// @initia/initia.proto/connect/oracle/v2/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: connect/oracle/v2/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.MsgRemoveCurrencyPairsDesc = exports.MsgAddCurrencyPairsDesc = exports.MsgDesc = exports.MsgClientImpl = exports.MsgRemoveCurrencyPairsResponse = exports.MsgRemoveCurrencyPairs = exports.MsgAddCurrencyPairsResponse = exports.MsgAddCurrencyPairs = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
const currency_pair_1 = require("../../types/v2/currency_pair");
exports.protobufPackage = "connect.oracle.v2";
function createBaseMsgAddCurrencyPairs() {
    return { authority: "", currencyPairs: [] };
}
exports.MsgAddCurrencyPairs = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        for (const v of message.currencyPairs) {
            currency_pair_1.CurrencyPair.encode(v, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgAddCurrencyPairs();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.currencyPairs.push(currency_pair_1.CurrencyPair.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            currencyPairs: gt.Array.isArray(object?.currencyPairs)
                ? object.currencyPairs.map((e) => currency_pair_1.CurrencyPair.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.currencyPairs?.length) {
            obj.currencyPairs = message.currencyPairs.map((e) => currency_pair_1.CurrencyPair.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgAddCurrencyPairs.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgAddCurrencyPairs();
        message.authority = object.authority ?? "";
        message.currencyPairs = object.currencyPairs?.map((e) => currency_pair_1.CurrencyPair.fromPartial(e)) || [];
        return message;
    },
};
function createBaseMsgAddCurrencyPairsResponse() {
    return {};
}
exports.MsgAddCurrencyPairsResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgAddCurrencyPairsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgAddCurrencyPairsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgAddCurrencyPairsResponse();
        return message;
    },
};
function createBaseMsgRemoveCurrencyPairs() {
    return { authority: "", currencyPairIds: [] };
}
exports.MsgRemoveCurrencyPairs = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        for (const v of message.currencyPairIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRemoveCurrencyPairs();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.currencyPairIds.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            currencyPairIds: gt.Array.isArray(object?.currencyPairIds)
                ? object.currencyPairIds.map((e) => gt.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.currencyPairIds?.length) {
            obj.currencyPairIds = message.currencyPairIds;
        }
        return obj;
    },
    create(base) {
        return exports.MsgRemoveCurrencyPairs.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgRemoveCurrencyPairs();
        message.authority = object.authority ?? "";
        message.currencyPairIds = object.currencyPairIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgRemoveCurrencyPairsResponse() {
    return {};
}
exports.MsgRemoveCurrencyPairsResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRemoveCurrencyPairsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgRemoveCurrencyPairsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgRemoveCurrencyPairsResponse();
        return message;
    },
};
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.AddCurrencyPairs = this.AddCurrencyPairs.bind(this);
        this.RemoveCurrencyPairs = this.RemoveCurrencyPairs.bind(this);
    }
    AddCurrencyPairs(request, metadata) {
        return this.rpc.unary(exports.MsgAddCurrencyPairsDesc, exports.MsgAddCurrencyPairs.fromPartial(request), metadata);
    }
    RemoveCurrencyPairs(request, metadata) {
        return this.rpc.unary(exports.MsgRemoveCurrencyPairsDesc, exports.MsgRemoveCurrencyPairs.fromPartial(request), metadata);
    }
}
exports.MsgClientImpl = MsgClientImpl;
exports.MsgDesc = { serviceName: "connect.oracle.v2.Msg" };
exports.MsgAddCurrencyPairsDesc = {
    methodName: "AddCurrencyPairs",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgAddCurrencyPairs.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgAddCurrencyPairsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgRemoveCurrencyPairsDesc = {
    methodName: "RemoveCurrencyPairs",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgRemoveCurrencyPairs.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgRemoveCurrencyPairsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
