// @initia/opinit.proto/opinit/opchild/v1/genesis.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v5.29.3
// source: opinit/opchild/v1/genesis.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.LastValidatorPower = exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const types_1 = require("./types");
exports.protobufPackage = "opinit.opchild.v1";
function createBaseGenesisState() {
    return {
        params: undefined,
        lastValidatorPowers: [],
        validators: [],
        nextL2Sequence: 0n,
        nextL1Sequence: 0n,
        bridgeInfo: undefined,
        exported: false,
        denomPairs: [],
        migrationInfos: [],
    };
}
exports.GenesisState = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.params !== undefined) {
            types_1.Params.encode(message.params, writer.uint32(10).fork()).join();
        }
        for (const v of message.lastValidatorPowers) {
            exports.LastValidatorPower.encode(v, writer.uint32(18).fork()).join();
        }
        for (const v of message.validators) {
            types_1.Validator.encode(v, writer.uint32(26).fork()).join();
        }
        if (message.nextL2Sequence !== 0n) {
            if (BigInt.asUintN(64, message.nextL2Sequence) !== message.nextL2Sequence) {
                throw new gt.Error("value provided for field message.nextL2Sequence of type uint64 too large");
            }
            writer.uint32(32).uint64(message.nextL2Sequence);
        }
        if (message.nextL1Sequence !== 0n) {
            if (BigInt.asUintN(64, message.nextL1Sequence) !== message.nextL1Sequence) {
                throw new gt.Error("value provided for field message.nextL1Sequence of type uint64 too large");
            }
            writer.uint32(40).uint64(message.nextL1Sequence);
        }
        if (message.bridgeInfo !== undefined) {
            types_1.BridgeInfo.encode(message.bridgeInfo, writer.uint32(50).fork()).join();
        }
        if (message.exported !== false) {
            writer.uint32(56).bool(message.exported);
        }
        for (const v of message.denomPairs) {
            types_1.DenomPair.encode(v, writer.uint32(66).fork()).join();
        }
        for (const v of message.migrationInfos) {
            types_1.MigrationInfo.encode(v, writer.uint32(74).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.params = types_1.Params.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.lastValidatorPowers.push(exports.LastValidatorPower.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.validators.push(types_1.Validator.decode(reader, reader.uint32()));
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.nextL2Sequence = reader.uint64();
                    continue;
                }
                case 5: {
                    if (tag !== 40) {
                        break;
                    }
                    message.nextL1Sequence = reader.uint64();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.bridgeInfo = types_1.BridgeInfo.decode(reader, reader.uint32());
                    continue;
                }
                case 7: {
                    if (tag !== 56) {
                        break;
                    }
                    message.exported = reader.bool();
                    continue;
                }
                case 8: {
                    if (tag !== 66) {
                        break;
                    }
                    message.denomPairs.push(types_1.DenomPair.decode(reader, reader.uint32()));
                    continue;
                }
                case 9: {
                    if (tag !== 74) {
                        break;
                    }
                    message.migrationInfos.push(types_1.MigrationInfo.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? types_1.Params.fromJSON(object.params) : undefined,
            lastValidatorPowers: gt.Array.isArray(object?.lastValidatorPowers)
                ? object.lastValidatorPowers.map((e) => exports.LastValidatorPower.fromJSON(e))
                : [],
            validators: gt.Array.isArray(object?.validators) ? object.validators.map((e) => types_1.Validator.fromJSON(e)) : [],
            nextL2Sequence: isSet(object.nextL2Sequence) ? BigInt(object.nextL2Sequence) : 0n,
            nextL1Sequence: isSet(object.nextL1Sequence) ? BigInt(object.nextL1Sequence) : 0n,
            bridgeInfo: isSet(object.bridgeInfo) ? types_1.BridgeInfo.fromJSON(object.bridgeInfo) : undefined,
            exported: isSet(object.exported) ? gt.Boolean(object.exported) : false,
            denomPairs: gt.Array.isArray(object?.denomPairs) ? object.denomPairs.map((e) => types_1.DenomPair.fromJSON(e)) : [],
            migrationInfos: gt.Array.isArray(object?.migrationInfos)
                ? object.migrationInfos.map((e) => types_1.MigrationInfo.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.params !== undefined) {
            obj.params = types_1.Params.toJSON(message.params);
        }
        if (message.lastValidatorPowers?.length) {
            obj.lastValidatorPowers = message.lastValidatorPowers.map((e) => exports.LastValidatorPower.toJSON(e));
        }
        if (message.validators?.length) {
            obj.validators = message.validators.map((e) => types_1.Validator.toJSON(e));
        }
        if (message.nextL2Sequence !== 0n) {
            obj.nextL2Sequence = message.nextL2Sequence.toString();
        }
        if (message.nextL1Sequence !== 0n) {
            obj.nextL1Sequence = message.nextL1Sequence.toString();
        }
        if (message.bridgeInfo !== undefined) {
            obj.bridgeInfo = types_1.BridgeInfo.toJSON(message.bridgeInfo);
        }
        if (message.exported !== false) {
            obj.exported = message.exported;
        }
        if (message.denomPairs?.length) {
            obj.denomPairs = message.denomPairs.map((e) => types_1.DenomPair.toJSON(e));
        }
        if (message.migrationInfos?.length) {
            obj.migrationInfos = message.migrationInfos.map((e) => types_1.MigrationInfo.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.params = (object.params !== undefined && object.params !== null)
            ? types_1.Params.fromPartial(object.params)
            : undefined;
        message.lastValidatorPowers = object.lastValidatorPowers?.map((e) => exports.LastValidatorPower.fromPartial(e)) || [];
        message.validators = object.validators?.map((e) => types_1.Validator.fromPartial(e)) || [];
        message.nextL2Sequence = object.nextL2Sequence ?? 0n;
        message.nextL1Sequence = object.nextL1Sequence ?? 0n;
        message.bridgeInfo = (object.bridgeInfo !== undefined && object.bridgeInfo !== null)
            ? types_1.BridgeInfo.fromPartial(object.bridgeInfo)
            : undefined;
        message.exported = object.exported ?? false;
        message.denomPairs = object.denomPairs?.map((e) => types_1.DenomPair.fromPartial(e)) || [];
        message.migrationInfos = object.migrationInfos?.map((e) => types_1.MigrationInfo.fromPartial(e)) || [];
        return message;
    },
};
function createBaseLastValidatorPower() {
    return { address: "", power: 0n };
}
exports.LastValidatorPower = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.power !== 0n) {
            if (BigInt.asIntN(64, message.power) !== message.power) {
                throw new gt.Error("value provided for field message.power of type int64 too large");
            }
            writer.uint32(16).int64(message.power);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLastValidatorPower();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.power = reader.int64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            power: isSet(object.power) ? BigInt(object.power) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.power !== 0n) {
            obj.power = message.power.toString();
        }
        return obj;
    },
    create(base) {
        return exports.LastValidatorPower.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseLastValidatorPower();
        message.address = object.address ?? "";
        message.power = object.power ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
