// @initia/initia.proto/noble/forwarding/v1/account.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: noble/forwarding/v1/account.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForwardingPubKey = exports.ForwardingAccount = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const auth_1 = require("../../../cosmos/auth/v1beta1/auth");
exports.protobufPackage = "noble.forwarding.v1";
function createBaseForwardingAccount() {
    return { baseAccount: undefined, channel: "", recipient: "", createdAt: 0n, fallback: "" };
}
exports.ForwardingAccount = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.baseAccount !== undefined) {
            auth_1.BaseAccount.encode(message.baseAccount, writer.uint32(10).fork()).join();
        }
        if (message.channel !== "") {
            writer.uint32(18).string(message.channel);
        }
        if (message.recipient !== "") {
            writer.uint32(26).string(message.recipient);
        }
        if (message.createdAt !== 0n) {
            if (BigInt.asIntN(64, message.createdAt) !== message.createdAt) {
                throw new gt.Error("value provided for field message.createdAt of type int64 too large");
            }
            writer.uint32(32).int64(message.createdAt);
        }
        if (message.fallback !== "") {
            writer.uint32(42).string(message.fallback);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseForwardingAccount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.baseAccount = auth_1.BaseAccount.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.channel = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.recipient = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.createdAt = reader.int64();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.fallback = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            baseAccount: isSet(object.baseAccount) ? auth_1.BaseAccount.fromJSON(object.baseAccount) : undefined,
            channel: isSet(object.channel) ? gt.String(object.channel) : "",
            recipient: isSet(object.recipient) ? gt.String(object.recipient) : "",
            createdAt: isSet(object.createdAt) ? BigInt(object.createdAt) : 0n,
            fallback: isSet(object.fallback) ? gt.String(object.fallback) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.baseAccount !== undefined) {
            obj.baseAccount = auth_1.BaseAccount.toJSON(message.baseAccount);
        }
        if (message.channel !== "") {
            obj.channel = message.channel;
        }
        if (message.recipient !== "") {
            obj.recipient = message.recipient;
        }
        if (message.createdAt !== 0n) {
            obj.createdAt = message.createdAt.toString();
        }
        if (message.fallback !== "") {
            obj.fallback = message.fallback;
        }
        return obj;
    },
    create(base) {
        return exports.ForwardingAccount.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseForwardingAccount();
        message.baseAccount = (object.baseAccount !== undefined && object.baseAccount !== null)
            ? auth_1.BaseAccount.fromPartial(object.baseAccount)
            : undefined;
        message.channel = object.channel ?? "";
        message.recipient = object.recipient ?? "";
        message.createdAt = object.createdAt ?? 0n;
        message.fallback = object.fallback ?? "";
        return message;
    },
};
function createBaseForwardingPubKey() {
    return { key: new Uint8Array(0) };
}
exports.ForwardingPubKey = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.key.length !== 0) {
            writer.uint32(10).bytes(message.key);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseForwardingPubKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { key: isSet(object.key) ? bytesFromBase64(object.key) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.key.length !== 0) {
            obj.key = base64FromBytes(message.key);
        }
        return obj;
    },
    create(base) {
        return exports.ForwardingPubKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseForwardingPubKey();
        message.key = object.key ?? new Uint8Array(0);
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
