// @initia/initia.proto/noble/forwarding/v1/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: noble/forwarding/v1/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.MsgSetMemoDesc = exports.MsgSetAllowedDenomsDesc = exports.MsgClearAccountDesc = exports.MsgRegisterAccountDesc = exports.MsgDesc = exports.MsgClientImpl = exports.MsgSetMemoResponse = exports.MsgSetMemo = exports.MsgSetAllowedDenomsResponse = exports.MsgSetAllowedDenoms = exports.MsgClearAccountResponse = exports.MsgClearAccount = exports.MsgRegisterAccountResponse = exports.MsgRegisterAccount = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
const memo_1 = require("./memo");
exports.protobufPackage = "noble.forwarding.v1";
function createBaseMsgRegisterAccount() {
    return { signer: "", recipient: "", channel: "", fallback: "", memos: [] };
}
exports.MsgRegisterAccount = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        if (message.recipient !== "") {
            writer.uint32(18).string(message.recipient);
        }
        if (message.channel !== "") {
            writer.uint32(26).string(message.channel);
        }
        if (message.fallback !== "") {
            writer.uint32(34).string(message.fallback);
        }
        for (const v of message.memos) {
            memo_1.MemoEntry.encode(v, writer.uint32(42).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRegisterAccount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.recipient = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.channel = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.fallback = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.memos.push(memo_1.MemoEntry.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            recipient: isSet(object.recipient) ? gt.String(object.recipient) : "",
            channel: isSet(object.channel) ? gt.String(object.channel) : "",
            fallback: isSet(object.fallback) ? gt.String(object.fallback) : "",
            memos: gt.Array.isArray(object?.memos) ? object.memos.map((e) => memo_1.MemoEntry.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.recipient !== "") {
            obj.recipient = message.recipient;
        }
        if (message.channel !== "") {
            obj.channel = message.channel;
        }
        if (message.fallback !== "") {
            obj.fallback = message.fallback;
        }
        if (message.memos?.length) {
            obj.memos = message.memos.map((e) => memo_1.MemoEntry.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgRegisterAccount.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgRegisterAccount();
        message.signer = object.signer ?? "";
        message.recipient = object.recipient ?? "";
        message.channel = object.channel ?? "";
        message.fallback = object.fallback ?? "";
        message.memos = object.memos?.map((e) => memo_1.MemoEntry.fromPartial(e)) || [];
        return message;
    },
};
function createBaseMsgRegisterAccountResponse() {
    return { address: "" };
}
exports.MsgRegisterAccountResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRegisterAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { address: isSet(object.address) ? gt.String(object.address) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create(base) {
        return exports.MsgRegisterAccountResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgRegisterAccountResponse();
        message.address = object.address ?? "";
        return message;
    },
};
function createBaseMsgClearAccount() {
    return { signer: "", address: "", fallback: false };
}
exports.MsgClearAccount = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        if (message.address !== "") {
            writer.uint32(18).string(message.address);
        }
        if (message.fallback !== false) {
            writer.uint32(24).bool(message.fallback);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgClearAccount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.fallback = reader.bool();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            address: isSet(object.address) ? gt.String(object.address) : "",
            fallback: isSet(object.fallback) ? gt.Boolean(object.fallback) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.fallback !== false) {
            obj.fallback = message.fallback;
        }
        return obj;
    },
    create(base) {
        return exports.MsgClearAccount.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgClearAccount();
        message.signer = object.signer ?? "";
        message.address = object.address ?? "";
        message.fallback = object.fallback ?? false;
        return message;
    },
};
function createBaseMsgClearAccountResponse() {
    return {};
}
exports.MsgClearAccountResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgClearAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgClearAccountResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgClearAccountResponse();
        return message;
    },
};
function createBaseMsgSetAllowedDenoms() {
    return { signer: "", denoms: [] };
}
exports.MsgSetAllowedDenoms = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        for (const v of message.denoms) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetAllowedDenoms();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denoms.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            denoms: gt.Array.isArray(object?.denoms) ? object.denoms.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.denoms?.length) {
            obj.denoms = message.denoms;
        }
        return obj;
    },
    create(base) {
        return exports.MsgSetAllowedDenoms.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgSetAllowedDenoms();
        message.signer = object.signer ?? "";
        message.denoms = object.denoms?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgSetAllowedDenomsResponse() {
    return {};
}
exports.MsgSetAllowedDenomsResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetAllowedDenomsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgSetAllowedDenomsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgSetAllowedDenomsResponse();
        return message;
    },
};
function createBaseMsgSetMemo() {
    return { signer: "", recipient: "", channel: "", fallback: "", denom: "", memo: "" };
}
exports.MsgSetMemo = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        if (message.recipient !== "") {
            writer.uint32(18).string(message.recipient);
        }
        if (message.channel !== "") {
            writer.uint32(26).string(message.channel);
        }
        if (message.fallback !== "") {
            writer.uint32(34).string(message.fallback);
        }
        if (message.denom !== "") {
            writer.uint32(42).string(message.denom);
        }
        if (message.memo !== "") {
            writer.uint32(50).string(message.memo);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMemo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.recipient = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.channel = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.fallback = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.memo = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            recipient: isSet(object.recipient) ? gt.String(object.recipient) : "",
            channel: isSet(object.channel) ? gt.String(object.channel) : "",
            fallback: isSet(object.fallback) ? gt.String(object.fallback) : "",
            denom: isSet(object.denom) ? gt.String(object.denom) : "",
            memo: isSet(object.memo) ? gt.String(object.memo) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.recipient !== "") {
            obj.recipient = message.recipient;
        }
        if (message.channel !== "") {
            obj.channel = message.channel;
        }
        if (message.fallback !== "") {
            obj.fallback = message.fallback;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        if (message.memo !== "") {
            obj.memo = message.memo;
        }
        return obj;
    },
    create(base) {
        return exports.MsgSetMemo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgSetMemo();
        message.signer = object.signer ?? "";
        message.recipient = object.recipient ?? "";
        message.channel = object.channel ?? "";
        message.fallback = object.fallback ?? "";
        message.denom = object.denom ?? "";
        message.memo = object.memo ?? "";
        return message;
    },
};
function createBaseMsgSetMemoResponse() {
    return {};
}
exports.MsgSetMemoResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMemoResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgSetMemoResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgSetMemoResponse();
        return message;
    },
};
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.RegisterAccount = this.RegisterAccount.bind(this);
        this.ClearAccount = this.ClearAccount.bind(this);
        this.SetAllowedDenoms = this.SetAllowedDenoms.bind(this);
        this.SetMemo = this.SetMemo.bind(this);
    }
    RegisterAccount(request, metadata) {
        return this.rpc.unary(exports.MsgRegisterAccountDesc, exports.MsgRegisterAccount.fromPartial(request), metadata);
    }
    ClearAccount(request, metadata) {
        return this.rpc.unary(exports.MsgClearAccountDesc, exports.MsgClearAccount.fromPartial(request), metadata);
    }
    SetAllowedDenoms(request, metadata) {
        return this.rpc.unary(exports.MsgSetAllowedDenomsDesc, exports.MsgSetAllowedDenoms.fromPartial(request), metadata);
    }
    SetMemo(request, metadata) {
        return this.rpc.unary(exports.MsgSetMemoDesc, exports.MsgSetMemo.fromPartial(request), metadata);
    }
}
exports.MsgClientImpl = MsgClientImpl;
exports.MsgDesc = { serviceName: "noble.forwarding.v1.Msg" };
exports.MsgRegisterAccountDesc = {
    methodName: "RegisterAccount",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgRegisterAccount.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgRegisterAccountResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgClearAccountDesc = {
    methodName: "ClearAccount",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgClearAccount.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgClearAccountResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgSetAllowedDenomsDesc = {
    methodName: "SetAllowedDenoms",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgSetAllowedDenoms.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgSetAllowedDenomsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgSetMemoDesc = {
    methodName: "SetMemo",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgSetMemo.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgSetMemoResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
