// @initia/initia.proto/celestia/blob/v1/blob.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: celestia/blob/v1/blob.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlobTx = exports.Blob = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "celestia.core.v1.blob";
function createBaseBlob() {
    return { namespaceId: new Uint8Array(0), data: new Uint8Array(0), shareVersion: 0, namespaceVersion: 0 };
}
exports.Blob = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.namespaceId.length !== 0) {
            writer.uint32(10).bytes(message.namespaceId);
        }
        if (message.data.length !== 0) {
            writer.uint32(18).bytes(message.data);
        }
        if (message.shareVersion !== 0) {
            writer.uint32(24).uint32(message.shareVersion);
        }
        if (message.namespaceVersion !== 0) {
            writer.uint32(32).uint32(message.namespaceVersion);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBlob();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.namespaceId = reader.bytes();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.shareVersion = reader.uint32();
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.namespaceVersion = reader.uint32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            namespaceId: isSet(object.namespaceId) ? bytesFromBase64(object.namespaceId) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            shareVersion: isSet(object.shareVersion) ? gt.Number(object.shareVersion) : 0,
            namespaceVersion: isSet(object.namespaceVersion) ? gt.Number(object.namespaceVersion) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.namespaceId.length !== 0) {
            obj.namespaceId = base64FromBytes(message.namespaceId);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.shareVersion !== 0) {
            obj.shareVersion = Math.round(message.shareVersion);
        }
        if (message.namespaceVersion !== 0) {
            obj.namespaceVersion = Math.round(message.namespaceVersion);
        }
        return obj;
    },
    create(base) {
        return exports.Blob.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBlob();
        message.namespaceId = object.namespaceId ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.shareVersion = object.shareVersion ?? 0;
        message.namespaceVersion = object.namespaceVersion ?? 0;
        return message;
    },
};
function createBaseBlobTx() {
    return { tx: new Uint8Array(0), blobs: [], typeId: "" };
}
exports.BlobTx = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.tx.length !== 0) {
            writer.uint32(10).bytes(message.tx);
        }
        for (const v of message.blobs) {
            exports.Blob.encode(v, writer.uint32(18).fork()).join();
        }
        if (message.typeId !== "") {
            writer.uint32(26).string(message.typeId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBlobTx();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.tx = reader.bytes();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.blobs.push(exports.Blob.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.typeId = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tx: isSet(object.tx) ? bytesFromBase64(object.tx) : new Uint8Array(0),
            blobs: gt.Array.isArray(object?.blobs) ? object.blobs.map((e) => exports.Blob.fromJSON(e)) : [],
            typeId: isSet(object.typeId) ? gt.String(object.typeId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.tx.length !== 0) {
            obj.tx = base64FromBytes(message.tx);
        }
        if (message.blobs?.length) {
            obj.blobs = message.blobs.map((e) => exports.Blob.toJSON(e));
        }
        if (message.typeId !== "") {
            obj.typeId = message.typeId;
        }
        return obj;
    },
    create(base) {
        return exports.BlobTx.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBlobTx();
        message.tx = object.tx ?? new Uint8Array(0);
        message.blobs = object.blobs?.map((e) => exports.Blob.fromPartial(e)) || [];
        message.typeId = object.typeId ?? "";
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
