// @initia/initia.proto/celestia/blob/v1/event.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: celestia/blob/v1/event.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventPayForBlobs = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "celestia.blob.v1";
function createBaseEventPayForBlobs() {
    return { signer: "", blobSizes: [], namespaces: [] };
}
exports.EventPayForBlobs = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        writer.uint32(18).fork();
        for (const v of message.blobSizes) {
            writer.uint32(v);
        }
        writer.join();
        for (const v of message.namespaces) {
            writer.uint32(26).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventPayForBlobs();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag === 16) {
                        message.blobSizes.push(reader.uint32());
                        continue;
                    }
                    if (tag === 18) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.blobSizes.push(reader.uint32());
                        }
                        continue;
                    }
                    break;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.namespaces.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            blobSizes: gt.Array.isArray(object?.blobSizes) ? object.blobSizes.map((e) => gt.Number(e)) : [],
            namespaces: gt.Array.isArray(object?.namespaces) ? object.namespaces.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.blobSizes?.length) {
            obj.blobSizes = message.blobSizes.map((e) => Math.round(e));
        }
        if (message.namespaces?.length) {
            obj.namespaces = message.namespaces.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.EventPayForBlobs.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventPayForBlobs();
        message.signer = object.signer ?? "";
        message.blobSizes = object.blobSizes?.map((e) => e) || [];
        message.namespaces = object.namespaces?.map((e) => e) || [];
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
