// @initia/initia.proto/celestia/blob/v1/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: celestia/blob/v1/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.MsgPayForBlobsDesc = exports.MsgDesc = exports.MsgClientImpl = exports.MsgPayForBlobsResponse = exports.MsgPayForBlobs = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
exports.protobufPackage = "celestia.blob.v1";
function createBaseMsgPayForBlobs() {
    return { signer: "", namespaces: [], blobSizes: [], shareCommitments: [], shareVersions: [] };
}
exports.MsgPayForBlobs = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.signer !== "") {
            writer.uint32(10).string(message.signer);
        }
        for (const v of message.namespaces) {
            writer.uint32(18).bytes(v);
        }
        writer.uint32(26).fork();
        for (const v of message.blobSizes) {
            writer.uint32(v);
        }
        writer.join();
        for (const v of message.shareCommitments) {
            writer.uint32(34).bytes(v);
        }
        writer.uint32(66).fork();
        for (const v of message.shareVersions) {
            writer.uint32(v);
        }
        writer.join();
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgPayForBlobs();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.signer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.namespaces.push(reader.bytes());
                    continue;
                }
                case 3: {
                    if (tag === 24) {
                        message.blobSizes.push(reader.uint32());
                        continue;
                    }
                    if (tag === 26) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.blobSizes.push(reader.uint32());
                        }
                        continue;
                    }
                    break;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.shareCommitments.push(reader.bytes());
                    continue;
                }
                case 8: {
                    if (tag === 64) {
                        message.shareVersions.push(reader.uint32());
                        continue;
                    }
                    if (tag === 66) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.shareVersions.push(reader.uint32());
                        }
                        continue;
                    }
                    break;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signer: isSet(object.signer) ? gt.String(object.signer) : "",
            namespaces: gt.Array.isArray(object?.namespaces) ? object.namespaces.map((e) => bytesFromBase64(e)) : [],
            blobSizes: gt.Array.isArray(object?.blobSizes) ? object.blobSizes.map((e) => gt.Number(e)) : [],
            shareCommitments: gt.Array.isArray(object?.shareCommitments)
                ? object.shareCommitments.map((e) => bytesFromBase64(e))
                : [],
            shareVersions: gt.Array.isArray(object?.shareVersions) ? object.shareVersions.map((e) => gt.Number(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signer !== "") {
            obj.signer = message.signer;
        }
        if (message.namespaces?.length) {
            obj.namespaces = message.namespaces.map((e) => base64FromBytes(e));
        }
        if (message.blobSizes?.length) {
            obj.blobSizes = message.blobSizes.map((e) => Math.round(e));
        }
        if (message.shareCommitments?.length) {
            obj.shareCommitments = message.shareCommitments.map((e) => base64FromBytes(e));
        }
        if (message.shareVersions?.length) {
            obj.shareVersions = message.shareVersions.map((e) => Math.round(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgPayForBlobs.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgPayForBlobs();
        message.signer = object.signer ?? "";
        message.namespaces = object.namespaces?.map((e) => e) || [];
        message.blobSizes = object.blobSizes?.map((e) => e) || [];
        message.shareCommitments = object.shareCommitments?.map((e) => e) || [];
        message.shareVersions = object.shareVersions?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgPayForBlobsResponse() {
    return {};
}
exports.MsgPayForBlobsResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgPayForBlobsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgPayForBlobsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgPayForBlobsResponse();
        return message;
    },
};
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.PayForBlobs = this.PayForBlobs.bind(this);
    }
    PayForBlobs(request, metadata) {
        return this.rpc.unary(exports.MsgPayForBlobsDesc, exports.MsgPayForBlobs.fromPartial(request), metadata);
    }
}
exports.MsgClientImpl = MsgClientImpl;
exports.MsgDesc = { serviceName: "celestia.blob.v1.Msg" };
exports.MsgPayForBlobsDesc = {
    methodName: "PayForBlobs",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgPayForBlobs.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgPayForBlobsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
