// @initia/initia.proto/initia/move/v1/authz.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/move/v1/authz.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecuteAuthorization = exports.PublishAuthorization = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const types_1 = require("./types");
exports.protobufPackage = "initia.move.v1";
function createBasePublishAuthorization() {
    return { moduleNames: [] };
}
exports.PublishAuthorization = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.moduleNames) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.moduleNames.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            moduleNames: gt.Array.isArray(object?.moduleNames) ? object.moduleNames.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.moduleNames?.length) {
            obj.moduleNames = message.moduleNames;
        }
        return obj;
    },
    create(base) {
        return exports.PublishAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublishAuthorization();
        message.moduleNames = object.moduleNames?.map((e) => e) || [];
        return message;
    },
};
function createBaseExecuteAuthorization() {
    return { items: [] };
}
exports.ExecuteAuthorization = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.items) {
            types_1.ExecuteAuthorizationItem.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseExecuteAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.items.push(types_1.ExecuteAuthorizationItem.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            items: gt.Array.isArray(object?.items) ? object.items.map((e) => types_1.ExecuteAuthorizationItem.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.items?.length) {
            obj.items = message.items.map((e) => types_1.ExecuteAuthorizationItem.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ExecuteAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseExecuteAuthorization();
        message.items = object.items?.map((e) => types_1.ExecuteAuthorizationItem.fromPartial(e)) || [];
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
