// @initia/initia.proto/initia/move/v1/query.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/move/v1/query.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryViewBatchDesc = exports.QueryViewDesc = exports.QueryLegacyViewDesc = exports.QueryTableEntriesDesc = exports.QueryTableEntryDesc = exports.QueryTableInfoDesc = exports.QueryResourcesDesc = exports.QueryResourceDesc = exports.QueryModulesDesc = exports.QueryModuleDesc = exports.QueryDesc = exports.QueryClientImpl = exports.QueryDexPairsResponse = exports.QueryDexPairsRequest = exports.QueryDexPairResponse = exports.QueryDexPairRequest = exports.QueryDenomResponse = exports.QueryDenomRequest = exports.QueryMetadataResponse = exports.QueryMetadataRequest = exports.QueryParamsResponse = exports.QueryParamsRequest = exports.QueryScriptABIResponse = exports.QueryScriptABIRequest = exports.VMEvent = exports.QueryViewJSONBatchResponse = exports.QueryViewJSONBatchRequest = exports.QueryViewJSONResponse = exports.QueryViewJSONRequest = exports.QueryViewBatchResponse = exports.QueryViewBatchRequest = exports.QueryViewResponse = exports.QueryViewRequest = exports.QueryLegacyViewResponse = exports.QueryLegacyViewRequest = exports.QueryTableEntriesResponse = exports.QueryTableEntriesRequest = exports.QueryTableEntryResponse = exports.QueryTableEntryRequest = exports.QueryTableInfoResponse = exports.QueryTableInfoRequest = exports.QueryResourcesResponse = exports.QueryResourcesRequest = exports.QueryResourceResponse = exports.QueryResourceRequest = exports.QueryModulesResponse = exports.QueryModulesRequest = exports.QueryModuleResponse = exports.QueryModuleRequest = exports.protobufPackage = void 0;
exports.GrpcWebError = exports.GrpcWebImpl = exports.QueryDexPairsDesc = exports.QueryDexPairDesc = exports.QueryDenomDesc = exports.QueryMetadataDesc = exports.QueryParamsDesc = exports.QueryScriptABIDesc = exports.QueryViewJSONBatchDesc = exports.QueryViewJSONDesc = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
const pagination_1 = require("../../../cosmos/base/query/v1beta1/pagination");
const types_1 = require("./types");
exports.protobufPackage = "initia.move.v1";
function createBaseQueryModuleRequest() {
    return { address: "", moduleName: "" };
}
exports.QueryModuleRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.moduleName !== "") {
            writer.uint32(18).string(message.moduleName);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryModuleRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        return obj;
    },
    create(base) {
        return exports.QueryModuleRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryModuleRequest();
        message.address = object.address ?? "";
        message.moduleName = object.moduleName ?? "";
        return message;
    },
};
function createBaseQueryModuleResponse() {
    return { module: undefined };
}
exports.QueryModuleResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.module !== undefined) {
            types_1.Module.encode(message.module, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryModuleResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.module = types_1.Module.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { module: isSet(object.module) ? types_1.Module.fromJSON(object.module) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.module !== undefined) {
            obj.module = types_1.Module.toJSON(message.module);
        }
        return obj;
    },
    create(base) {
        return exports.QueryModuleResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryModuleResponse();
        message.module = (object.module !== undefined && object.module !== null)
            ? types_1.Module.fromPartial(object.module)
            : undefined;
        return message;
    },
};
function createBaseQueryModulesRequest() {
    return { address: "", pagination: undefined };
}
exports.QueryModulesRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryModulesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            pagination: isSet(object.pagination) ? pagination_1.PageRequest.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageRequest.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryModulesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryModulesRequest();
        message.address = object.address ?? "";
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageRequest.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryModulesResponse() {
    return { modules: [], pagination: undefined };
}
exports.QueryModulesResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.modules) {
            types_1.Module.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryModulesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.modules.push(types_1.Module.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            modules: gt.Array.isArray(object?.modules) ? object.modules.map((e) => types_1.Module.fromJSON(e)) : [],
            pagination: isSet(object.pagination) ? pagination_1.PageResponse.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.modules?.length) {
            obj.modules = message.modules.map((e) => types_1.Module.toJSON(e));
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageResponse.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryModulesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryModulesResponse();
        message.modules = object.modules?.map((e) => types_1.Module.fromPartial(e)) || [];
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageResponse.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryResourceRequest() {
    return { address: "", structTag: "" };
}
exports.QueryResourceRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.structTag !== "") {
            writer.uint32(18).string(message.structTag);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryResourceRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.structTag = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            structTag: isSet(object.structTag) ? gt.String(object.structTag) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.structTag !== "") {
            obj.structTag = message.structTag;
        }
        return obj;
    },
    create(base) {
        return exports.QueryResourceRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryResourceRequest();
        message.address = object.address ?? "";
        message.structTag = object.structTag ?? "";
        return message;
    },
};
function createBaseQueryResourceResponse() {
    return { resource: undefined };
}
exports.QueryResourceResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.resource !== undefined) {
            types_1.Resource.encode(message.resource, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryResourceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.resource = types_1.Resource.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { resource: isSet(object.resource) ? types_1.Resource.fromJSON(object.resource) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.resource !== undefined) {
            obj.resource = types_1.Resource.toJSON(message.resource);
        }
        return obj;
    },
    create(base) {
        return exports.QueryResourceResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryResourceResponse();
        message.resource = (object.resource !== undefined && object.resource !== null)
            ? types_1.Resource.fromPartial(object.resource)
            : undefined;
        return message;
    },
};
function createBaseQueryResourcesRequest() {
    return { address: "", pagination: undefined };
}
exports.QueryResourcesRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryResourcesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            pagination: isSet(object.pagination) ? pagination_1.PageRequest.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageRequest.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryResourcesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryResourcesRequest();
        message.address = object.address ?? "";
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageRequest.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryResourcesResponse() {
    return { resources: [], pagination: undefined };
}
exports.QueryResourcesResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.resources) {
            types_1.Resource.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryResourcesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.resources.push(types_1.Resource.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            resources: gt.Array.isArray(object?.resources) ? object.resources.map((e) => types_1.Resource.fromJSON(e)) : [],
            pagination: isSet(object.pagination) ? pagination_1.PageResponse.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.resources?.length) {
            obj.resources = message.resources.map((e) => types_1.Resource.toJSON(e));
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageResponse.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryResourcesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryResourcesResponse();
        message.resources = object.resources?.map((e) => types_1.Resource.fromPartial(e)) || [];
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageResponse.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryTableInfoRequest() {
    return { address: "" };
}
exports.QueryTableInfoRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableInfoRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { address: isSet(object.address) ? gt.String(object.address) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableInfoRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableInfoRequest();
        message.address = object.address ?? "";
        return message;
    },
};
function createBaseQueryTableInfoResponse() {
    return { tableInfo: undefined };
}
exports.QueryTableInfoResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.tableInfo !== undefined) {
            types_1.TableInfo.encode(message.tableInfo, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableInfoResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.tableInfo = types_1.TableInfo.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { tableInfo: isSet(object.tableInfo) ? types_1.TableInfo.fromJSON(object.tableInfo) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.tableInfo !== undefined) {
            obj.tableInfo = types_1.TableInfo.toJSON(message.tableInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableInfoResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableInfoResponse();
        message.tableInfo = (object.tableInfo !== undefined && object.tableInfo !== null)
            ? types_1.TableInfo.fromPartial(object.tableInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryTableEntryRequest() {
    return { address: "", keyBytes: new Uint8Array(0) };
}
exports.QueryTableEntryRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.keyBytes.length !== 0) {
            writer.uint32(18).bytes(message.keyBytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableEntryRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.keyBytes = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            keyBytes: isSet(object.keyBytes) ? bytesFromBase64(object.keyBytes) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.keyBytes.length !== 0) {
            obj.keyBytes = base64FromBytes(message.keyBytes);
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableEntryRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableEntryRequest();
        message.address = object.address ?? "";
        message.keyBytes = object.keyBytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseQueryTableEntryResponse() {
    return { tableEntry: undefined };
}
exports.QueryTableEntryResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.tableEntry !== undefined) {
            types_1.TableEntry.encode(message.tableEntry, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableEntryResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.tableEntry = types_1.TableEntry.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { tableEntry: isSet(object.tableEntry) ? types_1.TableEntry.fromJSON(object.tableEntry) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.tableEntry !== undefined) {
            obj.tableEntry = types_1.TableEntry.toJSON(message.tableEntry);
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableEntryResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableEntryResponse();
        message.tableEntry = (object.tableEntry !== undefined && object.tableEntry !== null)
            ? types_1.TableEntry.fromPartial(object.tableEntry)
            : undefined;
        return message;
    },
};
function createBaseQueryTableEntriesRequest() {
    return { address: "", pagination: undefined };
}
exports.QueryTableEntriesRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableEntriesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            pagination: isSet(object.pagination) ? pagination_1.PageRequest.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageRequest.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableEntriesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableEntriesRequest();
        message.address = object.address ?? "";
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageRequest.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryTableEntriesResponse() {
    return { tableEntries: [], pagination: undefined };
}
exports.QueryTableEntriesResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.tableEntries) {
            types_1.TableEntry.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTableEntriesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.tableEntries.push(types_1.TableEntry.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tableEntries: gt.Array.isArray(object?.tableEntries)
                ? object.tableEntries.map((e) => types_1.TableEntry.fromJSON(e))
                : [],
            pagination: isSet(object.pagination) ? pagination_1.PageResponse.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.tableEntries?.length) {
            obj.tableEntries = message.tableEntries.map((e) => types_1.TableEntry.toJSON(e));
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageResponse.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryTableEntriesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTableEntriesResponse();
        message.tableEntries = object.tableEntries?.map((e) => types_1.TableEntry.fromPartial(e)) || [];
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageResponse.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryLegacyViewRequest() {
    return { address: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.QueryLegacyViewRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.moduleName !== "") {
            writer.uint32(18).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(26).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(34).string(v);
        }
        for (const v of message.args) {
            writer.uint32(42).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryLegacyViewRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryLegacyViewRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryLegacyViewRequest();
        message.address = object.address ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseQueryLegacyViewResponse() {
    return { data: "", events: [], gasUsed: 0n };
}
exports.QueryLegacyViewResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.data !== "") {
            writer.uint32(10).string(message.data);
        }
        for (const v of message.events) {
            exports.VMEvent.encode(v, writer.uint32(18).fork()).join();
        }
        if (message.gasUsed !== 0n) {
            if (BigInt.asUintN(64, message.gasUsed) !== message.gasUsed) {
                throw new gt.Error("value provided for field message.gasUsed of type uint64 too large");
            }
            writer.uint32(24).uint64(message.gasUsed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryLegacyViewResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.events.push(exports.VMEvent.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.gasUsed = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            data: isSet(object.data) ? gt.String(object.data) : "",
            events: gt.Array.isArray(object?.events) ? object.events.map((e) => exports.VMEvent.fromJSON(e)) : [],
            gasUsed: isSet(object.gasUsed) ? BigInt(object.gasUsed) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.data !== "") {
            obj.data = message.data;
        }
        if (message.events?.length) {
            obj.events = message.events.map((e) => exports.VMEvent.toJSON(e));
        }
        if (message.gasUsed !== 0n) {
            obj.gasUsed = message.gasUsed.toString();
        }
        return obj;
    },
    create(base) {
        return exports.QueryLegacyViewResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryLegacyViewResponse();
        message.data = object.data ?? "";
        message.events = object.events?.map((e) => exports.VMEvent.fromPartial(e)) || [];
        message.gasUsed = object.gasUsed ?? 0n;
        return message;
    },
};
function createBaseQueryViewRequest() {
    return { address: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.QueryViewRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.moduleName !== "") {
            writer.uint32(18).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(26).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(34).string(v);
        }
        for (const v of message.args) {
            writer.uint32(42).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewRequest();
        message.address = object.address ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseQueryViewResponse() {
    return { data: "", events: [], gasUsed: 0n };
}
exports.QueryViewResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.data !== "") {
            writer.uint32(10).string(message.data);
        }
        for (const v of message.events) {
            exports.VMEvent.encode(v, writer.uint32(18).fork()).join();
        }
        if (message.gasUsed !== 0n) {
            if (BigInt.asUintN(64, message.gasUsed) !== message.gasUsed) {
                throw new gt.Error("value provided for field message.gasUsed of type uint64 too large");
            }
            writer.uint32(24).uint64(message.gasUsed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.events.push(exports.VMEvent.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.gasUsed = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            data: isSet(object.data) ? gt.String(object.data) : "",
            events: gt.Array.isArray(object?.events) ? object.events.map((e) => exports.VMEvent.fromJSON(e)) : [],
            gasUsed: isSet(object.gasUsed) ? BigInt(object.gasUsed) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.data !== "") {
            obj.data = message.data;
        }
        if (message.events?.length) {
            obj.events = message.events.map((e) => exports.VMEvent.toJSON(e));
        }
        if (message.gasUsed !== 0n) {
            obj.gasUsed = message.gasUsed.toString();
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewResponse();
        message.data = object.data ?? "";
        message.events = object.events?.map((e) => exports.VMEvent.fromPartial(e)) || [];
        message.gasUsed = object.gasUsed ?? 0n;
        return message;
    },
};
function createBaseQueryViewBatchRequest() {
    return { requests: [] };
}
exports.QueryViewBatchRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.requests) {
            exports.QueryViewRequest.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewBatchRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.QueryViewRequest.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: gt.Array.isArray(object?.requests) ? object.requests.map((e) => exports.QueryViewRequest.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.QueryViewRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewBatchRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewBatchRequest();
        message.requests = object.requests?.map((e) => exports.QueryViewRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBaseQueryViewBatchResponse() {
    return { responses: [] };
}
exports.QueryViewBatchResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.responses) {
            exports.QueryViewResponse.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewBatchResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.QueryViewResponse.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: gt.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.QueryViewResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.QueryViewResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewBatchResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewBatchResponse();
        message.responses = object.responses?.map((e) => exports.QueryViewResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseQueryViewJSONRequest() {
    return { address: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.QueryViewJSONRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.moduleName !== "") {
            writer.uint32(18).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(26).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(34).string(v);
        }
        for (const v of message.args) {
            writer.uint32(42).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewJSONRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.args.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args;
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewJSONRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewJSONRequest();
        message.address = object.address ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseQueryViewJSONResponse() {
    return { data: "", events: [], gasUsed: 0n };
}
exports.QueryViewJSONResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.data !== "") {
            writer.uint32(10).string(message.data);
        }
        for (const v of message.events) {
            exports.VMEvent.encode(v, writer.uint32(18).fork()).join();
        }
        if (message.gasUsed !== 0n) {
            if (BigInt.asUintN(64, message.gasUsed) !== message.gasUsed) {
                throw new gt.Error("value provided for field message.gasUsed of type uint64 too large");
            }
            writer.uint32(24).uint64(message.gasUsed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewJSONResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.events.push(exports.VMEvent.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.gasUsed = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            data: isSet(object.data) ? gt.String(object.data) : "",
            events: gt.Array.isArray(object?.events) ? object.events.map((e) => exports.VMEvent.fromJSON(e)) : [],
            gasUsed: isSet(object.gasUsed) ? BigInt(object.gasUsed) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.data !== "") {
            obj.data = message.data;
        }
        if (message.events?.length) {
            obj.events = message.events.map((e) => exports.VMEvent.toJSON(e));
        }
        if (message.gasUsed !== 0n) {
            obj.gasUsed = message.gasUsed.toString();
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewJSONResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewJSONResponse();
        message.data = object.data ?? "";
        message.events = object.events?.map((e) => exports.VMEvent.fromPartial(e)) || [];
        message.gasUsed = object.gasUsed ?? 0n;
        return message;
    },
};
function createBaseQueryViewJSONBatchRequest() {
    return { requests: [] };
}
exports.QueryViewJSONBatchRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.requests) {
            exports.QueryViewJSONRequest.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewJSONBatchRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.QueryViewJSONRequest.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: gt.Array.isArray(object?.requests)
                ? object.requests.map((e) => exports.QueryViewJSONRequest.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.QueryViewJSONRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewJSONBatchRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewJSONBatchRequest();
        message.requests = object.requests?.map((e) => exports.QueryViewJSONRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBaseQueryViewJSONBatchResponse() {
    return { responses: [] };
}
exports.QueryViewJSONBatchResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.responses) {
            exports.QueryViewJSONResponse.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryViewJSONBatchResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.QueryViewJSONResponse.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: gt.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.QueryViewJSONResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.QueryViewJSONResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryViewJSONBatchResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryViewJSONBatchResponse();
        message.responses = object.responses?.map((e) => exports.QueryViewJSONResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseVMEvent() {
    return { typeTag: "", data: "" };
}
exports.VMEvent = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.typeTag !== "") {
            writer.uint32(10).string(message.typeTag);
        }
        if (message.data !== "") {
            writer.uint32(18).string(message.data);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVMEvent();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.typeTag = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            typeTag: isSet(object.typeTag) ? gt.String(object.typeTag) : "",
            data: isSet(object.data) ? gt.String(object.data) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.typeTag !== "") {
            obj.typeTag = message.typeTag;
        }
        if (message.data !== "") {
            obj.data = message.data;
        }
        return obj;
    },
    create(base) {
        return exports.VMEvent.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVMEvent();
        message.typeTag = object.typeTag ?? "";
        message.data = object.data ?? "";
        return message;
    },
};
function createBaseQueryScriptABIRequest() {
    return { codeBytes: new Uint8Array(0) };
}
exports.QueryScriptABIRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.codeBytes.length !== 0) {
            writer.uint32(10).bytes(message.codeBytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryScriptABIRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.codeBytes = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { codeBytes: isSet(object.codeBytes) ? bytesFromBase64(object.codeBytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.codeBytes.length !== 0) {
            obj.codeBytes = base64FromBytes(message.codeBytes);
        }
        return obj;
    },
    create(base) {
        return exports.QueryScriptABIRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryScriptABIRequest();
        message.codeBytes = object.codeBytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseQueryScriptABIResponse() {
    return { abi: new Uint8Array(0) };
}
exports.QueryScriptABIResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.abi.length !== 0) {
            writer.uint32(10).bytes(message.abi);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryScriptABIResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.abi = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { abi: isSet(object.abi) ? bytesFromBase64(object.abi) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.abi.length !== 0) {
            obj.abi = base64FromBytes(message.abi);
        }
        return obj;
    },
    create(base) {
        return exports.QueryScriptABIResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryScriptABIResponse();
        message.abi = object.abi ?? new Uint8Array(0);
        return message;
    },
};
function createBaseQueryParamsRequest() {
    return {};
}
exports.QueryParamsRequest = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QueryParamsRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseQueryParamsRequest();
        return message;
    },
};
function createBaseQueryParamsResponse() {
    return { params: undefined };
}
exports.QueryParamsResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.params !== undefined) {
            types_1.Params.encode(message.params, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.params = types_1.Params.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { params: isSet(object.params) ? types_1.Params.fromJSON(object.params) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.params !== undefined) {
            obj.params = types_1.Params.toJSON(message.params);
        }
        return obj;
    },
    create(base) {
        return exports.QueryParamsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryParamsResponse();
        message.params = (object.params !== undefined && object.params !== null)
            ? types_1.Params.fromPartial(object.params)
            : undefined;
        return message;
    },
};
function createBaseQueryMetadataRequest() {
    return { denom: "" };
}
exports.QueryMetadataRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.denom !== "") {
            writer.uint32(10).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMetadataRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { denom: isSet(object.denom) ? gt.String(object.denom) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.QueryMetadataRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryMetadataRequest();
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseQueryMetadataResponse() {
    return { metadata: "" };
}
exports.QueryMetadataResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.metadata !== "") {
            writer.uint32(10).string(message.metadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMetadataResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.metadata = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { metadata: isSet(object.metadata) ? gt.String(object.metadata) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.metadata !== "") {
            obj.metadata = message.metadata;
        }
        return obj;
    },
    create(base) {
        return exports.QueryMetadataResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryMetadataResponse();
        message.metadata = object.metadata ?? "";
        return message;
    },
};
function createBaseQueryDenomRequest() {
    return { metadata: "" };
}
exports.QueryDenomRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.metadata !== "") {
            writer.uint32(10).string(message.metadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDenomRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.metadata = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { metadata: isSet(object.metadata) ? gt.String(object.metadata) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.metadata !== "") {
            obj.metadata = message.metadata;
        }
        return obj;
    },
    create(base) {
        return exports.QueryDenomRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDenomRequest();
        message.metadata = object.metadata ?? "";
        return message;
    },
};
function createBaseQueryDenomResponse() {
    return { denom: "" };
}
exports.QueryDenomResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.denom !== "") {
            writer.uint32(10).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDenomResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { denom: isSet(object.denom) ? gt.String(object.denom) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.QueryDenomResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDenomResponse();
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseQueryDexPairRequest() {
    return { metadataQuote: "" };
}
exports.QueryDexPairRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.metadataQuote !== "") {
            writer.uint32(10).string(message.metadataQuote);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDexPairRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.metadataQuote = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { metadataQuote: isSet(object.metadataQuote) ? gt.String(object.metadataQuote) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.metadataQuote !== "") {
            obj.metadataQuote = message.metadataQuote;
        }
        return obj;
    },
    create(base) {
        return exports.QueryDexPairRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDexPairRequest();
        message.metadataQuote = object.metadataQuote ?? "";
        return message;
    },
};
function createBaseQueryDexPairResponse() {
    return { dexPair: undefined };
}
exports.QueryDexPairResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.dexPair !== undefined) {
            types_1.DexPair.encode(message.dexPair, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDexPairResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.dexPair = types_1.DexPair.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { dexPair: isSet(object.dexPair) ? types_1.DexPair.fromJSON(object.dexPair) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.dexPair !== undefined) {
            obj.dexPair = types_1.DexPair.toJSON(message.dexPair);
        }
        return obj;
    },
    create(base) {
        return exports.QueryDexPairResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDexPairResponse();
        message.dexPair = (object.dexPair !== undefined && object.dexPair !== null)
            ? types_1.DexPair.fromPartial(object.dexPair)
            : undefined;
        return message;
    },
};
function createBaseQueryDexPairsRequest() {
    return { pagination: undefined };
}
exports.QueryDexPairsRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDexPairsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { pagination: isSet(object.pagination) ? pagination_1.PageRequest.fromJSON(object.pagination) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageRequest.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryDexPairsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDexPairsRequest();
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageRequest.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseQueryDexPairsResponse() {
    return { dexPairs: [], pagination: undefined };
}
exports.QueryDexPairsResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.dexPairs) {
            types_1.DexPair.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryDexPairsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.dexPairs.push(types_1.DexPair.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            dexPairs: gt.Array.isArray(object?.dexPairs) ? object.dexPairs.map((e) => types_1.DexPair.fromJSON(e)) : [],
            pagination: isSet(object.pagination) ? pagination_1.PageResponse.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.dexPairs?.length) {
            obj.dexPairs = message.dexPairs.map((e) => types_1.DexPair.toJSON(e));
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageResponse.toJSON(message.pagination);
        }
        return obj;
    },
    create(base) {
        return exports.QueryDexPairsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryDexPairsResponse();
        message.dexPairs = object.dexPairs?.map((e) => types_1.DexPair.fromPartial(e)) || [];
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageResponse.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.Module = this.Module.bind(this);
        this.Modules = this.Modules.bind(this);
        this.Resource = this.Resource.bind(this);
        this.Resources = this.Resources.bind(this);
        this.TableInfo = this.TableInfo.bind(this);
        this.TableEntry = this.TableEntry.bind(this);
        this.TableEntries = this.TableEntries.bind(this);
        this.LegacyView = this.LegacyView.bind(this);
        this.View = this.View.bind(this);
        this.ViewBatch = this.ViewBatch.bind(this);
        this.ViewJSON = this.ViewJSON.bind(this);
        this.ViewJSONBatch = this.ViewJSONBatch.bind(this);
        this.ScriptABI = this.ScriptABI.bind(this);
        this.Params = this.Params.bind(this);
        this.Metadata = this.Metadata.bind(this);
        this.Denom = this.Denom.bind(this);
        this.DexPair = this.DexPair.bind(this);
        this.DexPairs = this.DexPairs.bind(this);
    }
    Module(request, metadata) {
        return this.rpc.unary(exports.QueryModuleDesc, exports.QueryModuleRequest.fromPartial(request), metadata);
    }
    Modules(request, metadata) {
        return this.rpc.unary(exports.QueryModulesDesc, exports.QueryModulesRequest.fromPartial(request), metadata);
    }
    Resource(request, metadata) {
        return this.rpc.unary(exports.QueryResourceDesc, exports.QueryResourceRequest.fromPartial(request), metadata);
    }
    Resources(request, metadata) {
        return this.rpc.unary(exports.QueryResourcesDesc, exports.QueryResourcesRequest.fromPartial(request), metadata);
    }
    TableInfo(request, metadata) {
        return this.rpc.unary(exports.QueryTableInfoDesc, exports.QueryTableInfoRequest.fromPartial(request), metadata);
    }
    TableEntry(request, metadata) {
        return this.rpc.unary(exports.QueryTableEntryDesc, exports.QueryTableEntryRequest.fromPartial(request), metadata);
    }
    TableEntries(request, metadata) {
        return this.rpc.unary(exports.QueryTableEntriesDesc, exports.QueryTableEntriesRequest.fromPartial(request), metadata);
    }
    LegacyView(request, metadata) {
        return this.rpc.unary(exports.QueryLegacyViewDesc, exports.QueryLegacyViewRequest.fromPartial(request), metadata);
    }
    View(request, metadata) {
        return this.rpc.unary(exports.QueryViewDesc, exports.QueryViewRequest.fromPartial(request), metadata);
    }
    ViewBatch(request, metadata) {
        return this.rpc.unary(exports.QueryViewBatchDesc, exports.QueryViewBatchRequest.fromPartial(request), metadata);
    }
    ViewJSON(request, metadata) {
        return this.rpc.unary(exports.QueryViewJSONDesc, exports.QueryViewJSONRequest.fromPartial(request), metadata);
    }
    ViewJSONBatch(request, metadata) {
        return this.rpc.unary(exports.QueryViewJSONBatchDesc, exports.QueryViewJSONBatchRequest.fromPartial(request), metadata);
    }
    ScriptABI(request, metadata) {
        return this.rpc.unary(exports.QueryScriptABIDesc, exports.QueryScriptABIRequest.fromPartial(request), metadata);
    }
    Params(request, metadata) {
        return this.rpc.unary(exports.QueryParamsDesc, exports.QueryParamsRequest.fromPartial(request), metadata);
    }
    Metadata(request, metadata) {
        return this.rpc.unary(exports.QueryMetadataDesc, exports.QueryMetadataRequest.fromPartial(request), metadata);
    }
    Denom(request, metadata) {
        return this.rpc.unary(exports.QueryDenomDesc, exports.QueryDenomRequest.fromPartial(request), metadata);
    }
    DexPair(request, metadata) {
        return this.rpc.unary(exports.QueryDexPairDesc, exports.QueryDexPairRequest.fromPartial(request), metadata);
    }
    DexPairs(request, metadata) {
        return this.rpc.unary(exports.QueryDexPairsDesc, exports.QueryDexPairsRequest.fromPartial(request), metadata);
    }
}
exports.QueryClientImpl = QueryClientImpl;
exports.QueryDesc = { serviceName: "initia.move.v1.Query" };
exports.QueryModuleDesc = {
    methodName: "Module",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryModuleRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryModuleResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryModulesDesc = {
    methodName: "Modules",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryModulesRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryModulesResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryResourceDesc = {
    methodName: "Resource",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryResourceRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryResourceResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryResourcesDesc = {
    methodName: "Resources",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryResourcesRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryResourcesResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryTableInfoDesc = {
    methodName: "TableInfo",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryTableInfoRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryTableInfoResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryTableEntryDesc = {
    methodName: "TableEntry",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryTableEntryRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryTableEntryResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryTableEntriesDesc = {
    methodName: "TableEntries",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryTableEntriesRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryTableEntriesResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryLegacyViewDesc = {
    methodName: "LegacyView",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryLegacyViewRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryLegacyViewResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryViewDesc = {
    methodName: "View",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryViewRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryViewResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryViewBatchDesc = {
    methodName: "ViewBatch",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryViewBatchRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryViewBatchResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryViewJSONDesc = {
    methodName: "ViewJSON",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryViewJSONRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryViewJSONResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryViewJSONBatchDesc = {
    methodName: "ViewJSONBatch",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryViewJSONBatchRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryViewJSONBatchResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryScriptABIDesc = {
    methodName: "ScriptABI",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryScriptABIRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryScriptABIResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryParamsDesc = {
    methodName: "Params",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryParamsRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryParamsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryMetadataDesc = {
    methodName: "Metadata",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryMetadataRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryMetadataResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryDenomDesc = {
    methodName: "Denom",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryDenomRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryDenomResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryDexPairDesc = {
    methodName: "DexPair",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryDexPairRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryDexPairResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryDexPairsDesc = {
    methodName: "DexPairs",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryDexPairsRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryDexPairsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
