// @initia/initia.proto/initia/move/v1/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/move/v1/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.MsgUpdateParamsDesc = exports.MsgDelistGasPriceDesc = exports.MsgDelistStakingDesc = exports.MsgWhitelistGasPriceDesc = exports.MsgWhitelistStakingDesc = exports.MsgGovScriptJSONDesc = exports.MsgGovScriptDesc = exports.MsgGovExecuteJSONDesc = exports.MsgGovExecuteDesc = exports.MsgGovPublishDesc = exports.MsgScriptJSONDesc = exports.MsgScriptDesc = exports.MsgExecuteJSONDesc = exports.MsgExecuteDesc = exports.MsgPublishDesc = exports.MsgDesc = exports.MsgClientImpl = exports.MsgUpdateParamsResponse = exports.MsgUpdateParams = exports.MsgDelistGasPriceResponse = exports.MsgDelistGasPrice = exports.MsgDelistStakingResponse = exports.MsgDelistStaking = exports.MsgWhitelistGasPriceResponse = exports.MsgWhitelistGasPrice = exports.MsgWhitelistStakingResponse = exports.MsgWhitelistStaking = exports.MsgGovScriptJSONResponse = exports.MsgGovScriptJSON = exports.MsgGovScriptResponse = exports.MsgGovScript = exports.MsgGovExecuteJSONResponse = exports.MsgGovExecuteJSON = exports.MsgGovExecuteResponse = exports.MsgGovExecute = exports.MsgGovPublishResponse = exports.MsgGovPublish = exports.MsgScriptJSONResponse = exports.MsgScriptJSON = exports.MsgScriptResponse = exports.MsgScript = exports.MsgExecuteJSONResponse = exports.MsgExecuteJSON = exports.MsgExecuteResponse = exports.MsgExecute = exports.MsgPublishResponse = exports.MsgPublish = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
const types_1 = require("./types");
exports.protobufPackage = "initia.move.v1";
function createBaseMsgPublish() {
    return { sender: "", codeBytes: [], upgradePolicy: 0 };
}
exports.MsgPublish = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        for (const v of message.codeBytes) {
            writer.uint32(18).bytes(v);
        }
        if (message.upgradePolicy !== 0) {
            writer.uint32(24).int32(message.upgradePolicy);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgPublish();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.codeBytes.push(reader.bytes());
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.upgradePolicy = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: gt.Array.isArray(object?.codeBytes) ? object.codeBytes.map((e) => bytesFromBase64(e)) : [],
            upgradePolicy: isSet(object.upgradePolicy) ? (0, types_1.upgradePolicyFromJSON)(object.upgradePolicy) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes?.length) {
            obj.codeBytes = message.codeBytes.map((e) => base64FromBytes(e));
        }
        if (message.upgradePolicy !== 0) {
            obj.upgradePolicy = (0, types_1.upgradePolicyToJSON)(message.upgradePolicy);
        }
        return obj;
    },
    create(base) {
        return exports.MsgPublish.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgPublish();
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes?.map((e) => e) || [];
        message.upgradePolicy = object.upgradePolicy ?? 0;
        return message;
    },
};
function createBaseMsgPublishResponse() {
    return {};
}
exports.MsgPublishResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgPublishResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgPublishResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgPublishResponse();
        return message;
    },
};
function createBaseMsgExecute() {
    return { sender: "", moduleAddress: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.MsgExecute = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.moduleAddress !== "") {
            writer.uint32(18).string(message.moduleAddress);
        }
        if (message.moduleName !== "") {
            writer.uint32(26).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(34).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(42).string(v);
        }
        for (const v of message.args) {
            writer.uint32(50).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgExecute();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleAddress = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            moduleAddress: isSet(object.moduleAddress) ? gt.String(object.moduleAddress) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.moduleAddress !== "") {
            obj.moduleAddress = message.moduleAddress;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgExecute.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgExecute();
        message.sender = object.sender ?? "";
        message.moduleAddress = object.moduleAddress ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgExecuteResponse() {
    return {};
}
exports.MsgExecuteResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgExecuteResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgExecuteResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgExecuteResponse();
        return message;
    },
};
function createBaseMsgExecuteJSON() {
    return { sender: "", moduleAddress: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.MsgExecuteJSON = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.moduleAddress !== "") {
            writer.uint32(18).string(message.moduleAddress);
        }
        if (message.moduleName !== "") {
            writer.uint32(26).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(34).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(42).string(v);
        }
        for (const v of message.args) {
            writer.uint32(50).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgExecuteJSON();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleAddress = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.args.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            moduleAddress: isSet(object.moduleAddress) ? gt.String(object.moduleAddress) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.moduleAddress !== "") {
            obj.moduleAddress = message.moduleAddress;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args;
        }
        return obj;
    },
    create(base) {
        return exports.MsgExecuteJSON.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgExecuteJSON();
        message.sender = object.sender ?? "";
        message.moduleAddress = object.moduleAddress ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgExecuteJSONResponse() {
    return {};
}
exports.MsgExecuteJSONResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgExecuteJSONResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgExecuteJSONResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgExecuteJSONResponse();
        return message;
    },
};
function createBaseMsgScript() {
    return { sender: "", codeBytes: new Uint8Array(0), typeArgs: [], args: [] };
}
exports.MsgScript = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.codeBytes.length !== 0) {
            writer.uint32(18).bytes(message.codeBytes);
        }
        for (const v of message.typeArgs) {
            writer.uint32(26).string(v);
        }
        for (const v of message.args) {
            writer.uint32(34).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgScript();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.codeBytes = reader.bytes();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: isSet(object.codeBytes) ? bytesFromBase64(object.codeBytes) : new Uint8Array(0),
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes.length !== 0) {
            obj.codeBytes = base64FromBytes(message.codeBytes);
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgScript.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgScript();
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes ?? new Uint8Array(0);
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgScriptResponse() {
    return {};
}
exports.MsgScriptResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgScriptResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgScriptResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgScriptResponse();
        return message;
    },
};
function createBaseMsgScriptJSON() {
    return { sender: "", codeBytes: new Uint8Array(0), typeArgs: [], args: [] };
}
exports.MsgScriptJSON = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.codeBytes.length !== 0) {
            writer.uint32(18).bytes(message.codeBytes);
        }
        for (const v of message.typeArgs) {
            writer.uint32(26).string(v);
        }
        for (const v of message.args) {
            writer.uint32(34).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgScriptJSON();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.codeBytes = reader.bytes();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.args.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: isSet(object.codeBytes) ? bytesFromBase64(object.codeBytes) : new Uint8Array(0),
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes.length !== 0) {
            obj.codeBytes = base64FromBytes(message.codeBytes);
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args;
        }
        return obj;
    },
    create(base) {
        return exports.MsgScriptJSON.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgScriptJSON();
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes ?? new Uint8Array(0);
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgScriptJSONResponse() {
    return {};
}
exports.MsgScriptJSONResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgScriptJSONResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgScriptJSONResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgScriptJSONResponse();
        return message;
    },
};
function createBaseMsgGovPublish() {
    return { authority: "", sender: "", codeBytes: [], upgradePolicy: 0 };
}
exports.MsgGovPublish = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.sender !== "") {
            writer.uint32(18).string(message.sender);
        }
        for (const v of message.codeBytes) {
            writer.uint32(26).bytes(v);
        }
        if (message.upgradePolicy !== 0) {
            writer.uint32(32).int32(message.upgradePolicy);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovPublish();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.codeBytes.push(reader.bytes());
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.upgradePolicy = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: gt.Array.isArray(object?.codeBytes) ? object.codeBytes.map((e) => bytesFromBase64(e)) : [],
            upgradePolicy: isSet(object.upgradePolicy) ? (0, types_1.upgradePolicyFromJSON)(object.upgradePolicy) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes?.length) {
            obj.codeBytes = message.codeBytes.map((e) => base64FromBytes(e));
        }
        if (message.upgradePolicy !== 0) {
            obj.upgradePolicy = (0, types_1.upgradePolicyToJSON)(message.upgradePolicy);
        }
        return obj;
    },
    create(base) {
        return exports.MsgGovPublish.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGovPublish();
        message.authority = object.authority ?? "";
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes?.map((e) => e) || [];
        message.upgradePolicy = object.upgradePolicy ?? 0;
        return message;
    },
};
function createBaseMsgGovPublishResponse() {
    return {};
}
exports.MsgGovPublishResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovPublishResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgGovPublishResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgGovPublishResponse();
        return message;
    },
};
function createBaseMsgGovExecute() {
    return { authority: "", sender: "", moduleAddress: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.MsgGovExecute = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.sender !== "") {
            writer.uint32(18).string(message.sender);
        }
        if (message.moduleAddress !== "") {
            writer.uint32(26).string(message.moduleAddress);
        }
        if (message.moduleName !== "") {
            writer.uint32(34).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(42).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(50).string(v);
        }
        for (const v of message.args) {
            writer.uint32(58).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovExecute();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.moduleAddress = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 7: {
                    if (tag !== 58) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            moduleAddress: isSet(object.moduleAddress) ? gt.String(object.moduleAddress) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.moduleAddress !== "") {
            obj.moduleAddress = message.moduleAddress;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgGovExecute.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGovExecute();
        message.authority = object.authority ?? "";
        message.sender = object.sender ?? "";
        message.moduleAddress = object.moduleAddress ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgGovExecuteResponse() {
    return {};
}
exports.MsgGovExecuteResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovExecuteResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgGovExecuteResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgGovExecuteResponse();
        return message;
    },
};
function createBaseMsgGovExecuteJSON() {
    return { authority: "", sender: "", moduleAddress: "", moduleName: "", functionName: "", typeArgs: [], args: [] };
}
exports.MsgGovExecuteJSON = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.sender !== "") {
            writer.uint32(18).string(message.sender);
        }
        if (message.moduleAddress !== "") {
            writer.uint32(26).string(message.moduleAddress);
        }
        if (message.moduleName !== "") {
            writer.uint32(34).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(42).string(message.functionName);
        }
        for (const v of message.typeArgs) {
            writer.uint32(50).string(v);
        }
        for (const v of message.args) {
            writer.uint32(58).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovExecuteJSON();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.moduleAddress = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 7: {
                    if (tag !== 58) {
                        break;
                    }
                    message.args.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            moduleAddress: isSet(object.moduleAddress) ? gt.String(object.moduleAddress) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.moduleAddress !== "") {
            obj.moduleAddress = message.moduleAddress;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args;
        }
        return obj;
    },
    create(base) {
        return exports.MsgGovExecuteJSON.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGovExecuteJSON();
        message.authority = object.authority ?? "";
        message.sender = object.sender ?? "";
        message.moduleAddress = object.moduleAddress ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgGovExecuteJSONResponse() {
    return {};
}
exports.MsgGovExecuteJSONResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovExecuteJSONResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgGovExecuteJSONResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgGovExecuteJSONResponse();
        return message;
    },
};
function createBaseMsgGovScript() {
    return { authority: "", sender: "", codeBytes: new Uint8Array(0), typeArgs: [], args: [] };
}
exports.MsgGovScript = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.sender !== "") {
            writer.uint32(18).string(message.sender);
        }
        if (message.codeBytes.length !== 0) {
            writer.uint32(26).bytes(message.codeBytes);
        }
        for (const v of message.typeArgs) {
            writer.uint32(34).string(v);
        }
        for (const v of message.args) {
            writer.uint32(42).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovScript();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.codeBytes = reader.bytes();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.args.push(reader.bytes());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: isSet(object.codeBytes) ? bytesFromBase64(object.codeBytes) : new Uint8Array(0),
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes.length !== 0) {
            obj.codeBytes = base64FromBytes(message.codeBytes);
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.MsgGovScript.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGovScript();
        message.authority = object.authority ?? "";
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes ?? new Uint8Array(0);
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgGovScriptResponse() {
    return {};
}
exports.MsgGovScriptResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovScriptResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgGovScriptResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgGovScriptResponse();
        return message;
    },
};
function createBaseMsgGovScriptJSON() {
    return { authority: "", sender: "", codeBytes: new Uint8Array(0), typeArgs: [], args: [] };
}
exports.MsgGovScriptJSON = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.sender !== "") {
            writer.uint32(18).string(message.sender);
        }
        if (message.codeBytes.length !== 0) {
            writer.uint32(26).bytes(message.codeBytes);
        }
        for (const v of message.typeArgs) {
            writer.uint32(34).string(v);
        }
        for (const v of message.args) {
            writer.uint32(42).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovScriptJSON();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.codeBytes = reader.bytes();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.typeArgs.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.args.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            codeBytes: isSet(object.codeBytes) ? bytesFromBase64(object.codeBytes) : new Uint8Array(0),
            typeArgs: gt.Array.isArray(object?.typeArgs) ? object.typeArgs.map((e) => gt.String(e)) : [],
            args: gt.Array.isArray(object?.args) ? object.args.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.codeBytes.length !== 0) {
            obj.codeBytes = base64FromBytes(message.codeBytes);
        }
        if (message.typeArgs?.length) {
            obj.typeArgs = message.typeArgs;
        }
        if (message.args?.length) {
            obj.args = message.args;
        }
        return obj;
    },
    create(base) {
        return exports.MsgGovScriptJSON.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGovScriptJSON();
        message.authority = object.authority ?? "";
        message.sender = object.sender ?? "";
        message.codeBytes = object.codeBytes ?? new Uint8Array(0);
        message.typeArgs = object.typeArgs?.map((e) => e) || [];
        message.args = object.args?.map((e) => e) || [];
        return message;
    },
};
function createBaseMsgGovScriptJSONResponse() {
    return {};
}
exports.MsgGovScriptJSONResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGovScriptJSONResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgGovScriptJSONResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgGovScriptJSONResponse();
        return message;
    },
};
function createBaseMsgWhitelistStaking() {
    return { authority: "", metadataLp: "", rewardWeight: "" };
}
exports.MsgWhitelistStaking = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.metadataLp !== "") {
            writer.uint32(18).string(message.metadataLp);
        }
        if (message.rewardWeight !== "") {
            writer.uint32(26).string(message.rewardWeight);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWhitelistStaking();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.metadataLp = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.rewardWeight = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            metadataLp: isSet(object.metadataLp) ? gt.String(object.metadataLp) : "",
            rewardWeight: isSet(object.rewardWeight) ? gt.String(object.rewardWeight) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.metadataLp !== "") {
            obj.metadataLp = message.metadataLp;
        }
        if (message.rewardWeight !== "") {
            obj.rewardWeight = message.rewardWeight;
        }
        return obj;
    },
    create(base) {
        return exports.MsgWhitelistStaking.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgWhitelistStaking();
        message.authority = object.authority ?? "";
        message.metadataLp = object.metadataLp ?? "";
        message.rewardWeight = object.rewardWeight ?? "";
        return message;
    },
};
function createBaseMsgWhitelistStakingResponse() {
    return {};
}
exports.MsgWhitelistStakingResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWhitelistStakingResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgWhitelistStakingResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgWhitelistStakingResponse();
        return message;
    },
};
function createBaseMsgWhitelistGasPrice() {
    return { authority: "", metadataQuote: "", metadataLp: "" };
}
exports.MsgWhitelistGasPrice = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.metadataQuote !== "") {
            writer.uint32(18).string(message.metadataQuote);
        }
        if (message.metadataLp !== "") {
            writer.uint32(26).string(message.metadataLp);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWhitelistGasPrice();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.metadataQuote = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.metadataLp = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            metadataQuote: isSet(object.metadataQuote) ? gt.String(object.metadataQuote) : "",
            metadataLp: isSet(object.metadataLp) ? gt.String(object.metadataLp) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.metadataQuote !== "") {
            obj.metadataQuote = message.metadataQuote;
        }
        if (message.metadataLp !== "") {
            obj.metadataLp = message.metadataLp;
        }
        return obj;
    },
    create(base) {
        return exports.MsgWhitelistGasPrice.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgWhitelistGasPrice();
        message.authority = object.authority ?? "";
        message.metadataQuote = object.metadataQuote ?? "";
        message.metadataLp = object.metadataLp ?? "";
        return message;
    },
};
function createBaseMsgWhitelistGasPriceResponse() {
    return {};
}
exports.MsgWhitelistGasPriceResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWhitelistGasPriceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgWhitelistGasPriceResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgWhitelistGasPriceResponse();
        return message;
    },
};
function createBaseMsgDelistStaking() {
    return { authority: "", metadataLp: "" };
}
exports.MsgDelistStaking = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.metadataLp !== "") {
            writer.uint32(18).string(message.metadataLp);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDelistStaking();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.metadataLp = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            metadataLp: isSet(object.metadataLp) ? gt.String(object.metadataLp) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.metadataLp !== "") {
            obj.metadataLp = message.metadataLp;
        }
        return obj;
    },
    create(base) {
        return exports.MsgDelistStaking.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgDelistStaking();
        message.authority = object.authority ?? "";
        message.metadataLp = object.metadataLp ?? "";
        return message;
    },
};
function createBaseMsgDelistStakingResponse() {
    return {};
}
exports.MsgDelistStakingResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDelistStakingResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgDelistStakingResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgDelistStakingResponse();
        return message;
    },
};
function createBaseMsgDelistGasPrice() {
    return { authority: "", metadataQuote: "", metadataLp: "" };
}
exports.MsgDelistGasPrice = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.metadataQuote !== "") {
            writer.uint32(18).string(message.metadataQuote);
        }
        if (message.metadataLp !== "") {
            writer.uint32(26).string(message.metadataLp);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDelistGasPrice();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.metadataQuote = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.metadataLp = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            metadataQuote: isSet(object.metadataQuote) ? gt.String(object.metadataQuote) : "",
            metadataLp: isSet(object.metadataLp) ? gt.String(object.metadataLp) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.metadataQuote !== "") {
            obj.metadataQuote = message.metadataQuote;
        }
        if (message.metadataLp !== "") {
            obj.metadataLp = message.metadataLp;
        }
        return obj;
    },
    create(base) {
        return exports.MsgDelistGasPrice.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgDelistGasPrice();
        message.authority = object.authority ?? "";
        message.metadataQuote = object.metadataQuote ?? "";
        message.metadataLp = object.metadataLp ?? "";
        return message;
    },
};
function createBaseMsgDelistGasPriceResponse() {
    return {};
}
exports.MsgDelistGasPriceResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDelistGasPriceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgDelistGasPriceResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgDelistGasPriceResponse();
        return message;
    },
};
function createBaseMsgUpdateParams() {
    return { authority: "", params: undefined };
}
exports.MsgUpdateParams = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            types_1.Params.encode(message.params, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.params = types_1.Params.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority) ? gt.String(object.authority) : "",
            params: isSet(object.params) ? types_1.Params.fromJSON(object.params) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.params !== undefined) {
            obj.params = types_1.Params.toJSON(message.params);
        }
        return obj;
    },
    create(base) {
        return exports.MsgUpdateParams.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateParams();
        message.authority = object.authority ?? "";
        message.params = (object.params !== undefined && object.params !== null)
            ? types_1.Params.fromPartial(object.params)
            : undefined;
        return message;
    },
};
function createBaseMsgUpdateParamsResponse() {
    return {};
}
exports.MsgUpdateParamsResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.MsgUpdateParamsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateParamsResponse();
        return message;
    },
};
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.Publish = this.Publish.bind(this);
        this.Execute = this.Execute.bind(this);
        this.ExecuteJSON = this.ExecuteJSON.bind(this);
        this.Script = this.Script.bind(this);
        this.ScriptJSON = this.ScriptJSON.bind(this);
        this.GovPublish = this.GovPublish.bind(this);
        this.GovExecute = this.GovExecute.bind(this);
        this.GovExecuteJSON = this.GovExecuteJSON.bind(this);
        this.GovScript = this.GovScript.bind(this);
        this.GovScriptJSON = this.GovScriptJSON.bind(this);
        this.WhitelistStaking = this.WhitelistStaking.bind(this);
        this.WhitelistGasPrice = this.WhitelistGasPrice.bind(this);
        this.DelistStaking = this.DelistStaking.bind(this);
        this.DelistGasPrice = this.DelistGasPrice.bind(this);
        this.UpdateParams = this.UpdateParams.bind(this);
    }
    Publish(request, metadata) {
        return this.rpc.unary(exports.MsgPublishDesc, exports.MsgPublish.fromPartial(request), metadata);
    }
    Execute(request, metadata) {
        return this.rpc.unary(exports.MsgExecuteDesc, exports.MsgExecute.fromPartial(request), metadata);
    }
    ExecuteJSON(request, metadata) {
        return this.rpc.unary(exports.MsgExecuteJSONDesc, exports.MsgExecuteJSON.fromPartial(request), metadata);
    }
    Script(request, metadata) {
        return this.rpc.unary(exports.MsgScriptDesc, exports.MsgScript.fromPartial(request), metadata);
    }
    ScriptJSON(request, metadata) {
        return this.rpc.unary(exports.MsgScriptJSONDesc, exports.MsgScriptJSON.fromPartial(request), metadata);
    }
    GovPublish(request, metadata) {
        return this.rpc.unary(exports.MsgGovPublishDesc, exports.MsgGovPublish.fromPartial(request), metadata);
    }
    GovExecute(request, metadata) {
        return this.rpc.unary(exports.MsgGovExecuteDesc, exports.MsgGovExecute.fromPartial(request), metadata);
    }
    GovExecuteJSON(request, metadata) {
        return this.rpc.unary(exports.MsgGovExecuteJSONDesc, exports.MsgGovExecuteJSON.fromPartial(request), metadata);
    }
    GovScript(request, metadata) {
        return this.rpc.unary(exports.MsgGovScriptDesc, exports.MsgGovScript.fromPartial(request), metadata);
    }
    GovScriptJSON(request, metadata) {
        return this.rpc.unary(exports.MsgGovScriptJSONDesc, exports.MsgGovScriptJSON.fromPartial(request), metadata);
    }
    WhitelistStaking(request, metadata) {
        return this.rpc.unary(exports.MsgWhitelistStakingDesc, exports.MsgWhitelistStaking.fromPartial(request), metadata);
    }
    WhitelistGasPrice(request, metadata) {
        return this.rpc.unary(exports.MsgWhitelistGasPriceDesc, exports.MsgWhitelistGasPrice.fromPartial(request), metadata);
    }
    DelistStaking(request, metadata) {
        return this.rpc.unary(exports.MsgDelistStakingDesc, exports.MsgDelistStaking.fromPartial(request), metadata);
    }
    DelistGasPrice(request, metadata) {
        return this.rpc.unary(exports.MsgDelistGasPriceDesc, exports.MsgDelistGasPrice.fromPartial(request), metadata);
    }
    UpdateParams(request, metadata) {
        return this.rpc.unary(exports.MsgUpdateParamsDesc, exports.MsgUpdateParams.fromPartial(request), metadata);
    }
}
exports.MsgClientImpl = MsgClientImpl;
exports.MsgDesc = { serviceName: "initia.move.v1.Msg" };
exports.MsgPublishDesc = {
    methodName: "Publish",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgPublish.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgPublishResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgExecuteDesc = {
    methodName: "Execute",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgExecute.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgExecuteResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgExecuteJSONDesc = {
    methodName: "ExecuteJSON",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgExecuteJSON.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgExecuteJSONResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgScriptDesc = {
    methodName: "Script",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgScript.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgScriptResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgScriptJSONDesc = {
    methodName: "ScriptJSON",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgScriptJSON.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgScriptJSONResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgGovPublishDesc = {
    methodName: "GovPublish",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgGovPublish.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgGovPublishResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgGovExecuteDesc = {
    methodName: "GovExecute",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgGovExecute.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgGovExecuteResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgGovExecuteJSONDesc = {
    methodName: "GovExecuteJSON",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgGovExecuteJSON.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgGovExecuteJSONResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgGovScriptDesc = {
    methodName: "GovScript",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgGovScript.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgGovScriptResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgGovScriptJSONDesc = {
    methodName: "GovScriptJSON",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgGovScriptJSON.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgGovScriptJSONResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgWhitelistStakingDesc = {
    methodName: "WhitelistStaking",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgWhitelistStaking.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgWhitelistStakingResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgWhitelistGasPriceDesc = {
    methodName: "WhitelistGasPrice",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgWhitelistGasPrice.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgWhitelistGasPriceResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgDelistStakingDesc = {
    methodName: "DelistStaking",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgDelistStaking.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgDelistStakingResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgDelistGasPriceDesc = {
    methodName: "DelistGasPrice",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgDelistGasPrice.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgDelistGasPriceResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.MsgUpdateParamsDesc = {
    methodName: "UpdateParams",
    service: exports.MsgDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.MsgUpdateParams.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.MsgUpdateParamsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
