// @initia/initia.proto/initia/abcipp/mempool/v1/query.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/abcipp/mempool/v1/query.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.QueryQueryTxHashDesc = exports.QueryQueryTxDistributionDesc = exports.QueryDesc = exports.QueryClientImpl = exports.QueryTxHashResponse = exports.QueryTxHashRequest = exports.QueryTxDistributionResponse_DistributionEntry = exports.QueryTxDistributionResponse = exports.QueryTxDistributionRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
exports.protobufPackage = "initia.abcipp.mempool.v1";
function createBaseQueryTxDistributionRequest() {
    return {};
}
exports.QueryTxDistributionRequest = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTxDistributionRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QueryTxDistributionRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseQueryTxDistributionRequest();
        return message;
    },
};
function createBaseQueryTxDistributionResponse() {
    return { distribution: new Map() };
}
exports.QueryTxDistributionResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        message.distribution.forEach((value, key) => {
            exports.QueryTxDistributionResponse_DistributionEntry.encode({ key: key, value }, writer.uint32(10).fork()).join();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTxDistributionResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = exports.QueryTxDistributionResponse_DistributionEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.distribution.set(entry1.key, entry1.value);
                    }
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            distribution: isObject(object.distribution)
                ? Object.entries(object.distribution).reduce((acc, [key, value]) => {
                    acc.set(key, BigInt(value));
                    return acc;
                }, new Map())
                : new Map(),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.distribution?.size) {
            obj.distribution = {};
            message.distribution.forEach((v, k) => {
                obj.distribution[k] = v.toString();
            });
        }
        return obj;
    },
    create(base) {
        return exports.QueryTxDistributionResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTxDistributionResponse();
        message.distribution = (() => {
            const m = new Map();
            (object.distribution ?? new Map()).forEach((value, key) => {
                if (value !== undefined) {
                    m.set(key, BigInt(value));
                }
            });
            return m;
        })();
        return message;
    },
};
function createBaseQueryTxDistributionResponse_DistributionEntry() {
    return { key: "", value: 0n };
}
exports.QueryTxDistributionResponse_DistributionEntry = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== 0n) {
            if (BigInt.asUintN(64, message.value) !== message.value) {
                throw new gt.Error("value provided for field message.value of type uint64 too large");
            }
            writer.uint32(16).uint64(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTxDistributionResponse_DistributionEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.value = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? gt.String(object.key) : "",
            value: isSet(object.value) ? BigInt(object.value) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== 0n) {
            obj.value = message.value.toString();
        }
        return obj;
    },
    create(base) {
        return exports.QueryTxDistributionResponse_DistributionEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTxDistributionResponse_DistributionEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? 0n;
        return message;
    },
};
function createBaseQueryTxHashRequest() {
    return { sender: "", sequence: "" };
}
exports.QueryTxHashRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.sequence !== "") {
            writer.uint32(18).string(message.sequence);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTxHashRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.sequence = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? gt.String(object.sender) : "",
            sequence: isSet(object.sequence) ? gt.String(object.sequence) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.sequence !== "") {
            obj.sequence = message.sequence;
        }
        return obj;
    },
    create(base) {
        return exports.QueryTxHashRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTxHashRequest();
        message.sender = object.sender ?? "";
        message.sequence = object.sequence ?? "";
        return message;
    },
};
function createBaseQueryTxHashResponse() {
    return { txHash: "" };
}
exports.QueryTxHashResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.txHash !== "") {
            writer.uint32(10).string(message.txHash);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryTxHashResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.txHash = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { txHash: isSet(object.txHash) ? gt.String(object.txHash) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.txHash !== "") {
            obj.txHash = message.txHash;
        }
        return obj;
    },
    create(base) {
        return exports.QueryTxHashResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryTxHashResponse();
        message.txHash = object.txHash ?? "";
        return message;
    },
};
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.QueryTxDistribution = this.QueryTxDistribution.bind(this);
        this.QueryTxHash = this.QueryTxHash.bind(this);
    }
    QueryTxDistribution(request, metadata) {
        return this.rpc.unary(exports.QueryQueryTxDistributionDesc, exports.QueryTxDistributionRequest.fromPartial(request), metadata);
    }
    QueryTxHash(request, metadata) {
        return this.rpc.unary(exports.QueryQueryTxHashDesc, exports.QueryTxHashRequest.fromPartial(request), metadata);
    }
}
exports.QueryClientImpl = QueryClientImpl;
exports.QueryDesc = { serviceName: "initia.abcipp.mempool.v1.Query" };
exports.QueryQueryTxDistributionDesc = {
    methodName: "QueryTxDistribution",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryTxDistributionRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryTxDistributionResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryQueryTxHashDesc = {
    methodName: "QueryTxHash",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryTxHashRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryTxHashResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
