// @initia/initia.proto/initia/gov/v1/genesis.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/gov/v1/genesis.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const gov_1 = require("../../../cosmos/gov/v1/gov");
const gov_2 = require("./gov");
exports.protobufPackage = "initia.gov.v1";
function createBaseGenesisState() {
    return { startingProposalId: 0n, deposits: [], votes: [], proposals: [], params: undefined, constitution: "" };
}
exports.GenesisState = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.startingProposalId !== 0n) {
            if (BigInt.asUintN(64, message.startingProposalId) !== message.startingProposalId) {
                throw new gt.Error("value provided for field message.startingProposalId of type uint64 too large");
            }
            writer.uint32(8).uint64(message.startingProposalId);
        }
        for (const v of message.deposits) {
            gov_1.Deposit.encode(v, writer.uint32(18).fork()).join();
        }
        for (const v of message.votes) {
            gov_1.Vote.encode(v, writer.uint32(26).fork()).join();
        }
        for (const v of message.proposals) {
            gov_2.Proposal.encode(v, writer.uint32(34).fork()).join();
        }
        if (message.params !== undefined) {
            gov_2.Params.encode(message.params, writer.uint32(42).fork()).join();
        }
        if (message.constitution !== "") {
            writer.uint32(50).string(message.constitution);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.startingProposalId = reader.uint64();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.deposits.push(gov_1.Deposit.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.votes.push(gov_1.Vote.decode(reader, reader.uint32()));
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.proposals.push(gov_2.Proposal.decode(reader, reader.uint32()));
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.params = gov_2.Params.decode(reader, reader.uint32());
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.constitution = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            startingProposalId: isSet(object.startingProposalId) ? BigInt(object.startingProposalId) : 0n,
            deposits: gt.Array.isArray(object?.deposits) ? object.deposits.map((e) => gov_1.Deposit.fromJSON(e)) : [],
            votes: gt.Array.isArray(object?.votes) ? object.votes.map((e) => gov_1.Vote.fromJSON(e)) : [],
            proposals: gt.Array.isArray(object?.proposals) ? object.proposals.map((e) => gov_2.Proposal.fromJSON(e)) : [],
            params: isSet(object.params) ? gov_2.Params.fromJSON(object.params) : undefined,
            constitution: isSet(object.constitution) ? gt.String(object.constitution) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.startingProposalId !== 0n) {
            obj.startingProposalId = message.startingProposalId.toString();
        }
        if (message.deposits?.length) {
            obj.deposits = message.deposits.map((e) => gov_1.Deposit.toJSON(e));
        }
        if (message.votes?.length) {
            obj.votes = message.votes.map((e) => gov_1.Vote.toJSON(e));
        }
        if (message.proposals?.length) {
            obj.proposals = message.proposals.map((e) => gov_2.Proposal.toJSON(e));
        }
        if (message.params !== undefined) {
            obj.params = gov_2.Params.toJSON(message.params);
        }
        if (message.constitution !== "") {
            obj.constitution = message.constitution;
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.startingProposalId = object.startingProposalId ?? 0n;
        message.deposits = object.deposits?.map((e) => gov_1.Deposit.fromPartial(e)) || [];
        message.votes = object.votes?.map((e) => gov_1.Vote.fromPartial(e)) || [];
        message.proposals = object.proposals?.map((e) => gov_2.Proposal.fromPartial(e)) || [];
        message.params = (object.params !== undefined && object.params !== null)
            ? gov_2.Params.fromPartial(object.params)
            : undefined;
        message.constitution = object.constitution ?? "";
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
