// @initia/initia.proto/initia/ibchooks/v1/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/ibchooks/v1/types.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferFunds = exports.ACL = exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
exports.protobufPackage = "initia.ibchooks.v1";
function createBaseParams() {
    return { defaultAllowed: false };
}
exports.Params = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.defaultAllowed !== false) {
            writer.uint32(8).bool(message.defaultAllowed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.defaultAllowed = reader.bool();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { defaultAllowed: isSet(object.defaultAllowed) ? gt.Boolean(object.defaultAllowed) : false };
    },
    toJSON(message) {
        const obj = {};
        if (message.defaultAllowed !== false) {
            obj.defaultAllowed = message.defaultAllowed;
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.defaultAllowed = object.defaultAllowed ?? false;
        return message;
    },
};
function createBaseACL() {
    return { address: "", allowed: false };
}
exports.ACL = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.allowed !== false) {
            writer.uint32(16).bool(message.allowed);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseACL();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.allowed = reader.bool();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            allowed: isSet(object.allowed) ? gt.Boolean(object.allowed) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.allowed !== false) {
            obj.allowed = message.allowed;
        }
        return obj;
    },
    create(base) {
        return exports.ACL.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseACL();
        message.address = object.address ?? "";
        message.allowed = object.allowed ?? false;
        return message;
    },
};
function createBaseTransferFunds() {
    return { balanceChange: undefined, amountInPacket: undefined };
}
exports.TransferFunds = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.balanceChange !== undefined) {
            coin_1.Coin.encode(message.balanceChange, writer.uint32(10).fork()).join();
        }
        if (message.amountInPacket !== undefined) {
            coin_1.Coin.encode(message.amountInPacket, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTransferFunds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.balanceChange = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.amountInPacket = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            balanceChange: isSet(object.balanceChange) ? coin_1.Coin.fromJSON(object.balanceChange) : undefined,
            amountInPacket: isSet(object.amountInPacket) ? coin_1.Coin.fromJSON(object.amountInPacket) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.balanceChange !== undefined) {
            obj.balanceChange = coin_1.Coin.toJSON(message.balanceChange);
        }
        if (message.amountInPacket !== undefined) {
            obj.amountInPacket = coin_1.Coin.toJSON(message.amountInPacket);
        }
        return obj;
    },
    create(base) {
        return exports.TransferFunds.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTransferFunds();
        message.balanceChange = (object.balanceChange !== undefined && object.balanceChange !== null)
            ? coin_1.Coin.fromPartial(object.balanceChange)
            : undefined;
        message.amountInPacket = (object.amountInPacket !== undefined && object.amountInPacket !== null)
            ? coin_1.Coin.fromPartial(object.amountInPacket)
            : undefined;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
