// @initia/initia.proto/cosmos/group/module/v1/module.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: cosmos/group/module/v1/module.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Module = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const duration_1 = require("../../../../google/protobuf/duration");
exports.protobufPackage = "cosmos.group.module.v1";
function createBaseModule() {
    return { maxExecutionPeriod: undefined, maxMetadataLen: 0n };
}
exports.Module = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.maxExecutionPeriod !== undefined) {
            duration_1.Duration.encode(message.maxExecutionPeriod, writer.uint32(10).fork()).join();
        }
        if (message.maxMetadataLen !== 0n) {
            if (BigInt.asUintN(64, message.maxMetadataLen) !== message.maxMetadataLen) {
                throw new gt.Error("value provided for field message.maxMetadataLen of type uint64 too large");
            }
            writer.uint32(16).uint64(message.maxMetadataLen);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModule();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.maxExecutionPeriod = duration_1.Duration.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.maxMetadataLen = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            maxExecutionPeriod: isSet(object.maxExecutionPeriod) ? duration_1.Duration.fromJSON(object.maxExecutionPeriod) : undefined,
            maxMetadataLen: isSet(object.maxMetadataLen) ? BigInt(object.maxMetadataLen) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.maxExecutionPeriod !== undefined) {
            obj.maxExecutionPeriod = duration_1.Duration.toJSON(message.maxExecutionPeriod);
        }
        if (message.maxMetadataLen !== 0n) {
            obj.maxMetadataLen = message.maxMetadataLen.toString();
        }
        return obj;
    },
    create(base) {
        return exports.Module.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseModule();
        message.maxExecutionPeriod = (object.maxExecutionPeriod !== undefined && object.maxExecutionPeriod !== null)
            ? duration_1.Duration.fromPartial(object.maxExecutionPeriod)
            : undefined;
        message.maxMetadataLen = object.maxMetadataLen ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
