// @initia/initia.proto/ibc/applications/nft_transfer/v1/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: ibc/applications/nft_transfer/v1/types.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.ClassTrace = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "ibc.applications.nft_transfer.v1";
function createBaseClassTrace() {
    return { path: "", baseClassId: "" };
}
exports.ClassTrace = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.path !== "") {
            writer.uint32(10).string(message.path);
        }
        if (message.baseClassId !== "") {
            writer.uint32(18).string(message.baseClassId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseClassTrace();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.path = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.baseClassId = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            path: isSet(object.path) ? gt.String(object.path) : "",
            baseClassId: isSet(object.baseClassId) ? gt.String(object.baseClassId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.path !== "") {
            obj.path = message.path;
        }
        if (message.baseClassId !== "") {
            obj.baseClassId = message.baseClassId;
        }
        return obj;
    },
    create(base) {
        return exports.ClassTrace.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseClassTrace();
        message.path = object.path ?? "";
        message.baseClassId = object.baseClassId ?? "";
        return message;
    },
};
function createBaseParams() {
    return { sendEnabled: false, receiveEnabled: false };
}
exports.Params = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.sendEnabled !== false) {
            writer.uint32(8).bool(message.sendEnabled);
        }
        if (message.receiveEnabled !== false) {
            writer.uint32(16).bool(message.receiveEnabled);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.sendEnabled = reader.bool();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.receiveEnabled = reader.bool();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sendEnabled: isSet(object.sendEnabled) ? gt.Boolean(object.sendEnabled) : false,
            receiveEnabled: isSet(object.receiveEnabled) ? gt.Boolean(object.receiveEnabled) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sendEnabled !== false) {
            obj.sendEnabled = message.sendEnabled;
        }
        if (message.receiveEnabled !== false) {
            obj.receiveEnabled = message.receiveEnabled;
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.sendEnabled = object.sendEnabled ?? false;
        message.receiveEnabled = object.receiveEnabled ?? false;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
