// @initia/initia.proto/initia/tx/v1/query.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/tx/v1/query.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebError = exports.GrpcWebImpl = exports.QueryTxsByEventsDesc = exports.QueryGasPriceDesc = exports.QueryGasPricesDesc = exports.QueryDesc = exports.QueryClientImpl = exports.TxsByEventsResponse = exports.TxsByEventsRequest = exports.QueryGasPriceResponse = exports.QueryGasPriceRequest = exports.QueryGasPricesResponse = exports.QueryGasPricesRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const grpc_web_1 = require("@improbable-eng/grpc-web");
const browser_headers_1 = require("browser-headers");
const abci_1 = require("../../../cosmos/base/abci/v1beta1/abci");
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
const tx_1 = require("../../../cosmos/tx/v1beta1/tx");
exports.protobufPackage = "initia.tx.v1";
function createBaseQueryGasPricesRequest() {
    return {};
}
exports.QueryGasPricesRequest = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGasPricesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QueryGasPricesRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseQueryGasPricesRequest();
        return message;
    },
};
function createBaseQueryGasPricesResponse() {
    return { gasPrices: [] };
}
exports.QueryGasPricesResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.gasPrices) {
            coin_1.DecCoin.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGasPricesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.gasPrices.push(coin_1.DecCoin.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            gasPrices: gt.Array.isArray(object?.gasPrices) ? object.gasPrices.map((e) => coin_1.DecCoin.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.gasPrices?.length) {
            obj.gasPrices = message.gasPrices.map((e) => coin_1.DecCoin.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.QueryGasPricesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGasPricesResponse();
        message.gasPrices = object.gasPrices?.map((e) => coin_1.DecCoin.fromPartial(e)) || [];
        return message;
    },
};
function createBaseQueryGasPriceRequest() {
    return { denom: "" };
}
exports.QueryGasPriceRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.denom !== "") {
            writer.uint32(10).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGasPriceRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { denom: isSet(object.denom) ? gt.String(object.denom) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.QueryGasPriceRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGasPriceRequest();
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseQueryGasPriceResponse() {
    return { gasPrice: undefined };
}
exports.QueryGasPriceResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.gasPrice !== undefined) {
            coin_1.DecCoin.encode(message.gasPrice, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGasPriceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.gasPrice = coin_1.DecCoin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { gasPrice: isSet(object.gasPrice) ? coin_1.DecCoin.fromJSON(object.gasPrice) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.gasPrice !== undefined) {
            obj.gasPrice = coin_1.DecCoin.toJSON(message.gasPrice);
        }
        return obj;
    },
    create(base) {
        return exports.QueryGasPriceResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGasPriceResponse();
        message.gasPrice = (object.gasPrice !== undefined && object.gasPrice !== null)
            ? coin_1.DecCoin.fromPartial(object.gasPrice)
            : undefined;
        return message;
    },
};
function createBaseTxsByEventsRequest() {
    return { page: 0n, limit: 0n, query: "" };
}
exports.TxsByEventsRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.page !== 0n) {
            if (BigInt.asUintN(64, message.page) !== message.page) {
                throw new gt.Error("value provided for field message.page of type uint64 too large");
            }
            writer.uint32(8).uint64(message.page);
        }
        if (message.limit !== 0n) {
            if (BigInt.asUintN(64, message.limit) !== message.limit) {
                throw new gt.Error("value provided for field message.limit of type uint64 too large");
            }
            writer.uint32(16).uint64(message.limit);
        }
        if (message.query !== "") {
            writer.uint32(26).string(message.query);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTxsByEventsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.page = reader.uint64();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.limit = reader.uint64();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.query = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            page: isSet(object.page) ? BigInt(object.page) : 0n,
            limit: isSet(object.limit) ? BigInt(object.limit) : 0n,
            query: isSet(object.query) ? gt.String(object.query) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.page !== 0n) {
            obj.page = message.page.toString();
        }
        if (message.limit !== 0n) {
            obj.limit = message.limit.toString();
        }
        if (message.query !== "") {
            obj.query = message.query;
        }
        return obj;
    },
    create(base) {
        return exports.TxsByEventsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTxsByEventsRequest();
        message.page = object.page ?? 0n;
        message.limit = object.limit ?? 0n;
        message.query = object.query ?? "";
        return message;
    },
};
function createBaseTxsByEventsResponse() {
    return { txs: [], txResponses: [], total: 0n };
}
exports.TxsByEventsResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.txs) {
            tx_1.Tx.encode(v, writer.uint32(10).fork()).join();
        }
        for (const v of message.txResponses) {
            abci_1.TxResponse.encode(v, writer.uint32(18).fork()).join();
        }
        if (message.total !== 0n) {
            if (BigInt.asUintN(64, message.total) !== message.total) {
                throw new gt.Error("value provided for field message.total of type uint64 too large");
            }
            writer.uint32(24).uint64(message.total);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTxsByEventsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.txs.push(tx_1.Tx.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.txResponses.push(abci_1.TxResponse.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.total = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            txs: gt.Array.isArray(object?.txs) ? object.txs.map((e) => tx_1.Tx.fromJSON(e)) : [],
            txResponses: gt.Array.isArray(object?.txResponses)
                ? object.txResponses.map((e) => abci_1.TxResponse.fromJSON(e))
                : [],
            total: isSet(object.total) ? BigInt(object.total) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.txs?.length) {
            obj.txs = message.txs.map((e) => tx_1.Tx.toJSON(e));
        }
        if (message.txResponses?.length) {
            obj.txResponses = message.txResponses.map((e) => abci_1.TxResponse.toJSON(e));
        }
        if (message.total !== 0n) {
            obj.total = message.total.toString();
        }
        return obj;
    },
    create(base) {
        return exports.TxsByEventsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTxsByEventsResponse();
        message.txs = object.txs?.map((e) => tx_1.Tx.fromPartial(e)) || [];
        message.txResponses = object.txResponses?.map((e) => abci_1.TxResponse.fromPartial(e)) || [];
        message.total = object.total ?? 0n;
        return message;
    },
};
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.GasPrices = this.GasPrices.bind(this);
        this.GasPrice = this.GasPrice.bind(this);
        this.TxsByEvents = this.TxsByEvents.bind(this);
    }
    GasPrices(request, metadata) {
        return this.rpc.unary(exports.QueryGasPricesDesc, exports.QueryGasPricesRequest.fromPartial(request), metadata);
    }
    GasPrice(request, metadata) {
        return this.rpc.unary(exports.QueryGasPriceDesc, exports.QueryGasPriceRequest.fromPartial(request), metadata);
    }
    TxsByEvents(request, metadata) {
        return this.rpc.unary(exports.QueryTxsByEventsDesc, exports.TxsByEventsRequest.fromPartial(request), metadata);
    }
}
exports.QueryClientImpl = QueryClientImpl;
exports.QueryDesc = { serviceName: "initia.tx.v1.Query" };
exports.QueryGasPricesDesc = {
    methodName: "GasPrices",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryGasPricesRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryGasPricesResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryGasPriceDesc = {
    methodName: "GasPrice",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryGasPriceRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.QueryGasPriceResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
exports.QueryTxsByEventsDesc = {
    methodName: "TxsByEvents",
    service: exports.QueryDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.TxsByEventsRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            const value = exports.TxsByEventsResponse.decode(data);
            return {
                ...value,
                toObject() {
                    return value;
                },
            };
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        const request = { ..._request, ...methodDesc.requestType };
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
            : metadata ?? this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata ?? {},
                ...(this.options.transport !== undefined ? { transport: this.options.transport } : {}),
                debug: this.options.debug ?? false,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message.toObject());
                    }
                    else {
                        const err = new GrpcWebError(response.statusMessage, response.status, response.trailers);
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
class GrpcWebError extends gt.Error {
    constructor(message, code, metadata) {
        super(message);
        this.code = code;
        this.metadata = metadata;
    }
}
exports.GrpcWebError = GrpcWebError;
