// @initia/initia.proto/initia/tx/v1/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/tx/v1/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionOptionQueuedTx = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "initia.tx.v1";
function createBaseExtensionOptionQueuedTx() {
    return {};
}
exports.ExtensionOptionQueuedTx = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseExtensionOptionQueuedTx();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ExtensionOptionQueuedTx.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseExtensionOptionQueuedTx();
        return message;
    },
};
