// @initia/initia.proto/initia/dynamicfee/v1/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/dynamicfee/v1/types.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "initia.dynamicfee.v1";
function createBaseParams() {
    return { baseGasPrice: "", minBaseGasPrice: "", maxBaseGasPrice: "", maxChangeRate: "", targetGas: 0n };
}
exports.Params = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.baseGasPrice !== "") {
            writer.uint32(10).string(message.baseGasPrice);
        }
        if (message.minBaseGasPrice !== "") {
            writer.uint32(18).string(message.minBaseGasPrice);
        }
        if (message.maxBaseGasPrice !== "") {
            writer.uint32(26).string(message.maxBaseGasPrice);
        }
        if (message.maxChangeRate !== "") {
            writer.uint32(34).string(message.maxChangeRate);
        }
        if (message.targetGas !== 0n) {
            if (BigInt.asIntN(64, message.targetGas) !== message.targetGas) {
                throw new gt.Error("value provided for field message.targetGas of type int64 too large");
            }
            writer.uint32(40).int64(message.targetGas);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.baseGasPrice = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.minBaseGasPrice = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.maxBaseGasPrice = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.maxChangeRate = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 40) {
                        break;
                    }
                    message.targetGas = reader.int64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            baseGasPrice: isSet(object.baseGasPrice) ? gt.String(object.baseGasPrice) : "",
            minBaseGasPrice: isSet(object.minBaseGasPrice) ? gt.String(object.minBaseGasPrice) : "",
            maxBaseGasPrice: isSet(object.maxBaseGasPrice) ? gt.String(object.maxBaseGasPrice) : "",
            maxChangeRate: isSet(object.maxChangeRate) ? gt.String(object.maxChangeRate) : "",
            targetGas: isSet(object.targetGas) ? BigInt(object.targetGas) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.baseGasPrice !== "") {
            obj.baseGasPrice = message.baseGasPrice;
        }
        if (message.minBaseGasPrice !== "") {
            obj.minBaseGasPrice = message.minBaseGasPrice;
        }
        if (message.maxBaseGasPrice !== "") {
            obj.maxBaseGasPrice = message.maxBaseGasPrice;
        }
        if (message.maxChangeRate !== "") {
            obj.maxChangeRate = message.maxChangeRate;
        }
        if (message.targetGas !== 0n) {
            obj.targetGas = message.targetGas.toString();
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.baseGasPrice = object.baseGasPrice ?? "";
        message.minBaseGasPrice = object.minBaseGasPrice ?? "";
        message.maxBaseGasPrice = object.maxBaseGasPrice ?? "";
        message.maxChangeRate = object.maxChangeRate ?? "";
        message.targetGas = object.targetGas ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
