// @initia/initia.proto/cosmos/store/v1beta1/listening.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: cosmos/store/v1beta1/listening.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockMetadata = exports.StoreKVPair = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const types_1 = require("../../../tendermint/abci/types");
exports.protobufPackage = "cosmos.store.v1beta1";
function createBaseStoreKVPair() {
    return { storeKey: "", delete: false, key: new Uint8Array(0), value: new Uint8Array(0) };
}
exports.StoreKVPair = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.storeKey !== "") {
            writer.uint32(10).string(message.storeKey);
        }
        if (message.delete !== false) {
            writer.uint32(16).bool(message.delete);
        }
        if (message.key.length !== 0) {
            writer.uint32(26).bytes(message.key);
        }
        if (message.value.length !== 0) {
            writer.uint32(34).bytes(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStoreKVPair();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.storeKey = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.delete = reader.bool();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.key = reader.bytes();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.value = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            storeKey: isSet(object.storeKey) ? gt.String(object.storeKey) : "",
            delete: isSet(object.delete) ? gt.Boolean(object.delete) : false,
            key: isSet(object.key) ? bytesFromBase64(object.key) : new Uint8Array(0),
            value: isSet(object.value) ? bytesFromBase64(object.value) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.storeKey !== "") {
            obj.storeKey = message.storeKey;
        }
        if (message.delete !== false) {
            obj.delete = message.delete;
        }
        if (message.key.length !== 0) {
            obj.key = base64FromBytes(message.key);
        }
        if (message.value.length !== 0) {
            obj.value = base64FromBytes(message.value);
        }
        return obj;
    },
    create(base) {
        return exports.StoreKVPair.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseStoreKVPair();
        message.storeKey = object.storeKey ?? "";
        message.delete = object.delete ?? false;
        message.key = object.key ?? new Uint8Array(0);
        message.value = object.value ?? new Uint8Array(0);
        return message;
    },
};
function createBaseBlockMetadata() {
    return { responseCommit: undefined, requestFinalizeBlock: undefined, responseFinalizeBlock: undefined };
}
exports.BlockMetadata = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.responseCommit !== undefined) {
            types_1.ResponseCommit.encode(message.responseCommit, writer.uint32(50).fork()).join();
        }
        if (message.requestFinalizeBlock !== undefined) {
            types_1.RequestFinalizeBlock.encode(message.requestFinalizeBlock, writer.uint32(58).fork()).join();
        }
        if (message.responseFinalizeBlock !== undefined) {
            types_1.ResponseFinalizeBlock.encode(message.responseFinalizeBlock, writer.uint32(66).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBlockMetadata();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.responseCommit = types_1.ResponseCommit.decode(reader, reader.uint32());
                    continue;
                }
                case 7: {
                    if (tag !== 58) {
                        break;
                    }
                    message.requestFinalizeBlock = types_1.RequestFinalizeBlock.decode(reader, reader.uint32());
                    continue;
                }
                case 8: {
                    if (tag !== 66) {
                        break;
                    }
                    message.responseFinalizeBlock = types_1.ResponseFinalizeBlock.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responseCommit: isSet(object.responseCommit) ? types_1.ResponseCommit.fromJSON(object.responseCommit) : undefined,
            requestFinalizeBlock: isSet(object.requestFinalizeBlock)
                ? types_1.RequestFinalizeBlock.fromJSON(object.requestFinalizeBlock)
                : undefined,
            responseFinalizeBlock: isSet(object.responseFinalizeBlock)
                ? types_1.ResponseFinalizeBlock.fromJSON(object.responseFinalizeBlock)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responseCommit !== undefined) {
            obj.responseCommit = types_1.ResponseCommit.toJSON(message.responseCommit);
        }
        if (message.requestFinalizeBlock !== undefined) {
            obj.requestFinalizeBlock = types_1.RequestFinalizeBlock.toJSON(message.requestFinalizeBlock);
        }
        if (message.responseFinalizeBlock !== undefined) {
            obj.responseFinalizeBlock = types_1.ResponseFinalizeBlock.toJSON(message.responseFinalizeBlock);
        }
        return obj;
    },
    create(base) {
        return exports.BlockMetadata.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBlockMetadata();
        message.responseCommit = (object.responseCommit !== undefined && object.responseCommit !== null)
            ? types_1.ResponseCommit.fromPartial(object.responseCommit)
            : undefined;
        message.requestFinalizeBlock = (object.requestFinalizeBlock !== undefined && object.requestFinalizeBlock !== null)
            ? types_1.RequestFinalizeBlock.fromPartial(object.requestFinalizeBlock)
            : undefined;
        message.responseFinalizeBlock =
            (object.responseFinalizeBlock !== undefined && object.responseFinalizeBlock !== null)
                ? types_1.ResponseFinalizeBlock.fromPartial(object.responseFinalizeBlock)
                : undefined;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
