// @initia/amino-converter/dist/index.js
'use strict';

var stargate = require('@cosmjs/stargate');
var cosmwasmStargate = require('@cosmjs/cosmwasm-stargate');
var protoSigning = require('@cosmjs/proto-signing');
var tx$3 = require('@initia/initia.proto/ibc/applications/nft_transfer/v1/tx');
var tx$4 = require('@initia/initia.proto/initia/bank/v1/tx');
var tx$5 = require('@initia/initia.proto/initia/distribution/v1/tx');
var tx$6 = require('@initia/initia.proto/initia/gov/v1/tx');
var tx$7 = require('@initia/initia.proto/initia/ibchooks/v1/tx');
var tx$8 = require('@initia/initia.proto/initia/intertx/v1/tx');
var tx$9 = require('@initia/initia.proto/initia/move/v1/tx');
var types$1 = require('@initia/initia.proto/initia/move/v1/types');
var tx$a = require('@initia/initia.proto/initia/mstaking/v1/tx');
var tx$b = require('@initia/initia.proto/initia/reward/v1/tx');
var tx$c = require('@initia/initia.proto/minievm/evm/v1/tx');
var tx$d = require('@initia/initia.proto/miniwasm/tokenfactory/v1/tx');
var tx$e = require('@initia/opinit.proto/opinit/opchild/v1/tx');
var types = require('@initia/opinit.proto/opinit/ophost/v1/types');
var tx$f = require('@initia/opinit.proto/opinit/ophost/v1/tx');
var tx = require('@initia/initia.proto/cosmos/authz/v1beta1/tx');
var authz = require('@initia/initia.proto/cosmwasm/wasm/v1/authz');
var types$2 = require('@initia/initia.proto/cosmwasm/wasm/v1/types');
var authz$4 = require('@initia/initia.proto/cosmos/authz/v1beta1/authz');
var authz$3 = require('@initia/initia.proto/cosmos/bank/v1beta1/authz');
var authz$2 = require('@initia/initia.proto/initia/mstaking/v1/authz');
var authz$1 = require('@initia/initia.proto/initia/move/v1/authz');
var tx$1 = require('@initia/initia.proto/cosmos/feegrant/v1beta1/tx');
var feegrant = require('@initia/initia.proto/cosmos/feegrant/v1beta1/feegrant');
var tx$2 = require('@initia/initia.proto/cosmos/gov/v1/tx');

// src/index.ts

// src/ibc/core/client/v1/height.ts
var Height = {
  toAmino: (height) => {
    if (height.revisionNumber === 0n && height.revisionHeight === 0n) {
      return {};
    }
    return {
      revision_number: height.revisionNumber.toString(),
      revision_height: height.revisionHeight.toString()
    };
  },
  fromAmino: (height) => {
    if (height.revision_height === void 0 && height.revision_height === void 0) {
      return {
        revisionHeight: 0n,
        revisionNumber: 0n
      };
    }
    if (height.revision_height === void 0 || height.revision_number === void 0) {
      throw new Error(
        "Both revision_height and revision_number must exist or neither must exist."
      );
    }
    return {
      revisionHeight: BigInt(height.revision_height),
      revisionNumber: BigInt(height.revision_number)
    };
  }
};

// src/ibc/applications/nft-transfer/types.ts
var Params = {
  toAmino: (params) => ({
    send_enabled: params.sendEnabled,
    receive_enabled: params.receiveEnabled
  }),
  fromAmino: (params) => ({
    sendEnabled: params.send_enabled,
    receiveEnabled: params.receive_enabled
  })
};

// src/ibc/applications/nft-transfer/tx.ts
var registry = [
  ["/ibc.applications.nft_transfer.v1.MsgTransfer", tx$3.MsgTransfer],
  ["/ibc.applications.nft_transfer.v1.Params", tx$3.MsgUpdateParams]
];
var aminoConverters = {
  "/ibc.applications.nft_transfer.v1.MsgTransfer": {
    aminoType: "nft-transfer/MsgTransfer",
    toAmino: (msg) => ({
      source_port: msg.sourcePort,
      source_channel: msg.sourceChannel,
      class_id: msg.classId,
      token_ids: msg.tokenIds.length === 0 ? void 0 : msg.tokenIds,
      sender: msg.sender,
      receiver: msg.receiver,
      timeout_height: msg.timeoutHeight ? Height.toAmino(msg.timeoutHeight) : void 0,
      timeout_timestamp: msg.timeoutTimestamp === BigInt(0) ? void 0 : msg.timeoutTimestamp.toString(),
      memo: msg.memo === "" ? void 0 : msg.memo
    }),
    fromAmino: (msg) => ({
      sourcePort: msg.source_port,
      sourceChannel: msg.source_channel,
      classId: msg.class_id,
      tokenIds: msg.token_ids ?? [],
      sender: msg.sender,
      receiver: msg.receiver,
      timeoutHeight: msg.timeout_height ? Height.fromAmino(msg.timeout_height) : void 0,
      timeoutTimestamp: msg.timeout_timestamp ? BigInt(msg.timeout_timestamp) : BigInt(0),
      memo: msg.memo ?? ""
    })
  },
  "/ibc.applications.nft_transfer.v1.Params": {
    aminoType: "nft-transfer/Params",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params.fromAmino(msg.params)
    })
  }
};

// src/ibc/applications/index.ts
var ibcApplicationsRegistry = [
  ...registry,
  ...registry
];
var ibcApplicationsAminoConverters = {
  ...aminoConverters,
  ...aminoConverters
};

// src/cosmos/bank/v1beta1/denomUnit.ts
var DenomUnit = {
  toAmino: (denomUnit) => ({
    denom: denomUnit.denom,
    exponent: denomUnit.exponent.toString(),
    aliases: denomUnit.aliases.length === 0 ? void 0 : denomUnit.aliases
  }),
  fromAmino: (denomUnit) => ({
    denom: denomUnit.denom,
    exponent: Number(denomUnit.exponent),
    aliases: denomUnit.aliases ?? []
  })
};

// src/cosmos/bank/v1beta1/metadata.ts
var Metadata = {
  toAmino: (metadata) => ({
    description: metadata.description,
    denom_units: metadata.denomUnits.length === 0 ? void 0 : metadata.denomUnits.map((denomUint) => DenomUnit.toAmino(denomUint)),
    base: metadata.base,
    display: metadata.display,
    name: metadata.name,
    symbol: metadata.symbol,
    uri: metadata.uri,
    uri_hash: metadata.uriHash
  }),
  fromAmino: (metadata) => ({
    description: metadata.description,
    denomUnits: metadata.denom_units ? metadata.denom_units.map((denomUnit) => DenomUnit.fromAmino(denomUnit)) : [],
    base: metadata.base,
    display: metadata.display,
    name: metadata.name,
    symbol: metadata.symbol,
    uri: metadata.uri,
    uriHash: metadata.uri_hash
  })
};

// src/initia/bank/v1/tx.ts
var registry2 = [
  ["/initia.bank.v1.MsgSetDenomMetadata", tx$4.MsgSetDenomMetadata]
];
var aminoConverters2 = {
  "/initia.bank.v1.MsgSetDenomMetadata": {
    aminoType: "bank/MsgSetDenomMetadata",
    toAmino: (msg) => ({
      authority: msg.authority,
      metadata: Metadata.toAmino(msg.metadata)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      metadata: Metadata.fromAmino(msg.metadata)
    })
  }
};

// src/initia/distribution/v1/types.ts
var Params2 = {
  toAmino: (params) => ({
    community_tax: params.communityTax,
    withdraw_addr_enabled: params.withdrawAddrEnabled,
    reward_weights: params.rewardWeights.length === 0 ? void 0 : params.rewardWeights
  }),
  fromAmino: (params) => ({
    communityTax: params.community_tax,
    withdrawAddrEnabled: params.withdraw_addr_enabled,
    rewardWeights: params.reward_weights ?? []
  })
};

// src/cosmos/base/v1beta1/coin.ts
var Coin = {
  toAmino: (coin) => ({
    denom: coin.denom,
    amount: coin.amount
  }),
  fromAmino: (coin) => ({
    denom: coin.denom,
    amount: coin.amount
  })
};
var DecCoin = {
  toAmino: (coin) => {
    const { denom, amount } = coin;
    const integerPart = amount.slice(0, -18);
    const decimalPart = amount.slice(-18).padStart(18, "0");
    return {
      denom,
      amount: `${integerPart === "" ? "0" : integerPart}.${decimalPart}`
    };
  },
  fromAmino: (coin) => {
    const { denom, amount } = coin;
    const [integerPart, decimalPart] = amount.split(".");
    const newAmount = BigInt(integerPart) * BigInt(10n ** 18n) + BigInt(decimalPart);
    return {
      denom,
      amount: newAmount.toString()
    };
  }
};

// src/initia/distribution/v1/tx.ts
var registry3 = [
  ["/initia.distribution.v1.MsgUpdateParams", tx$5.MsgUpdateParams],
  [
    "/initia.distribution.v1.MsgDepositValidatorRewardsPool",
    tx$5.MsgDepositValidatorRewardsPool
  ]
];
var aminoConverters3 = {
  "/initia.distribution.v1.MsgUpdateParams": {
    aminoType: "distribution/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params2.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params2.fromAmino(msg.params)
    })
  },
  "/initia.distribution.v1.MsgDepositValidatorRewardsPool": {
    aminoType: "move/MsgDepositValidatorRewardsPool",
    toAmino: (msg) => ({
      depositor: msg.depositor,
      validator_address: msg.validatorAddress,
      denom: msg.denom,
      amount: msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      depositor: msg.depositor,
      validatorAddress: msg.validator_address,
      denom: msg.denom,
      amount: msg.amount.map((coin) => Coin.fromAmino(coin))
    })
  }
};

// src/google/protobuf/duration.ts
var Duration = {
  toAmino: (duration) => {
    const { seconds, nanos } = duration;
    let str = seconds * BigInt(1e9) + BigInt(nanos) < 0 ? "-" : "";
    str += seconds < BigInt(0) ? (-seconds).toString() : seconds.toString();
    if (nanos !== 0) {
      const nanosStr = nanos.toString().replace(/0+$/, "");
      str += "." + nanosStr;
    }
    str += "s";
    return str;
  },
  fromAmino: (duration) => {
    const negative = duration.startsWith("-");
    const [sec, nano = "0"] = duration.slice(negative ? 1 : 0).split(".");
    const paddedNano = nano.padEnd(9, "0");
    return {
      seconds: (negative ? BigInt(-1) : BigInt(1)) * BigInt(sec),
      nanos: (negative ? -1 : 1) * Number(paddedNano)
    };
  }
};

// src/initia/gov/v1/types.ts
var Vesting = {
  toAmino: (vesting) => ({
    module_addr: vesting.moduleAddr,
    module_name: vesting.moduleName,
    creator_addr: vesting.creatorAddr
  }),
  fromAmino: (vesting) => ({
    moduleAddr: vesting.module_addr,
    moduleName: vesting.module_name,
    creatorAddr: vesting.creator_addr
  })
};
var Params3 = {
  toAmino: (params) => ({
    min_deposit: params.minDeposit.length === 0 ? null : params.minDeposit.map((coin) => Coin.toAmino(coin)),
    max_deposit_period: Duration.toAmino(
      params.maxDepositPeriod
    ),
    voting_period: Duration.toAmino(params.votingPeriod),
    quorum: params.quorum,
    threshold: params.threshold,
    veto_threshold: params.vetoThreshold,
    min_initial_deposit_ratio: params.minInitialDepositRatio,
    proposal_cancel_ratio: params.proposalCancelRatio,
    proposal_cancel_dest: params.proposalCancelDest,
    expedited_voting_period: Duration.toAmino(
      params.expeditedVotingPeriod
    ),
    expedited_threshold: params.expeditedThreshold,
    expedited_min_deposit: params.expeditedMinDeposit.length === 0 ? null : params.expeditedMinDeposit.map((coin) => Coin.toAmino(coin)),
    burn_vote_quorum: params.burnVoteQuorum,
    burn_proposal_deposit_prevote: params.burnProposalDepositPrevote,
    burn_vote_veto: params.burnVoteVeto,
    min_deposit_ratio: params.minDepositRatio,
    emergency_min_deposit: params.emergencyMinDeposit.length === 0 ? null : params.emergencyMinDeposit.map((coin) => Coin.toAmino(coin)),
    emergency_tally_interval: Duration.toAmino(
      params.emergencyTallyInterval
    ),
    emergency_submitters: params.emergencySubmitters.length === 0 ? void 0 : params.emergencySubmitters,
    low_threshold_functions: params.lowThresholdFunctions.length === 0 ? void 0 : params.lowThresholdFunctions,
    vesting: params.vesting ? Vesting.toAmino(params.vesting) : void 0
  }),
  fromAmino: (params) => ({
    minDeposit: params.min_deposit ? params.min_deposit.map((coin) => Coin.fromAmino(coin)) : [],
    maxDepositPeriod: Duration.fromAmino(params.max_deposit_period),
    votingPeriod: Duration.fromAmino(params.voting_period),
    quorum: params.quorum,
    threshold: params.threshold,
    vetoThreshold: params.veto_threshold,
    minInitialDepositRatio: params.min_initial_deposit_ratio,
    proposalCancelRatio: params.proposal_cancel_ratio,
    proposalCancelDest: params.proposal_cancel_dest,
    expeditedVotingPeriod: Duration.fromAmino(params.expedited_voting_period),
    expeditedThreshold: params.expedited_threshold,
    expeditedMinDeposit: params.expedited_min_deposit ? params.expedited_min_deposit.map((coin) => Coin.fromAmino(coin)) : [],
    burnVoteQuorum: params.burn_vote_quorum,
    burnProposalDepositPrevote: params.burn_proposal_deposit_prevote,
    burnVoteVeto: params.burn_vote_veto,
    minDepositRatio: params.min_deposit_ratio,
    emergencyMinDeposit: params.emergency_min_deposit ? params.emergency_min_deposit.map((coin) => Coin.fromAmino(coin)) : [],
    emergencyTallyInterval: Duration.fromAmino(params.emergency_tally_interval),
    emergencySubmitters: params.emergency_submitters ?? [],
    lowThresholdFunctions: params.low_threshold_functions ?? [],
    vesting: params.vesting ? Vesting.fromAmino(params.vesting) : void 0
  })
};

// src/initia/gov/v1/tx.ts
var registry4 = [
  ["/initia.gov.v1.MsgUpdateParams", tx$6.MsgUpdateParams]
];
var aminoConverters4 = {
  "/initia.gov.v1.MsgUpdateParams": {
    aminoType: "gov/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params3.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params3.fromAmino(msg.params)
    })
  }
};

// src/initia/ibchooks/v1/types.ts
var Params4 = {
  toAmino: (params) => ({
    default_allowed: params.defaultAllowed
  }),
  fromAmino: (params) => ({
    defaultAllowed: params.default_allowed
  })
};

// src/initia/ibchooks/v1/tx.ts
var registry5 = [
  ["/initia.ibchooks.v1.MsgUpdateACL", tx$7.MsgUpdateACL],
  ["/initia.ibchooks.v1.MsgUpdateParams", tx$7.MsgUpdateParams]
];
var aminoConverters5 = {
  "/initia.ibchooks.v1.MsgUpdateACL": {
    aminoType: "ibchooks/MsgUpdateACL",
    toAmino: (msg) => ({
      authority: msg.authority,
      address: msg.address,
      allowed: msg.allowed
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      address: msg.address,
      allowed: msg.allowed
    })
  },
  "/initia.ibchooks.v1.MsgUpdateParams": {
    aminoType: "ibchooks/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params4.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params4.fromAmino(msg.params)
    })
  }
};

// src/utils/index.ts
function bytesToBase64(uint8Array) {
  const binaryString = String.fromCharCode(...uint8Array);
  return btoa(binaryString);
}
function base64ToBytes(base64Str) {
  const binaryString = atob(base64Str);
  const uint8Array = new Uint8Array(binaryString.length);
  for (let i = 0; i < binaryString.length; i++) {
    uint8Array[i] = binaryString.charCodeAt(i);
  }
  return uint8Array;
}
function createMsgConverter(registry18, aminoConverters18) {
  const toAminoMsgs = {};
  const toProtoMsgs = {};
  for (const [typeUrl, proto] of registry18) {
    const aminoConverter = aminoConverters18[typeUrl];
    if (aminoConverter === void 0) {
      continue;
    }
    toAminoMsgs[typeUrl] = (msg) => {
      return {
        type: aminoConverter.aminoType,
        value: aminoConverter.toAmino(proto.decode(msg.value))
      };
    };
    toProtoMsgs[aminoConverter.aminoType] = (msg) => {
      return {
        typeUrl,
        value: proto.encode(aminoConverter.fromAmino(msg.value)).finish()
      };
    };
  }
  return {
    toAminoMsg: (msg) => {
      const typeUrl = msg.typeUrl;
      if (toAminoMsgs[typeUrl] === void 0) {
        throw new Error(`Unknown TypeUrl`);
      }
      return toAminoMsgs[typeUrl](msg);
    },
    toProtoMsg: (msg) => {
      const type = msg.type;
      if (toProtoMsgs[type] === void 0) {
        throw new Error(`Unknown Amino Type`);
      }
      return toProtoMsgs[type](msg);
    }
  };
}

// src/initia/intertx/v1/tx.ts
var registry6 = [
  ["/initia.intertx.v1.MsgRegisterAccount", tx$8.MsgRegisterAccount],
  ["/initia.intertx.v1.MsgSubmitTx", tx$8.MsgSubmitTx]
];
var aminoConverters6 = {
  "/initia.intertx.v1.MsgRegisterAccount": {
    aminoType: "intertx/MsgRegisterAccount",
    toAmino: (msg) => ({
      owner: msg.owner,
      connection_id: msg.connectionId,
      version: msg.version,
      ordering: msg.ordering
    }),
    fromAmino: (msg) => ({
      owner: msg.owner,
      connectionId: msg.connection_id,
      version: msg.version,
      ordering: msg.ordering
    })
  }
};
function generateMsgSubmitTxAminoConverter(registry18, aminoConverters18) {
  const { toAminoMsg, toProtoMsg } = createMsgConverter(
    registry18,
    aminoConverters18
  );
  return {
    "/initia.intertx.v1.MsgSubmitTx": {
      aminoType: "intertx/MsgSubmitTx",
      toAmino: (msg) => ({
        owner: msg.owner,
        connection_id: msg.connectionId,
        msg: toAminoMsg(msg.msg)
      }),
      fromAmino: (msg) => ({
        owner: msg.owner,
        connectionId: msg.connection_id,
        msg: toProtoMsg(msg.msg)
      })
    }
  };
}

// src/initia/move/v1/types.ts
var Params5 = {
  toAmino: (msg) => ({
    base_denom: msg.baseDenom,
    base_min_gas_price: msg.baseMinGasPrice,
    contract_shared_revenue_ratio: msg.contractSharedRevenueRatio,
    script_enabled: msg.scriptEnabled,
    allowed_publishers: msg.allowedPublishers.length === 0 ? null : msg.allowedPublishers
  }),
  fromAmino: (msg) => ({
    baseDenom: msg.base_denom,
    baseMinGasPrice: msg.base_min_gas_price,
    contractSharedRevenueRatio: msg.contract_shared_revenue_ratio,
    scriptEnabled: msg.script_enabled,
    allowedPublishers: msg.allowed_publishers ?? []
  })
};
var ExecuteAuthorizationItem = {
  toAmino: (msg) => ({
    module_address: msg.moduleAddress,
    module_name: msg.moduleName,
    function_name: msg.functionNames.length === 0 ? void 0 : msg.functionNames
  }),
  fromAmino: (msg) => ({
    moduleAddress: msg.module_address,
    moduleName: msg.module_name,
    functionNames: msg.function_name ?? []
  })
};
var registry7 = [
  ["/initia.move.v1.MsgPublish", tx$9.MsgPublish],
  ["/initia.move.v1.MsgExecute", tx$9.MsgExecute],
  ["/initia.move.v1.MsgExecuteJSON", tx$9.MsgExecuteJSON],
  ["/initia.move.v1.MsgScript", tx$9.MsgScript],
  ["/initia.move.v1.MsgScriptJSON", tx$9.MsgScriptJSON],
  ["/initia.move.v1.MsgGovPublish", tx$9.MsgGovPublish],
  ["/initia.move.v1.MsgGovExecute", tx$9.MsgGovExecute],
  ["/initia.move.v1.MsgGovExecuteJSON", tx$9.MsgGovExecuteJSON],
  ["/initia.move.v1.MsgGovScript", tx$9.MsgGovScript],
  ["/initia.move.v1.MsgGovScriptJSON", tx$9.MsgGovScriptJSON],
  ["/initia.move.v1.MsgWhitelist", tx$9.MsgWhitelist],
  ["/initia.move.v1.MsgDelist", tx$9.MsgDelist],
  ["/initia.move.v1.MsgUpdateParams", tx$9.MsgUpdateParams]
];
var aminoConverters7 = {
  "/initia.move.v1.MsgPublish": {
    aminoType: "move/MsgPublish",
    toAmino: (msg) => ({
      sender: msg.sender,
      code_bytes: msg.codeBytes.length === 0 ? void 0 : msg.codeBytes.map((bytes) => bytesToBase64(bytes)),
      upgrade_policy: msg.upgradePolicy === types$1.UpgradePolicy.UNSPECIFIED ? void 0 : msg.upgradePolicy
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      codeBytes: msg.code_bytes ? msg.code_bytes.map((str) => base64ToBytes(str)) : [],
      upgradePolicy: msg.upgrade_policy ?? types$1.UpgradePolicy.UNSPECIFIED
    })
  },
  "/initia.move.v1.MsgExecute": {
    aminoType: "move/MsgExecute",
    toAmino: (msg) => ({
      sender: msg.sender,
      module_address: msg.moduleAddress,
      module_name: msg.moduleName,
      function_name: msg.functionName,
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args.map((bytes) => bytesToBase64(bytes))
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      moduleAddress: msg.module_address,
      moduleName: msg.module_name,
      functionName: msg.function_name,
      typeArgs: msg.type_args ?? [],
      args: msg.args ? msg.args.map((arg) => base64ToBytes(arg)) : []
    })
  },
  "/initia.move.v1.MsgExecuteJSON": {
    aminoType: "move/MsgExecuteJSON",
    toAmino: (msg) => ({
      sender: msg.sender,
      module_address: msg.moduleAddress,
      module_name: msg.moduleName,
      function_name: msg.functionName,
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      moduleAddress: msg.module_address,
      moduleName: msg.module_name,
      functionName: msg.function_name,
      typeArgs: msg.type_args ?? [],
      args: msg.args ?? []
    })
  },
  "/initia.move.v1.MsgScript": {
    aminoType: "move/MsgScript",
    toAmino: (msg) => ({
      sender: msg.sender,
      code_bytes: bytesToBase64(msg.codeBytes),
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args.map((bytes) => bytesToBase64(bytes))
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      codeBytes: base64ToBytes(msg.code_bytes),
      typeArgs: msg.type_args ?? [],
      args: msg.args ? msg.args.map((arg) => base64ToBytes(arg)) : []
    })
  },
  "/initia.move.v1.MsgScriptJSON": {
    aminoType: "move/MsgScriptJSON",
    toAmino: (msg) => ({
      sender: msg.sender,
      code_bytes: bytesToBase64(msg.codeBytes),
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      codeBytes: base64ToBytes(msg.code_bytes),
      typeArgs: msg.type_args ?? [],
      args: msg.args ?? []
    })
  },
  "/initia.move.v1.MsgGovPublish": {
    aminoType: "move/MsgGovPublish",
    toAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      code_bytes: msg.codeBytes.length === 0 ? void 0 : msg.codeBytes.map((bytes) => bytesToBase64(bytes)),
      upgrade_policy: msg.upgradePolicy === types$1.UpgradePolicy.UNSPECIFIED ? void 0 : msg.upgradePolicy
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      codeBytes: msg.code_bytes ? msg.code_bytes.map((str) => base64ToBytes(str)) : [],
      upgradePolicy: msg.upgrade_policy ?? types$1.UpgradePolicy.UNSPECIFIED
    })
  },
  "/initia.move.v1.MsgGovExecute": {
    aminoType: "move/MsgGovExecute",
    toAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      module_address: msg.moduleAddress,
      module_name: msg.moduleName,
      function_name: msg.functionName,
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args.map((bytes) => bytesToBase64(bytes))
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      moduleAddress: msg.module_address,
      moduleName: msg.module_name,
      functionName: msg.function_name,
      typeArgs: msg.type_args ?? [],
      args: msg.args ? msg.args.map((arg) => base64ToBytes(arg)) : []
    })
  },
  "/initia.move.v1.MsgGovExecuteJSON": {
    aminoType: "move/MsgGovExecuteJSON",
    toAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      module_address: msg.moduleAddress,
      module_name: msg.moduleName,
      function_name: msg.functionName,
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      moduleAddress: msg.module_address,
      moduleName: msg.module_name,
      functionName: msg.function_name,
      typeArgs: msg.type_args ?? [],
      args: msg.args ?? []
    })
  },
  "/initia.move.v1.MsgGovScript": {
    aminoType: "move/MsgGovScript",
    toAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      code_bytes: bytesToBase64(msg.codeBytes),
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args.map((bytes) => bytesToBase64(bytes))
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      codeBytes: base64ToBytes(msg.code_bytes),
      typeArgs: msg.type_args ?? [],
      args: msg.args ? msg.args.map((arg) => base64ToBytes(arg)) : []
    })
  },
  "/initia.move.v1.MsgGovScriptJSON": {
    aminoType: "move/MsgGovScriptJSON",
    toAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      code_bytes: bytesToBase64(msg.codeBytes),
      type_args: msg.typeArgs.length === 0 ? void 0 : msg.typeArgs,
      args: msg.args.length === 0 ? void 0 : msg.args
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      sender: msg.sender,
      codeBytes: base64ToBytes(msg.code_bytes),
      typeArgs: msg.type_args ?? [],
      args: msg.args ?? []
    })
  },
  "/initia.move.v1.MsgWhitelist": {
    aminoType: "move/MsgWhitelist",
    toAmino: (msg) => ({
      authority: msg.authority,
      metadata_lp: msg.metadataLp,
      reward_weight: msg.rewardWeight
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      metadataLp: msg.metadata_lp,
      rewardWeight: msg.reward_weight
    })
  },
  "/initia.move.v1.MsgDelist": {
    aminoType: "move/MsgDelist",
    toAmino: (msg) => ({
      authority: msg.authority,
      metadata_lp: msg.metadataLp
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      metadataLp: msg.metadata_lp
    })
  },
  "/initia.move.v1.MsgUpdateParams": {
    aminoType: "move/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params5.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params5.fromAmino(msg.params)
    })
  }
};

// src/initia/mstaking/v1/types.ts
var Description = {
  toAmino: (description) => ({
    moniker: description.moniker,
    identity: description.identity,
    website: description.website,
    details: description.details,
    security_contact: description.securityContact
  }),
  fromAmino: (description) => ({
    moniker: description.moniker,
    identity: description.identity,
    website: description.website,
    details: description.details,
    securityContact: description.security_contact
  })
};
var CommissionRates = {
  toAmino: (commissionRates) => ({
    rate: commissionRates.rate,
    max_rate: commissionRates.maxRate,
    max_change_rate: commissionRates.maxChangeRate
  }),
  fromAmino: (commissionRates) => ({
    rate: commissionRates.rate,
    maxRate: commissionRates.max_rate,
    maxChangeRate: commissionRates.max_change_rate
  })
};
var Params6 = {
  toAmino: (params) => ({
    unbonding_time: Duration.toAmino(params.unbondingTime),
    max_validators: params.maxValidators,
    max_entries: params.maxEntries,
    historical_entries: params.historicalEntries,
    bond_denoms: params.bondDenoms.length === 0 ? void 0 : params.bondDenoms,
    min_voting_power: params.minVotingPower.toString(),
    min_commission_rate: params.minCommissionRate
  }),
  fromAmino: (params) => ({
    unbondingTime: Duration.fromAmino(params.unbonding_time),
    maxValidators: params.max_validators,
    maxEntries: params.max_entries,
    historicalEntries: params.historical_entries,
    bondDenoms: params.bond_denoms ?? [],
    minVotingPower: BigInt(params.min_voting_power),
    minCommissionRate: params.min_commission_rate
  })
};

// src/cosmos/crypto/ed25519/keys.ts
var PubKey = {
  toAmino: (pubKey) => ({
    key: bytesToBase64(pubKey.key)
  }),
  fromAmino: (pubKey) => ({
    key: base64ToBytes(pubKey.key)
  }),
  toProtoMsg: (pubKey) => ({
    typeUrl: "/cosmos.crypto.ed25519.PubKey",
    value: pubKey.key
  }),
  fromProtoMsg: (pubKey) => ({
    key: pubKey.value
  }),
  toAminoMsg: (pubKey) => ({
    type: "tendermint/PubKeySecp256k1",
    value: PubKey.toAmino(pubKey).key
  }),
  fromAminoMsg: (pubKey) => ({
    key: base64ToBytes(pubKey.value)
  })
};

// src/initia/mstaking/v1/tx.ts
var registry8 = [
  ["/initia.mstaking.v1.MsgCreateValidator", tx$a.MsgCreateValidator],
  ["/initia.mstaking.v1.MsgDelegate", tx$a.MsgDelegate],
  ["/initia.mstaking.v1.MsgBeginRedelegate", tx$a.MsgBeginRedelegate],
  ["/initia.mstaking.v1.MsgUndelegate", tx$a.MsgUndelegate],
  [
    "/initia.mstaking.v1.MsgCancelUnbondingDelegation",
    tx$a.MsgCancelUnbondingDelegation
  ],
  ["/initia.mstaking.v1.MsgUpdateParams", tx$a.MsgUpdateParams]
];
var aminoConverters8 = {
  "/initia.mstaking.v1.MsgCreateValidator": {
    aminoType: "mstaking/MsgCreateValidator",
    toAmino: (msg) => ({
      description: Description.toAmino(msg.description),
      commission: CommissionRates.toAmino(msg.commission),
      validator_address: msg.validatorAddress,
      pubkey: PubKey.toAminoMsg(
        PubKey.fromProtoMsg(msg.pubkey)
      ),
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      description: Description.fromAmino(msg.description),
      commission: CommissionRates.fromAmino(msg.commission),
      validatorAddress: msg.validator_address,
      pubkey: PubKey.toProtoMsg(PubKey.fromAminoMsg(msg.pubkey)),
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : []
    })
  },
  "/initia.mstaking.v1.MsgDelegate": {
    aminoType: "mstaking/MsgDelegate",
    toAmino: (msg) => ({
      delegator_address: msg.delegatorAddress,
      validator_address: msg.validatorAddress,
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      delegatorAddress: msg.delegator_address,
      validatorAddress: msg.validator_address,
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : []
    })
  },
  "/initia.mstaking.v1.MsgBeginRedelegate": {
    aminoType: "mstaking/MsgBeginRedelegate",
    toAmino: (msg) => ({
      delegator_address: msg.delegatorAddress,
      validator_src_address: msg.validatorSrcAddress,
      validator_dst_address: msg.validatorDstAddress,
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      delegatorAddress: msg.delegator_address,
      validatorSrcAddress: msg.validator_src_address,
      validatorDstAddress: msg.validator_dst_address,
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : []
    })
  },
  "/initia.mstaking.v1.MsgUndelegate": {
    aminoType: "mstaking/MsgUndelegate",
    toAmino: (msg) => ({
      delegator_address: msg.delegatorAddress,
      validator_address: msg.validatorAddress,
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      delegatorAddress: msg.delegator_address,
      validatorAddress: msg.validator_address,
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : []
    })
  },
  "/initia.mstaking.v1.MsgCancelUnbondingDelegation": {
    aminoType: "mstaking/MsgCancelUnbondingDelegation",
    toAmino: (msg) => ({
      delegator_address: msg.delegatorAddress,
      validator_address: msg.validatorAddress,
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin)),
      creation_height: msg.creationHeight.toString()
    }),
    fromAmino: (msg) => ({
      delegatorAddress: msg.delegator_address,
      validatorAddress: msg.validator_address,
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : [],
      creationHeight: BigInt(msg.creation_height)
    })
  },
  "/initia.mstaking.v1.MsgUpdateParams": {
    aminoType: "mstaking/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params6.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params6.fromAmino(msg.params)
    })
  }
};

// src/initia/reward/v1/types.ts
var Params7 = {
  toAmino: (params) => ({
    reward_denom: params.rewardDenom,
    dilution_period: Duration.toAmino(params.dilutionPeriod),
    release_rate: params.releaseRate,
    dilution_rate: params.dilutionRate,
    release_enabled: params.releaseEnabled
  }),
  fromAmino: (params) => ({
    rewardDenom: params.reward_denom,
    dilutionPeriod: Duration.fromAmino(params.dilution_period),
    releaseRate: params.release_rate,
    dilutionRate: params.dilution_rate,
    releaseEnabled: params.release_enabled
  })
};

// src/initia/reward/v1/tx.ts
var registry9 = [
  ["/initia.reward.v1.MsgUpdateParams", tx$b.MsgUpdateParams]
];
var aminoConverters9 = {
  "/initia.reward.v1.MsgUpdateParams": {
    aminoType: "reward/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params7.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params7.fromAmino(msg.params)
    })
  }
};

// src/initia/index.ts
var initiaRegistry = [
  ...registry2,
  ...registry3,
  ...registry4,
  ...registry5,
  ...registry6,
  ...registry7,
  ...registry7,
  ...registry8,
  ...registry9
];
var initiaAminoConverters = {
  ...aminoConverters2,
  ...aminoConverters3,
  ...aminoConverters4,
  ...aminoConverters5,
  ...aminoConverters6,
  ...aminoConverters7,
  ...aminoConverters8,
  ...aminoConverters9
};

// src/minievm/evm/v1/types.ts
var Params8 = {
  toAmino: (params) => ({
    extra_eips: params.extraEips.length === 0 ? void 0 : params.extraEips.map((eip) => eip.toString()),
    allowed_publishers: params.allowedPublishers.length === 0 ? null : params.allowedPublishers,
    allow_custom_erc20: params.allowCustomErc20,
    allowed_custom_erc20s: params.allowedCustomErc20s.length === 0 ? null : params.allowedCustomErc20s,
    fee_denom: params.feeDenom,
    gas_refund_ratio: params.gasRefundRatio,
    num_retain_block_hashes: params.numRetainBlockHashes.toString()
  }),
  fromAmino: (params) => ({
    extraEips: params.extra_eips ? params.extra_eips.map((eip) => BigInt(eip)) : [],
    allowedPublishers: params.allowed_publishers ?? [],
    allowCustomErc20: params.allow_custom_erc20,
    allowedCustomErc20s: params.allowed_custom_erc20s ?? [],
    feeDenom: params.fee_denom,
    gasRefundRatio: params.gas_refund_ratio,
    numRetainBlockHashes: BigInt(params.num_retain_block_hashes)
  })
};
var AccessTuple = {
  toAmino: (accessTuple) => ({
    address: accessTuple.address,
    storage_keys: accessTuple.storageKeys.length === 0 ? void 0 : []
  }),
  fromAmino: (accessTuple) => ({
    address: accessTuple.address,
    storageKeys: accessTuple.storage_keys ?? []
  })
};
var SetCodeAuthorization = {
  toAmino: (accessTuple) => ({
    chain_id: accessTuple.chainId,
    address: accessTuple.address,
    nonce: accessTuple.nonce.toString(),
    signature: bytesToBase64(accessTuple.signature)
  }),
  fromAmino: (accessTuple) => ({
    chainId: accessTuple.chain_id,
    address: accessTuple.address,
    nonce: BigInt(accessTuple.nonce),
    signature: base64ToBytes(accessTuple.signature)
  })
};

// src/minievm/evm/v1/tx.ts
var registry10 = [
  ["/minievm.evm.v1.MsgCreate", tx$c.MsgCreate],
  ["/minievm.evm.v1.MsgCreate2", tx$c.MsgCreate2],
  ["/minievm.evm.v1.MsgCall", tx$c.MsgCall],
  ["/minievm.evm.v1.MsgUpdateParams", tx$c.MsgUpdateParams]
];
var aminoConverters10 = {
  "/minievm.evm.v1.MsgCreate": {
    aminoType: "evm/MsgCreate",
    toAmino: (msg) => ({
      sender: msg.sender,
      code: msg.code,
      value: msg.value,
      access_list: msg.accessList.length === 0 ? null : msg.accessList.map(
        (accessTuple) => AccessTuple.toAmino(accessTuple)
      )
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      code: msg.code,
      value: msg.value,
      accessList: msg.access_list ? msg.access_list.map(
        (accessTuple) => AccessTuple.fromAmino(accessTuple)
      ) : []
    })
  },
  "/minievm.evm.v1.MsgCreate2": {
    aminoType: "evm/MsgCreate2",
    toAmino: (msg) => ({
      sender: msg.sender,
      code: msg.code,
      value: msg.value,
      salt: msg.salt,
      access_list: msg.accessList.length === 0 ? null : msg.accessList.map(
        (accessTuple) => AccessTuple.toAmino(accessTuple)
      )
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      code: msg.code,
      value: msg.value,
      salt: msg.salt,
      accessList: msg.access_list ? msg.access_list.map(
        (accessTuple) => AccessTuple.fromAmino(accessTuple)
      ) : []
    })
  },
  "/minievm.evm.v1.MsgCall": {
    aminoType: "evm/MsgCall",
    toAmino: (msg) => ({
      sender: msg.sender,
      contract_addr: msg.contractAddr,
      input: msg.input,
      value: msg.value,
      auth_list: msg.accessList.length === 0 ? void 0 : msg.authList.map(
        (setCodeAuthorization) => SetCodeAuthorization.toAmino(setCodeAuthorization)
      ),
      access_list: msg.accessList.length === 0 ? null : msg.accessList.map(
        (accessTuple) => AccessTuple.toAmino(accessTuple)
      )
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      contractAddr: msg.contract_addr,
      input: msg.input,
      value: msg.value,
      authList: msg.auth_list ? msg.auth_list.map(
        (setCodeAuthorization) => SetCodeAuthorization.fromAmino(setCodeAuthorization)
      ) : [],
      accessList: msg.access_list ? msg.access_list.map(
        (accessTuple) => AccessTuple.fromAmino(accessTuple)
      ) : []
    })
  },
  "/minievm.evm.v1.MsgUpdateParams": {
    aminoType: "evm/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params8.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params8.fromAmino(msg.params)
    })
  }
};

// src/miniwasm/tokenfactory/v1/types.ts
var Params9 = {
  toAmino: (params) => ({
    denom_creation_fee: params.denomCreationFee.length === 0 ? void 0 : params.denomCreationFee.map((coin) => Coin.toAmino(coin)),
    denom_creation_gas_consume: params.denomCreationGasConsume.toString()
  }),
  fromAmino: (params) => ({
    denomCreationFee: params.denom_creation_fee ? params.denom_creation_fee.map((coin) => Coin.fromAmino(coin)) : [],
    denomCreationGasConsume: BigInt(params.denom_creation_gas_consume)
  })
};

// src/miniwasm/tokenfactory/v1/tx.ts
var registry11 = [
  ["/miniwasm.tokenfactory.v1.MsgCreateDenom", tx$d.MsgCreateDenom],
  ["/miniwasm.tokenfactory.v1.MsgMint", tx$d.MsgMint],
  ["/miniwasm.tokenfactory.v1.MsgBurn", tx$d.MsgBurn],
  ["/miniwasm.tokenfactory.v1.MsgChangeAdmin", tx$d.MsgChangeAdmin],
  ["/miniwasm.tokenfactory.v1.MsgSetBeforeSendHook", tx$d.MsgSetBeforeSendHook],
  ["/miniwasm.tokenfactory.v1.MsgSetDenomMetadata", tx$d.MsgSetDenomMetadata],
  ["/miniwasm.tokenfactory.v1.MsgUpdateParams", tx$d.MsgUpdateParams]
];
var aminoConverters11 = {
  "/miniwasm.tokenfactory.v1.MsgCreateDenom": {
    aminoType: "tokenfactory/MsgCreateDenom",
    toAmino: (msg) => ({
      sender: msg.sender,
      subdenom: msg.subdenom
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      subdenom: msg.subdenom
    })
  },
  "/miniwasm.tokenfactory.v1.MsgMint": {
    aminoType: "tokenfactory/MsgMint",
    toAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.toAmino(msg.amount),
      mint_to_address: msg.mintToAddress
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.fromAmino(msg.amount),
      mintToAddress: msg.mint_to_address
    })
  },
  "/miniwasm.tokenfactory.v1.MsgBurn": {
    aminoType: "tokenfactory/MsgBurn",
    toAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.toAmino(msg.amount)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.fromAmino(msg.amount)
    })
  },
  "/miniwasm.tokenfactory.v1.MsgChangeAdmin": {
    aminoType: "tokenfactory/MsgChangeAdmin",
    toAmino: (msg) => ({
      sender: msg.sender,
      denom: msg.denom,
      new_admin: msg.newAdmin
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      denom: msg.denom,
      newAdmin: msg.new_admin
    })
  },
  "/miniwasm.tokenfactory.v1.MsgSetBeforeSendHook": {
    aminoType: "tokenfactory/MsgSetBeforeSendHook",
    toAmino: (msg) => ({
      sender: msg.sender,
      denom: msg.denom,
      cosmwasm_address: msg.cosmwasmAddress
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      denom: msg.denom,
      cosmwasmAddress: msg.cosmwasm_address
    })
  },
  "/miniwasm.tokenfactory.v1.MsgSetDenomMetadata": {
    aminoType: "tokenfactory/MsgSetDenomMetadata",
    toAmino: (msg) => ({
      sender: msg.sender,
      metadata: Metadata.toAmino(msg.metadata)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      metadata: Metadata.fromAmino(msg.metadata)
    })
  },
  "/miniwasm.tokenfactory.v1.MsgUpdateParams": {
    aminoType: "tokenfactory/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params9.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params9.fromAmino(msg.params)
    })
  }
};
var BridgeConfig = {
  toAmino: (bridgeConfig) => ({
    challenger: bridgeConfig.challenger,
    proposer: bridgeConfig.proposer,
    batch_info: BatchInfo.toAmino(bridgeConfig.batchInfo),
    submission_interval: Duration.toAmino(
      bridgeConfig.submissionInterval
    ),
    finalization_period: Duration.toAmino(
      bridgeConfig.finalizationPeriod
    ),
    submission_start_height: bridgeConfig.submissionStartHeight.toString(),
    oracle_enabled: bridgeConfig.oracleEnabled,
    metadata: bridgeConfig.metadata.length === 0 ? void 0 : bytesToBase64(bridgeConfig.metadata)
  }),
  fromAmino: (bridgeConfig) => ({
    challenger: bridgeConfig.challenger,
    proposer: bridgeConfig.proposer,
    batchInfo: BatchInfo.fromAmino(bridgeConfig.batch_info),
    submissionInterval: Duration.fromAmino(bridgeConfig.submission_interval),
    finalizationPeriod: Duration.fromAmino(bridgeConfig.finalization_period),
    submissionStartHeight: BigInt(bridgeConfig.submission_start_height),
    oracleEnabled: bridgeConfig.oracle_enabled,
    metadata: bridgeConfig.metadata ? base64ToBytes(bridgeConfig.metadata) : new Uint8Array([])
  })
};
var BatchInfo = {
  toAmino: (batchInfo) => ({
    submitter: batchInfo.submitter,
    chain_type: types.batchInfo_ChainTypeToJSON(batchInfo.chainType)
  }),
  fromAmino: (batchInfo) => ({
    submitter: batchInfo.submitter,
    chainType: types.batchInfo_ChainTypeFromJSON(batchInfo.chain_type)
  })
};
var Params10 = {
  toAmino: (params) => ({
    registration_fee: params.registrationFee.length === 0 ? null : params.registrationFee.map((coin) => Coin.toAmino(coin))
  }),
  fromAmino: (params) => ({
    registrationFee: params.registration_fee ? params.registration_fee.map((coin) => Coin.fromAmino(coin)) : []
  })
};

// src/opinit/opchild/v1/types.ts
var BridgeInfo = {
  toAmino: (bridgeInfo) => ({
    bridge_id: bridgeInfo.bridgeId.toString(),
    bridge_addr: bridgeInfo.bridgeAddr,
    l1_chain_id: bridgeInfo.l1ChainId,
    l1_client_id: bridgeInfo.l1ClientId,
    bridge_config: BridgeConfig.toAmino(
      bridgeInfo.bridgeConfig
    )
  }),
  fromAmino: (bridgeInfo) => ({
    bridgeId: BigInt(bridgeInfo.bridge_id),
    bridgeAddr: bridgeInfo.bridge_addr,
    l1ChainId: bridgeInfo.l1_chain_id,
    l1ClientId: bridgeInfo.l1_client_id,
    bridgeConfig: BridgeConfig.fromAmino(bridgeInfo.bridge_config)
  })
};
var Params11 = {
  toAmino: (params) => ({
    max_validators: params.maxValidators,
    historical_entries: params.historicalEntries,
    min_gas_prices: params.minGasPrices.length === 0 ? void 0 : params.minGasPrices.map((coin) => DecCoin.toAmino(coin)),
    bridge_executors: params.bridgeExecutors.length === 0 ? null : params.bridgeExecutors,
    admin: params.admin,
    fee_whitelist: params.feeWhitelist.length === 0 ? null : params.feeWhitelist,
    hook_max_gas: params.hookMaxGas.toString()
  }),
  fromAmino: (params) => ({
    maxValidators: params.max_validators,
    historicalEntries: params.historical_entries,
    minGasPrices: params.min_gas_prices ? params.min_gas_prices.map((coin) => DecCoin.fromAmino(coin)) : [],
    bridgeExecutors: params.bridge_executors ?? [],
    admin: params.admin,
    feeWhitelist: params.fee_whitelist ?? [],
    hookMaxGas: BigInt(params.hook_max_gas)
  })
};

// src/opinit/opchild/v1/tx.ts
var registry12 = [
  ["/opinit.opchild.v1.MsgExecuteMessages", tx$e.MsgExecuteMessages],
  ["/opinit.opchild.v1.MsgFinalizeTokenDeposit", tx$e.MsgFinalizeTokenDeposit],
  ["/opinit.opchild.v1.MsgInitiateTokenWithdrawal", tx$e.MsgInitiateTokenWithdrawal],
  ["/opinit.opchild.v1.MsgSetBridgeInfo", tx$e.MsgSetBridgeInfo],
  ["/opinit.opchild.v1.MsgSpendFeePool", tx$e.MsgSpendFeePool],
  ["/opinit.opchild.v1.MsgUpdateParams", tx$e.MsgUpdateParams],
  ["/opinit.opchild.v1.MsgUpdateOracle", tx$e.MsgUpdateOracle],
  ["/opinit.opchild.v1.MsgMigrateToken", tx$e.MsgMigrateToken]
];
var aminoConverters12 = {
  "/opinit.opchild.v1.MsgFinalizeTokenDeposit": {
    aminoType: "opchild/MsgFinalizeTokenDeposit",
    toAmino: (msg) => ({
      sender: msg.sender,
      from: msg.from,
      to: msg.to,
      amount: Coin.toAmino(msg.amount),
      sequence: msg.sequence.toString(),
      height: msg.height.toString(),
      base_denom: msg.baseDenom,
      data: msg.data.length === 0 ? void 0 : bytesToBase64(msg.data)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      from: msg.from,
      to: msg.to,
      amount: Coin.fromAmino(msg.amount),
      sequence: BigInt(msg.sender),
      height: BigInt(msg.height),
      baseDenom: msg.base_denom,
      data: msg.data ? base64ToBytes(msg.data) : new Uint8Array([])
    })
  },
  "/opinit.opchild.v1.MsgInitiateTokenWithdrawal": {
    aminoType: "opchild/MsgInitiateTokenWithdrawal",
    toAmino: (msg) => ({
      sender: msg.sender,
      to: msg.to,
      amount: Coin.toAmino(msg.amount)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      to: msg.to,
      amount: Coin.fromAmino(msg.amount)
    })
  },
  "/opinit.opchild.v1.MsgSetBridgeInfo": {
    aminoType: "opchild/MsgSetBridgeInfo",
    toAmino: (msg) => ({
      sender: msg.sender,
      bridge_info: BridgeInfo.toAmino(msg.bridgeInfo)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      bridgeInfo: BridgeInfo.fromAmino(msg.bridge_info)
    })
  },
  "/opinit.opchild.v1.MsgSpendFeePool": {
    aminoType: "opchild/MsgSpendFeePool",
    toAmino: (msg) => ({
      authority: msg.authority,
      recipient: msg.recipient,
      amount: msg.amount.length === 0 ? null : msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      recipient: msg.recipient,
      amount: msg.amount ? msg.amount.map((coin) => Coin.fromAmino(coin)) : []
    })
  },
  "/opinit.opchild.v1.MsgUpdateParams": {
    aminoType: "opchild/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params11.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params11.fromAmino(msg.params)
    })
  },
  "/opinit.opchild.v1.MsgUpdateOracle": {
    aminoType: "opchild/MsgUpdateOracle",
    toAmino: (msg) => ({
      sender: msg.sender,
      height: msg.height.toString(),
      data: msg.data.length === 0 ? void 0 : bytesToBase64(msg.data)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      height: BigInt(msg.height),
      data: msg.data ? base64ToBytes(msg.data) : new Uint8Array([])
    })
  },
  "/opinit.opchild.v1.MsgMigrateToken": {
    aminoType: "opchild/MsgMigrateToken",
    toAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.toAmino(msg.amount)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      amount: Coin.fromAmino(msg.amount)
    })
  }
};
function generateMsgExecuteMessagesAminoConverter(registry18, aminoConverters18) {
  const { toAminoMsg, toProtoMsg } = createMsgConverter(
    registry18,
    aminoConverters18
  );
  return {
    "/opinit.opchild.v1.MsgExecuteMessages": {
      aminoType: "opchild/MsgExecuteMessages",
      toAmino: (msg) => ({
        sender: msg.sender,
        messages: msg.messages.map((msg2) => toAminoMsg(msg2))
      }),
      fromAmino: (msg) => ({
        sender: msg.sender,
        messages: msg.messages.map((msg2) => toProtoMsg(msg2))
      })
    }
  };
}
var registry13 = [
  ["/opinit.ophost.v1.MsgRecordBatch", tx$f.MsgRecordBatch],
  ["/opinit.ophost.v1.MsgCreateBridge", tx$f.MsgCreateBridge],
  ["/opinit.ophost.v1.MsgProposeOutput", tx$f.MsgProposeOutput],
  ["/opinit.ophost.v1.MsgDeleteOutput", tx$f.MsgDeleteOutput],
  ["/opinit.ophost.v1.MsgInitiateTokenDeposit", tx$f.MsgInitiateTokenDeposit],
  ["/opinit.ophost.v1.MsgFinalizeTokenWithdrawal", tx$f.MsgFinalizeTokenWithdrawal],
  ["/opinit.ophost.v1.MsgUpdateProposer", tx$f.MsgUpdateProposer],
  ["/opinit.ophost.v1.MsgUpdateChallenger", tx$f.MsgUpdateChallenger],
  ["/opinit.ophost.v1.MsgUpdateBatchInfo", tx$f.MsgUpdateBatchInfo],
  ["/opinit.ophost.v1.MsgUpdateOracleConfig", tx$f.MsgUpdateOracleConfig],
  ["/opinit.ophost.v1.MsgUpdateMetadata", tx$f.MsgUpdateMetadata],
  ["/opinit.ophost.v1.MsgUpdateParams", tx$f.MsgUpdateParams]
];
var aminoConverters13 = {
  "/opinit.ophost.v1.MsgRecordBatch": {
    aminoType: "ophost/MsgRecordBatch",
    toAmino: (msg) => ({
      submitter: msg.submitter,
      bridge_id: msg.bridgeId.toString(),
      batch_bytes: msg.batchBytes.length === 0 ? void 0 : bytesToBase64(msg.batchBytes)
    }),
    fromAmino: (msg) => ({
      submitter: msg.submitter,
      bridgeId: BigInt(msg.bridge_id),
      batchBytes: msg.batch_bytes ? base64ToBytes(msg.batch_bytes) : new Uint8Array([])
    })
  },
  "/opinit.ophost.v1.MsgCreateBridge": {
    aminoType: "ophost/MsgCreateBridge",
    toAmino: (msg) => ({
      creator: msg.creator,
      config: BridgeConfig.toAmino(msg.config)
    }),
    fromAmino: (msg) => ({
      creator: msg.creator,
      config: BridgeConfig.fromAmino(msg.config)
    })
  },
  "/opinit.ophost.v1.MsgProposeOutput": {
    aminoType: "ophost/MsgProposeOutput",
    toAmino: (msg) => ({
      proposer: msg.proposer,
      bridge_id: msg.bridgeId.toString(),
      output_index: msg.outputIndex.toString(),
      l2_block_number: msg.l2BlockNumber.toString(),
      output_root: msg.outputRoot.length === 0 ? void 0 : bytesToBase64(msg.outputRoot)
    }),
    fromAmino: (msg) => ({
      proposer: msg.proposer,
      bridgeId: BigInt(msg.bridge_id),
      outputIndex: BigInt(msg.output_index),
      l2BlockNumber: BigInt(msg.l2_block_number),
      outputRoot: msg.output_root ? base64ToBytes(msg.output_root) : new Uint8Array([])
    })
  },
  "/opinit.ophost.v1.MsgDeleteOutput": {
    aminoType: "ophost/MsgDeleteOutput",
    toAmino: (msg) => ({
      challenger: msg.challenger,
      bridge_id: msg.bridgeId.toString(),
      output_index: msg.outputIndex.toString()
    }),
    fromAmino: (msg) => ({
      challenger: msg.challenger,
      bridgeId: BigInt(msg.bridge_id),
      outputIndex: BigInt(msg.output_index)
    })
  },
  "/opinit.ophost.v1.MsgInitiateTokenDeposit": {
    aminoType: "ophost/MsgInitiateTokenDeposit",
    toAmino: (msg) => ({
      sender: msg.sender,
      bridge_id: msg.bridgeId.toString(),
      to: msg.to,
      amount: Coin.toAmino(msg.amount),
      data: msg.data.length === 0 ? void 0 : bytesToBase64(msg.data)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      bridgeId: BigInt(msg.bridge_id),
      to: msg.to,
      amount: Coin.fromAmino(msg.amount),
      data: msg.data ? base64ToBytes(msg.data) : new Uint8Array([])
    })
  },
  "/opinit.ophost.v1.MsgFinalizeTokenWithdrawal": {
    aminoType: "ophost/MsgFinalizeTokenWithdrawal",
    toAmino: (msg) => ({
      sender: msg.sender,
      bridge_id: msg.bridgeId.toString(),
      output_index: msg.outputIndex.toString(),
      withdrawal_proofs: msg.withdrawalProofs.length === 0 ? void 0 : msg.withdrawalProofs.map((proof) => bytesToBase64(proof)),
      from: msg.from,
      to: msg.to,
      sequence: msg.sequence.toString(),
      amount: Coin.toAmino(msg.amount),
      version: bytesToBase64(msg.version),
      storage_root: bytesToBase64(msg.storageRoot),
      last_block_hash: bytesToBase64(msg.lastBlockHash)
    }),
    fromAmino: (msg) => ({
      sender: msg.sender,
      bridgeId: BigInt(msg.bridge_id),
      outputIndex: BigInt(msg.output_index),
      withdrawalProofs: msg.withdrawal_proofs ? msg.withdrawal_proofs.map((proof) => base64ToBytes(proof)) : [],
      from: msg.from,
      to: msg.to,
      sequence: BigInt(msg.sequence),
      amount: Coin.fromAmino(msg.amount),
      version: base64ToBytes(msg.version),
      storageRoot: base64ToBytes(msg.storage_root),
      lastBlockHash: base64ToBytes(msg.last_block_hash)
    })
  },
  "/opinit.ophost.v1.MsgUpdateProposer": {
    aminoType: "ophost/MsgUpdateProposer",
    toAmino: (msg) => ({
      authority: msg.authority,
      bridge_id: msg.bridgeId.toString(),
      new_proposer: msg.newProposer
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      bridgeId: BigInt(msg.bridge_id),
      newProposer: msg.new_proposer
    })
  },
  "/opinit.ophost.v1.MsgUpdateChallenger": {
    aminoType: "ophost/MsgUpdateChallenger",
    toAmino: (msg) => ({
      authority: msg.authority,
      bridge_id: msg.bridgeId.toString(),
      challenger: msg.challenger
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      bridgeId: BigInt(msg.bridge_id),
      challenger: msg.challenger
    })
  },
  "/opinit.ophost.v1.MsgUpdateBatchInfo": {
    aminoType: "ophost/MsgUpdateBatchInfo",
    toAmino: (msg) => ({
      authority: msg.authority,
      bridge_id: msg.bridgeId.toString(),
      new_batch_info: BatchInfo.toAmino(msg.newBatchInfo)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      bridgeId: BigInt(msg.bridge_id),
      newBatchInfo: BatchInfo.fromAmino(msg.new_batch_info)
    })
  },
  "/opinit.ophost.v1.MsgUpdateOracleConfig": {
    aminoType: "ophost/MsgUpdateOracleConfig",
    toAmino: (msg) => ({
      authority: msg.authority,
      bridge_id: msg.bridgeId.toString(),
      oracle_enabled: msg.oracleEnabled
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      bridgeId: BigInt(msg.bridge_id),
      oracleEnabled: msg.oracle_enabled
    })
  },
  "/opinit.ophost.v1.MsgUpdateMetadata": {
    aminoType: "ophost/MsgUpdateMetadata",
    toAmino: (msg) => ({
      authority: msg.authority,
      bridge_id: msg.bridgeId.toString(),
      metadata: bytesToBase64(msg.metadata)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      bridgeId: BigInt(msg.bridge_id),
      metadata: base64ToBytes(msg.metadata)
    })
  },
  "/opinit.ophost.v1.MsgUpdateParams": {
    aminoType: "ophost/MsgUpdateParams",
    toAmino: (msg) => ({
      authority: msg.authority,
      params: Params10.toAmino(msg.params)
    }),
    fromAmino: (msg) => ({
      authority: msg.authority,
      params: Params10.fromAmino(msg.params)
    })
  }
};

// src/opinit/index.ts
var opinitRegistry = [
  ...registry12,
  ...registry13
];
var opinitAminoConverters = {
  ...aminoConverters12,
  ...aminoConverters13
};
var AccessConfig = {
  toAmino: (msg) => ({
    permission: types$2.accessTypeToJSON(msg.permission),
    addresses: msg.addresses
  }),
  fromAmino: (msg) => ({
    permission: types$2.accessTypeFromJSON(msg.permission),
    addresses: msg.addresses
  })
};

// src/miniwasm/wasm/v1/authz.ts
var CodeGrant = {
  toAmino: (msg) => ({
    code_hash: bytesToBase64(msg.codeHash),
    instantiate_permission: msg.instantiatePermission ? AccessConfig.toAmino(msg.instantiatePermission) : void 0
  }),
  fromAmino: (msg) => ({
    codeHash: base64ToBytes(msg.code_hash),
    instantiatePermission: msg.instantiate_permission ? AccessConfig.fromAmino(msg.instantiate_permission) : void 0
  })
};
var ContractGrant = {
  toAmino: (msg) => ({
    contract: msg.contract,
    limit: ContractLimit.toAmino(msg.limit),
    filter: ContractFilter.toAmino(msg.filter)
  }),
  fromAmino: (msg) => ({
    contract: msg.contract,
    limit: ContractLimit.fromAmino(msg.limit),
    filter: ContractFilter.fromAmino(msg.filter)
  })
};
var ContractLimit;
((ContractLimit2) => {
  function toAmino(msg) {
    switch (msg.typeUrl) {
      case "/cosmwasm.wasm.v1.MaxCallsLimit": {
        const maxCallsLimit = authz.MaxCallsLimit.decode(msg.value);
        return {
          type: "wasm/MaxCallsLimit",
          value: { remaining: maxCallsLimit.remaining.toString() }
        };
      }
      case "/cosmwasm.wasm.v1.MaxFundsLimit": {
        const maxFundsLimit = authz.MaxFundsLimit.decode(msg.value);
        return {
          type: "wasm/MaxFundsLimit",
          value: {
            amounts: maxFundsLimit.amounts.map((coin) => Coin.toAmino(coin))
          }
        };
      }
      case "/cosmwasm.wasm.v1.CombinedLimit": {
        const combinedLimit = authz.CombinedLimit.decode(msg.value);
        return {
          type: "wasm/CombinedLimit",
          value: {
            calls_remaining: combinedLimit.callsRemaining.toString(),
            amounts: combinedLimit.amounts.map((coin) => Coin.toAmino(coin))
          }
        };
      }
      default:
        throw new Error(`Unsupported typeUrl: ${msg.typeUrl}`);
    }
  }
  ContractLimit2.toAmino = toAmino;
  function fromAmino(msg) {
    switch (msg.type) {
      case "wasm/MaxCallsLimit":
        return {
          typeUrl: "/cosmwasm.wasm.v1.MaxCallsLimit",
          value: authz.MaxCallsLimit.encode({
            remaining: BigInt(msg.value.remaining)
          }).finish()
        };
      case "wasm/MaxFundsLimit":
        return {
          typeUrl: "/cosmwasm.wasm.v1.MaxFundsLimit",
          value: authz.MaxFundsLimit.encode({
            amounts: msg.value.amounts.map((coin) => Coin.fromAmino(coin))
          }).finish()
        };
      case "wasm/CombinedLimit":
        return {
          typeUrl: "/cosmwasm.wasm.v1.CombinedLimit",
          value: authz.CombinedLimit.encode({
            callsRemaining: BigInt(msg.value.calls_remaining),
            amounts: msg.value.amounts.map((coin) => Coin.fromAmino(coin))
          }).finish()
        };
      default:
        throw new Error(`Unsupported type`);
    }
  }
  ContractLimit2.fromAmino = fromAmino;
})(ContractLimit || (ContractLimit = {}));
var ContractFilter;
((ContractFilter2) => {
  function toAmino(msg) {
    switch (msg.typeUrl) {
      case "/cosmwasm.wasm.v1.AllowAllMessagesFilter": {
        return {
          type: "wasm/AllowAllMessagesFilter",
          value: {}
        };
      }
      case "/cosmwasm.wasm.v1.AcceptedMessageKeysFilter": {
        const maxFundsLimit = authz.AcceptedMessageKeysFilter.decode(msg.value);
        return {
          type: "wasm/AcceptedMessageKeysFilter",
          value: {
            keys: maxFundsLimit.keys
          }
        };
      }
      case "/cosmwasm.wasm.v1.AcceptedMessagesFilter": {
        const acceptedMessagesFilter = authz.AcceptedMessagesFilter.decode(msg.value);
        return {
          type: "wasm/AcceptedMessagesFilter",
          value: {
            messages: acceptedMessagesFilter.messages.map(
              (message) => bytesToBase64(message)
            )
          }
        };
      }
      default:
        throw new Error(`Unsupported typeUrl: ${msg.typeUrl}`);
    }
  }
  ContractFilter2.toAmino = toAmino;
  function fromAmino(msg) {
    switch (msg.type) {
      case "wasm/AllowAllMessagesFilter":
        return {
          typeUrl: "/cosmwasm.wasm.v1.AllowAllMessagesFilter",
          value: authz.AllowAllMessagesFilter.encode({}).finish()
        };
      case "wasm/AcceptedMessageKeysFilter":
        return {
          typeUrl: "/cosmwasm.wasm.v1.AcceptedMessageKeysFilter",
          value: authz.AcceptedMessageKeysFilter.encode({
            keys: msg.value.keys
          }).finish()
        };
      case "wasm/AcceptedMessagesFilter":
        return {
          typeUrl: "/cosmwasm.wasm.v1.AcceptedMessagesFilter",
          value: authz.AcceptedMessagesFilter.encode({
            messages: msg.value.messages.map(
              (message) => base64ToBytes(message)
            )
          }).finish()
        };
      default:
        throw new Error(`Unsupported type`);
    }
  }
  ContractFilter2.fromAmino = fromAmino;
})(ContractFilter || (ContractFilter = {}));
var AuthorizationMsg = {
  toAmino: (msg) => {
    switch (msg.typeUrl) {
      case "/cosmos.authz.v1beta1.GenericAuthorization": {
        const genericAuthorization = authz$4.GenericAuthorization.decode(msg.value);
        return {
          type: "cosmos-sdk/GenericAuthorization",
          value: { msg: genericAuthorization.msg }
        };
      }
      case "/cosmos.bank.v1beta1.SendAuthorization": {
        const sendAuthorization = authz$3.SendAuthorization.decode(msg.value);
        return {
          type: "cosmos-sdk/SendAuthorization",
          value: {
            spend_limit: sendAuthorization.spendLimit.map(
              (coin) => Coin.toAmino(coin)
            ),
            allow_list: sendAuthorization.allowList
          }
        };
      }
      case "/initia.mstaking.v1.StakeAuthorization": {
        const stakeAuthorization = authz$2.StakeAuthorization.decode(msg.value);
        return {
          type: "mstaking/StakeAuthorization",
          value: {
            max_tokens: stakeAuthorization.maxTokens.map(
              (coin) => Coin.toAmino(coin)
            ),
            allow_list: stakeAuthorization.allowList,
            deny_list: stakeAuthorization.denyList,
            authorization_type: authz$2.authorizationTypeToJSON(
              stakeAuthorization.authorizationType
            )
          }
        };
      }
      case "/initia.move.v1.PublishAuthorization": {
        const publishAuthorization = authz$1.PublishAuthorization.decode(msg.value);
        return {
          type: "move/PublishAuthorization",
          value: {
            module_names: publishAuthorization.moduleNames
          }
        };
      }
      case "/initia.move.v1.ExecuteAuthorization": {
        const executeAuthorization = authz$1.ExecuteAuthorization.decode(msg.value);
        return {
          type: "move/ExecuteAuthorization",
          value: {
            items: executeAuthorization.items.map(
              (item) => ExecuteAuthorizationItem.toAmino(item)
            )
          }
        };
      }
      case "/initia.wasm.v1.StoreCodeAuthorization": {
        const storeCodeAuthorization = authz.StoreCodeAuthorization.decode(msg.value);
        return {
          type: "wasm/StoreCodeAuthorization",
          value: {
            grants: storeCodeAuthorization.grants.length === 0 ? null : storeCodeAuthorization.grants.map(
              (grant) => CodeGrant.toAmino(grant)
            )
          }
        };
      }
      case "/initia.wasm.v1.ContractExecutionAuthorization": {
        const contractExecutionAuthorization = authz.ContractExecutionAuthorization.decode(msg.value);
        return {
          type: "wasm/ContractExecutionAuthorization",
          value: {
            grants: contractExecutionAuthorization.grants.length === 0 ? null : contractExecutionAuthorization.grants.map(
              (grant) => ContractGrant.toAmino(grant)
            )
          }
        };
      }
      case "/initia.wasm.v1.ContractMigrationAuthorization": {
        const contractMigrationAuthorization = authz.ContractMigrationAuthorization.decode(msg.value);
        return {
          type: "wasm/ContractMigrationAuthorization",
          value: {
            grants: contractMigrationAuthorization.grants.length === 0 ? null : contractMigrationAuthorization.grants.map(
              (grant) => ContractGrant.toAmino(grant)
            )
          }
        };
      }
      default:
        throw new Error(`Unsupported typeUrl: ${msg.typeUrl}`);
    }
  },
  fromAmino: (msg) => {
    switch (msg.type) {
      case "cosmos-sdk/GenericAuthorization": {
        const value = authz$4.GenericAuthorization.fromPartial({
          msg: msg.value.msg
        });
        return {
          typeUrl: "/cosmos.authz.v1beta1.GenericAuthorization",
          value: authz$4.GenericAuthorization.encode(value).finish()
        };
      }
      case "cosmos-sdk/SendAuthorization": {
        const value = authz$3.SendAuthorization.fromPartial({
          spendLimit: msg.value.spend_limit.map((coin) => Coin.fromAmino(coin)),
          allowList: msg.value.allow_list
        });
        return {
          typeUrl: "/cosmos.bank.v1beta1.SendAuthorization",
          value: authz$3.SendAuthorization.encode(value).finish()
        };
      }
      case "mstaking/StakeAuthorization": {
        const value = authz$2.StakeAuthorization.fromPartial({
          maxTokens: msg.value.max_tokens.map((coin) => Coin.fromAmino(coin)),
          allowList: msg.value.allow_list,
          denyList: msg.value.deny_list,
          authorizationType: authz$2.authorizationTypeFromJSON(
            msg.value.authorization_type
          )
        });
        return {
          typeUrl: "/initia.mstaking.v1.StakeAuthorization",
          value: authz$2.StakeAuthorization.encode(value).finish()
        };
      }
      case "move/PublishAuthorization": {
        const value = authz$1.PublishAuthorization.fromPartial({
          moduleNames: msg.value.module_names
        });
        return {
          typeUrl: "/initia.move.v1.PublishAuthorization",
          value: authz$1.PublishAuthorization.encode(value).finish()
        };
      }
      case "move/ExecuteAuthorization": {
        const value = authz$1.ExecuteAuthorization.fromPartial({
          items: msg.value.items.map(
            (item) => ExecuteAuthorizationItem.fromAmino(item)
          )
        });
        return {
          typeUrl: "/initia.move.v1.ExecuteAuthorization",
          value: authz$1.ExecuteAuthorization.encode(value).finish()
        };
      }
      case "wasm/StoreCodeAuthorization": {
        const value = authz.StoreCodeAuthorization.fromPartial({
          grants: msg.value.grants ? msg.value.grants.map((grant) => CodeGrant.fromAmino(grant)) : []
        });
        return {
          typeUrl: "/initia.wasm.v1.StoreCodeAuthorization",
          value: authz.StoreCodeAuthorization.encode(value).finish()
        };
      }
      case "wasm/ContractExecutionAuthorization": {
        const value = authz.ContractExecutionAuthorization.fromPartial({
          grants: msg.value.grants ? msg.value.grants.map((grant) => ContractGrant.fromAmino(grant)) : []
        });
        return {
          typeUrl: "/initia.wasm.v1.ContractExecutionAuthorization",
          value: authz.ContractExecutionAuthorization.encode(value).finish()
        };
      }
      case "wasm/ContractMigrationAuthorization": {
        const value = authz.ContractMigrationAuthorization.fromPartial({
          grants: msg.value.grants ? msg.value.grants.map((grant) => ContractGrant.fromAmino(grant)) : []
        });
        return {
          typeUrl: "/initia.wasm.v1.ContractMigrationAuthorization",
          value: authz.ContractMigrationAuthorization.encode(value).finish()
        };
      }
      default:
        throw new Error(`Unsupported type`);
    }
  }
};

// src/cosmos/authz/v1beta1/authz.ts
var Grant = {
  toAmino: (msg) => ({
    authorization: AuthorizationMsg.toAmino(msg.authorization),
    expiration: msg.expiration ? msg.expiration.toISOString() : void 0
  }),
  fromAmino: (msg) => ({
    authorization: AuthorizationMsg.fromAmino(msg.authorization),
    expiration: msg.expiration ? new Date(msg.expiration) : void 0
  })
};

// src/cosmos/authz/v1beta1/tx.ts
var registry14 = [
  ["/cosmos.authz.v1beta1.MsgGrant", tx.MsgGrant],
  ["/cosmos.authz.v1beta1.MsgExec", tx.MsgExec],
  ["/cosmos.authz.v1beta1.MsgRevoke", tx.MsgRevoke]
];
var aminoConverters14 = {
  "/cosmos.authz.v1beta1.MsgGrant": {
    aminoType: "cosmos-sdk/MsgGrant",
    toAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      grant: Grant.toAmino(msg.grant)
    }),
    fromAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      grant: Grant.fromAmino(msg.grant)
    })
  },
  "/cosmos.authz.v1beta1.MsgRevoke": {
    aminoType: "cosmos-sdk/MsgRevoke",
    toAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      msg_type_url: msg.msgTypeUrl
    }),
    fromAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      msgTypeUrl: msg.msg_type_url
    })
  }
};
function generateMsgExecAminoConverter(registry18, aminoConverters18) {
  const { toAminoMsg, toProtoMsg } = createMsgConverter(
    registry18,
    aminoConverters18
  );
  return {
    "/cosmos.authz.v1beta1.MsgExec": {
      aminoType: "cosmos-sdk/MsgExec",
      toAmino: (msg) => ({
        grantee: msg.grantee,
        msgs: msg.msgs.map((msg2) => toAminoMsg(msg2))
      }),
      fromAmino: (msg) => ({
        grantee: msg.grantee,
        msgs: msg.msgs.map((msg2) => toProtoMsg(msg2))
      })
    }
  };
}
var BasicAllowance = {
  toAmino: (msg) => ({
    spend_limit: msg.spendLimit.map((coin) => Coin.toAmino(coin)),
    expiration: msg.expiration ? msg.expiration.toISOString() : void 0
  }),
  fromAmino: (msg) => ({
    spendLimit: msg.spend_limit.map((coin) => Coin.fromAmino(coin)),
    expiration: msg.expiration ? new Date(msg.expiration) : void 0
  })
};
var PeriodicAllowance = {
  toAmino: (msg) => ({
    basic: BasicAllowance.toAmino(msg.basic),
    period: Duration.toAmino(msg.period),
    period_spend_limit: msg.periodSpendLimit.map((coin) => Coin.toAmino(coin)),
    period_can_spend: msg.periodCanSpend.map((coin) => Coin.toAmino(coin)),
    period_reset: msg.periodReset.toISOString()
  }),
  fromAmino: (msg) => ({
    basic: BasicAllowance.fromAmino(msg.basic),
    period: Duration.fromAmino(msg.period),
    periodSpendLimit: msg.period_spend_limit.map(
      (coin) => Coin.fromAmino(coin)
    ),
    periodCanSpend: msg.period_can_spend.map((coin) => Coin.fromAmino(coin)),
    periodReset: new Date(msg.period_reset)
  })
};
var Allowance;
((Allowance2) => {
  function toAmino(msg) {
    switch (msg.typeUrl) {
      case "/cosmos.feegrant.v1beta1.BasicAllowance": {
        const basicAllowance = feegrant.BasicAllowance.decode(msg.value);
        return {
          type: "cosmos-sdk/BasicAllowance",
          value: BasicAllowance.toAmino(basicAllowance)
        };
      }
      case "/cosmos.feegrant.v1beta1.PeriodicAllowance": {
        const periodicAllowance = feegrant.PeriodicAllowance.decode(msg.value);
        return {
          type: "cosmos-sdk/PeriodicAllowance",
          value: PeriodicAllowance.toAmino(periodicAllowance)
        };
      }
      default:
        throw new Error(`Unsupported typeUrl: ${msg.typeUrl}`);
    }
  }
  Allowance2.toAmino = toAmino;
  function fromAmino(msg) {
    switch (msg.type) {
      case "cosmos-sdk/BasicAllowance":
        return {
          typeUrl: "/cosmos.feegrant.v1beta1.BasicAllowance",
          value: feegrant.BasicAllowance.encode(
            BasicAllowance.fromAmino(msg.value)
          ).finish()
        };
      case "cosmos-sdk/PeriodicAllowance":
        return {
          typeUrl: "/cosmos.feegrant.v1beta1.PeriodicAllowance",
          value: feegrant.PeriodicAllowance.encode(
            PeriodicAllowance.fromAmino(msg.value)
          ).finish()
        };
      default:
        throw new Error(`Unsupported type`);
    }
  }
  Allowance2.fromAmino = fromAmino;
})(Allowance || (Allowance = {}));

// src/cosmos/feegrant/v1beta1/tx.ts
var registry15 = [
  ["/cosmos.feegrant.v1beta1.MsgGrantAllowance", tx$1.MsgGrantAllowance],
  ["/cosmos.feegrant.v1beta1.MsgRevokeAllowance", tx$1.MsgRevokeAllowance],
  ["/cosmos.feegrant.v1beta1.MsgPruneAllowances", tx$1.MsgPruneAllowances]
];
var aminoConverters15 = {
  "/cosmos.feegrant.v1beta1.MsgGrantAllowance": {
    aminoType: "cosmos-sdk/MsgGrantAllowance",
    toAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      allowance: Allowance.toAmino(msg.allowance)
    }),
    fromAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee,
      allowance: Allowance.fromAmino(msg.allowance)
    })
  },
  "/cosmos.feegrant.v1beta1.MsgRevokeAllowance": {
    aminoType: "cosmos-sdk/MsgRevokeAllowance",
    toAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee
    }),
    fromAmino: (msg) => ({
      granter: msg.granter,
      grantee: msg.grantee
    })
  },
  "/cosmos.feegrant.v1beta1.MsgPruneAllowances": {
    aminoType: "cosmos-sdk/MsgPruneAllowances",
    toAmino: (msg) => ({
      pruner: msg.pruner
    }),
    fromAmino: (msg) => ({
      pruner: msg.pruner
    })
  }
};
function generateMsgSubmitProposalAminoConverter(registry18, aminoConverters18) {
  const { toAminoMsg, toProtoMsg } = createMsgConverter(
    registry18,
    aminoConverters18
  );
  return {
    "/cosmos.gov.v1.MsgSubmitProposal": {
      aminoType: "cosmos-sdk/v1/MsgSubmitProposal",
      toAmino: (msg) => ({
        messages: msg.messages.length === 0 ? void 0 : msg.messages.map((msg2) => toAminoMsg(msg2)),
        initial_deposit: msg.initialDeposit.map((coin) => Coin.toAmino(coin)),
        proposer: msg.proposer,
        metadata: msg.metadata === "" ? void 0 : msg.metadata,
        title: msg.title,
        summary: msg.summary,
        expedited: msg.expedited ? true : void 0
      }),
      fromAmino: (msg) => ({
        messages: msg.messages ? msg.messages.map((msg2) => toProtoMsg(msg2)) : [],
        initialDeposit: msg.initial_deposit.map((coin) => Coin.fromAmino(coin)),
        proposer: msg.proposer,
        metadata: msg.metadata ?? "",
        title: msg.title,
        summary: msg.summary,
        expedited: msg.expedited ?? false
      })
    }
  };
}
var registry16 = [
  ["/cosmos.gov.v1.MsgVote", tx$2.MsgVote],
  ["/cosmos.gov.v1.MsgDeposit", tx$2.MsgDeposit]
];
var aminoConverters16 = {
  "/cosmos.gov.v1.MsgVote": {
    aminoType: "cosmos-sdk/v1/MsgVote",
    toAmino: (msg) => ({
      proposal_id: msg.proposalId.toString(),
      voter: msg.voter,
      option: msg.option,
      metadata: msg.metadata === "" ? void 0 : msg.metadata
    }),
    fromAmino: (msg) => ({
      proposalId: BigInt(msg.proposal_id),
      voter: msg.voter,
      option: msg.option,
      metadata: msg.metadata ?? ""
    })
  },
  "/cosmos.gov.v1.MsgDeposit": {
    aminoType: "cosmos-sdk/v1/MsgDeposit",
    toAmino: (msg) => ({
      proposal_id: msg.proposalId.toString(),
      depositor: msg.depositor,
      amount: msg.amount.map((coin) => Coin.toAmino(coin))
    }),
    fromAmino: (msg) => ({
      proposalId: BigInt(msg.proposal_id),
      depositor: msg.depositor,
      amount: msg.amount.map((coin) => Coin.fromAmino(coin))
    })
  }
};

// src/cosmos/index.ts
var cosmosRegistry = [
  ...registry14,
  ...registry15,
  ...registry16
];
var cosmosAminoConverters = {
  ...aminoConverters14,
  ...aminoConverters15,
  ...aminoConverters16
};

// src/index.ts
var protoRegistry = [
  ...stargate.defaultRegistryTypes,
  ...cosmosRegistry,
  ...cosmwasmStargate.wasmTypes,
  ...ibcApplicationsRegistry,
  ...initiaRegistry,
  ...registry10,
  ...registry11,
  ...opinitRegistry
];
var registry17 = new protoSigning.Registry(protoRegistry);
var tmpAminoConverters = {
  ...stargate.createDefaultAminoConverters(),
  ...cosmwasmStargate.createWasmAminoConverters(),
  ...cosmosAminoConverters,
  ...ibcApplicationsAminoConverters,
  ...initiaAminoConverters,
  ...aminoConverters10,
  ...aminoConverters11,
  ...opinitAminoConverters
};
var msgSubmitTxAminoConverter = generateMsgSubmitTxAminoConverter(
  protoRegistry,
  tmpAminoConverters
);
var msgExecuteMessagesAminoConverter = generateMsgExecuteMessagesAminoConverter(protoRegistry, tmpAminoConverters);
var msgExecAminoConverter = generateMsgExecAminoConverter(
  protoRegistry,
  tmpAminoConverters
);
var msgSubmitProposalAminoConverter = generateMsgSubmitProposalAminoConverter(
  protoRegistry,
  tmpAminoConverters
);
var aminoConverters17 = {
  ...tmpAminoConverters,
  ...msgSubmitTxAminoConverter,
  ...msgExecuteMessagesAminoConverter,
  ...msgExecAminoConverter,
  ...msgSubmitProposalAminoConverter
};
var aminoTypes = new stargate.AminoTypes(aminoConverters17);

exports.aminoConverters = aminoConverters17;
exports.aminoTypes = aminoTypes;
exports.generateMsgExecuteMessagesAminoConverter = generateMsgExecuteMessagesAminoConverter;
exports.generateMsgSubmitProposalAminoConverter = generateMsgSubmitProposalAminoConverter;
exports.generateMsgSubmitTxAminoConverter = generateMsgSubmitTxAminoConverter;
exports.protoRegistry = protoRegistry;
exports.registry = registry17;
