// @initia/initia.proto/minievm/evm/v1/authz.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: minievm/evm/v1/authz.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallAuthorization = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "minievm.evm.v1";
function createBaseCallAuthorization() {
    return { contracts: [] };
}
exports.CallAuthorization = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.contracts) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCallAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.contracts.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { contracts: gt.Array.isArray(object?.contracts) ? object.contracts.map((e) => gt.String(e)) : [] };
    },
    toJSON(message) {
        const obj = {};
        if (message.contracts?.length) {
            obj.contracts = message.contracts;
        }
        return obj;
    },
    create(base) {
        return exports.CallAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCallAuthorization();
        message.contracts = object.contracts?.map((e) => e) || [];
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
