// @initia/initia.proto/minievm/evm/v1/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: minievm/evm/v1/types.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.SetCodeAuthorization = exports.ERC721ClassInfo = exports.ERC721OriginTokenInfo = exports.AccessTuple = exports.Log = exports.GasEnforcement = exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "minievm.evm.v1";
function createBaseParams() {
    return {
        extraEips: [],
        allowedPublishers: [],
        allowCustomErc20: false,
        allowedCustomErc20s: [],
        feeDenom: "",
        gasRefundRatio: "",
        numRetainBlockHashes: 0n,
        gasEnforcement: undefined,
    };
}
exports.Params = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        writer.uint32(10).fork();
        for (const v of message.extraEips) {
            if (BigInt.asIntN(64, v) !== v) {
                throw new gt.Error("a value provided in array field extraEips of type int64 is too large");
            }
            writer.int64(v);
        }
        writer.join();
        for (const v of message.allowedPublishers) {
            writer.uint32(18).string(v);
        }
        if (message.allowCustomErc20 !== false) {
            writer.uint32(24).bool(message.allowCustomErc20);
        }
        for (const v of message.allowedCustomErc20s) {
            writer.uint32(34).string(v);
        }
        if (message.feeDenom !== "") {
            writer.uint32(42).string(message.feeDenom);
        }
        if (message.gasRefundRatio !== "") {
            writer.uint32(50).string(message.gasRefundRatio);
        }
        if (message.numRetainBlockHashes !== 0n) {
            if (BigInt.asUintN(64, message.numRetainBlockHashes) !== message.numRetainBlockHashes) {
                throw new gt.Error("value provided for field message.numRetainBlockHashes of type uint64 too large");
            }
            writer.uint32(56).uint64(message.numRetainBlockHashes);
        }
        if (message.gasEnforcement !== undefined) {
            exports.GasEnforcement.encode(message.gasEnforcement, writer.uint32(66).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag === 8) {
                        message.extraEips.push(reader.int64());
                        continue;
                    }
                    if (tag === 10) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.extraEips.push(reader.int64());
                        }
                        continue;
                    }
                    break;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.allowedPublishers.push(reader.string());
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.allowCustomErc20 = reader.bool();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.allowedCustomErc20s.push(reader.string());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.feeDenom = reader.string();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.gasRefundRatio = reader.string();
                    continue;
                }
                case 7: {
                    if (tag !== 56) {
                        break;
                    }
                    message.numRetainBlockHashes = reader.uint64();
                    continue;
                }
                case 8: {
                    if (tag !== 66) {
                        break;
                    }
                    message.gasEnforcement = exports.GasEnforcement.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            extraEips: gt.Array.isArray(object?.extraEips) ? object.extraEips.map((e) => BigInt(e)) : [],
            allowedPublishers: gt.Array.isArray(object?.allowedPublishers)
                ? object.allowedPublishers.map((e) => gt.String(e))
                : [],
            allowCustomErc20: isSet(object.allowCustomErc20) ? gt.Boolean(object.allowCustomErc20) : false,
            allowedCustomErc20s: gt.Array.isArray(object?.allowedCustomErc20s)
                ? object.allowedCustomErc20s.map((e) => gt.String(e))
                : [],
            feeDenom: isSet(object.feeDenom) ? gt.String(object.feeDenom) : "",
            gasRefundRatio: isSet(object.gasRefundRatio) ? gt.String(object.gasRefundRatio) : "",
            numRetainBlockHashes: isSet(object.numRetainBlockHashes) ? BigInt(object.numRetainBlockHashes) : 0n,
            gasEnforcement: isSet(object.gasEnforcement) ? exports.GasEnforcement.fromJSON(object.gasEnforcement) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.extraEips?.length) {
            obj.extraEips = message.extraEips.map((e) => e.toString());
        }
        if (message.allowedPublishers?.length) {
            obj.allowedPublishers = message.allowedPublishers;
        }
        if (message.allowCustomErc20 !== false) {
            obj.allowCustomErc20 = message.allowCustomErc20;
        }
        if (message.allowedCustomErc20s?.length) {
            obj.allowedCustomErc20s = message.allowedCustomErc20s;
        }
        if (message.feeDenom !== "") {
            obj.feeDenom = message.feeDenom;
        }
        if (message.gasRefundRatio !== "") {
            obj.gasRefundRatio = message.gasRefundRatio;
        }
        if (message.numRetainBlockHashes !== 0n) {
            obj.numRetainBlockHashes = message.numRetainBlockHashes.toString();
        }
        if (message.gasEnforcement !== undefined) {
            obj.gasEnforcement = exports.GasEnforcement.toJSON(message.gasEnforcement);
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.extraEips = object.extraEips?.map((e) => e) || [];
        message.allowedPublishers = object.allowedPublishers?.map((e) => e) || [];
        message.allowCustomErc20 = object.allowCustomErc20 ?? false;
        message.allowedCustomErc20s = object.allowedCustomErc20s?.map((e) => e) || [];
        message.feeDenom = object.feeDenom ?? "";
        message.gasRefundRatio = object.gasRefundRatio ?? "";
        message.numRetainBlockHashes = object.numRetainBlockHashes ?? 0n;
        message.gasEnforcement = (object.gasEnforcement !== undefined && object.gasEnforcement !== null)
            ? exports.GasEnforcement.fromPartial(object.gasEnforcement)
            : undefined;
        return message;
    },
};
function createBaseGasEnforcement() {
    return { maxGasFeeCap: "", maxGasLimit: 0n, unlimitedGasSenders: [] };
}
exports.GasEnforcement = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.maxGasFeeCap !== "") {
            writer.uint32(10).string(message.maxGasFeeCap);
        }
        if (message.maxGasLimit !== 0n) {
            if (BigInt.asUintN(64, message.maxGasLimit) !== message.maxGasLimit) {
                throw new gt.Error("value provided for field message.maxGasLimit of type uint64 too large");
            }
            writer.uint32(16).uint64(message.maxGasLimit);
        }
        for (const v of message.unlimitedGasSenders) {
            writer.uint32(26).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGasEnforcement();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.maxGasFeeCap = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.maxGasLimit = reader.uint64();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.unlimitedGasSenders.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            maxGasFeeCap: isSet(object.maxGasFeeCap) ? gt.String(object.maxGasFeeCap) : "",
            maxGasLimit: isSet(object.maxGasLimit) ? BigInt(object.maxGasLimit) : 0n,
            unlimitedGasSenders: gt.Array.isArray(object?.unlimitedGasSenders)
                ? object.unlimitedGasSenders.map((e) => gt.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.maxGasFeeCap !== "") {
            obj.maxGasFeeCap = message.maxGasFeeCap;
        }
        if (message.maxGasLimit !== 0n) {
            obj.maxGasLimit = message.maxGasLimit.toString();
        }
        if (message.unlimitedGasSenders?.length) {
            obj.unlimitedGasSenders = message.unlimitedGasSenders;
        }
        return obj;
    },
    create(base) {
        return exports.GasEnforcement.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGasEnforcement();
        message.maxGasFeeCap = object.maxGasFeeCap ?? "";
        message.maxGasLimit = object.maxGasLimit ?? 0n;
        message.unlimitedGasSenders = object.unlimitedGasSenders?.map((e) => e) || [];
        return message;
    },
};
function createBaseLog() {
    return { address: "", topics: [], data: "" };
}
exports.Log = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        for (const v of message.topics) {
            writer.uint32(18).string(v);
        }
        if (message.data !== "") {
            writer.uint32(26).string(message.data);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLog();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.topics.push(reader.string());
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            topics: gt.Array.isArray(object?.topics) ? object.topics.map((e) => gt.String(e)) : [],
            data: isSet(object.data) ? gt.String(object.data) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.topics?.length) {
            obj.topics = message.topics;
        }
        if (message.data !== "") {
            obj.data = message.data;
        }
        return obj;
    },
    create(base) {
        return exports.Log.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseLog();
        message.address = object.address ?? "";
        message.topics = object.topics?.map((e) => e) || [];
        message.data = object.data ?? "";
        return message;
    },
};
function createBaseAccessTuple() {
    return { address: "", storageKeys: [] };
}
exports.AccessTuple = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        for (const v of message.storageKeys) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAccessTuple();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.storageKeys.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            address: isSet(object.address) ? gt.String(object.address) : "",
            storageKeys: gt.Array.isArray(object?.storageKeys) ? object.storageKeys.map((e) => gt.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.storageKeys?.length) {
            obj.storageKeys = message.storageKeys;
        }
        return obj;
    },
    create(base) {
        return exports.AccessTuple.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAccessTuple();
        message.address = object.address ?? "";
        message.storageKeys = object.storageKeys?.map((e) => e) || [];
        return message;
    },
};
function createBaseERC721OriginTokenInfo() {
    return { tokenOriginId: "", tokenUri: "" };
}
exports.ERC721OriginTokenInfo = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.tokenOriginId !== "") {
            writer.uint32(10).string(message.tokenOriginId);
        }
        if (message.tokenUri !== "") {
            writer.uint32(18).string(message.tokenUri);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseERC721OriginTokenInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.tokenOriginId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.tokenUri = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tokenOriginId: isSet(object.tokenOriginId) ? gt.String(object.tokenOriginId) : "",
            tokenUri: isSet(object.tokenUri) ? gt.String(object.tokenUri) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.tokenOriginId !== "") {
            obj.tokenOriginId = message.tokenOriginId;
        }
        if (message.tokenUri !== "") {
            obj.tokenUri = message.tokenUri;
        }
        return obj;
    },
    create(base) {
        return exports.ERC721OriginTokenInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseERC721OriginTokenInfo();
        message.tokenOriginId = object.tokenOriginId ?? "";
        message.tokenUri = object.tokenUri ?? "";
        return message;
    },
};
function createBaseERC721ClassInfo() {
    return { classId: "", className: "", classUri: "", classDescs: "" };
}
exports.ERC721ClassInfo = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.classId !== "") {
            writer.uint32(10).string(message.classId);
        }
        if (message.className !== "") {
            writer.uint32(18).string(message.className);
        }
        if (message.classUri !== "") {
            writer.uint32(26).string(message.classUri);
        }
        if (message.classDescs !== "") {
            writer.uint32(34).string(message.classDescs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseERC721ClassInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.classId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.className = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.classUri = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.classDescs = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            classId: isSet(object.classId) ? gt.String(object.classId) : "",
            className: isSet(object.className) ? gt.String(object.className) : "",
            classUri: isSet(object.classUri) ? gt.String(object.classUri) : "",
            classDescs: isSet(object.classDescs) ? gt.String(object.classDescs) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.classId !== "") {
            obj.classId = message.classId;
        }
        if (message.className !== "") {
            obj.className = message.className;
        }
        if (message.classUri !== "") {
            obj.classUri = message.classUri;
        }
        if (message.classDescs !== "") {
            obj.classDescs = message.classDescs;
        }
        return obj;
    },
    create(base) {
        return exports.ERC721ClassInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseERC721ClassInfo();
        message.classId = object.classId ?? "";
        message.className = object.className ?? "";
        message.classUri = object.classUri ?? "";
        message.classDescs = object.classDescs ?? "";
        return message;
    },
};
function createBaseSetCodeAuthorization() {
    return { chainId: "", address: "", nonce: 0n, signature: new Uint8Array(0) };
}
exports.SetCodeAuthorization = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.chainId !== "") {
            writer.uint32(10).string(message.chainId);
        }
        if (message.address !== "") {
            writer.uint32(18).string(message.address);
        }
        if (message.nonce !== 0n) {
            if (BigInt.asUintN(64, message.nonce) !== message.nonce) {
                throw new gt.Error("value provided for field message.nonce of type uint64 too large");
            }
            writer.uint32(24).uint64(message.nonce);
        }
        if (message.signature.length !== 0) {
            writer.uint32(34).bytes(message.signature);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetCodeAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.chainId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.nonce = reader.uint64();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            chainId: isSet(object.chainId) ? gt.String(object.chainId) : "",
            address: isSet(object.address) ? gt.String(object.address) : "",
            nonce: isSet(object.nonce) ? BigInt(object.nonce) : 0n,
            signature: isSet(object.signature) ? bytesFromBase64(object.signature) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.chainId !== "") {
            obj.chainId = message.chainId;
        }
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.nonce !== 0n) {
            obj.nonce = message.nonce.toString();
        }
        if (message.signature.length !== 0) {
            obj.signature = base64FromBytes(message.signature);
        }
        return obj;
    },
    create(base) {
        return exports.SetCodeAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetCodeAuthorization();
        message.chainId = object.chainId ?? "";
        message.address = object.address ?? "";
        message.nonce = object.nonce ?? 0n;
        message.signature = object.signature ?? new Uint8Array(0);
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
