// @initia/initia.proto/miniwasm/tokenfactory/v1/params.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: miniwasm/tokenfactory/v1/params.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
exports.protobufPackage = "miniwasm.tokenfactory.v1";
function createBaseParams() {
    return { denomCreationFee: [], denomCreationGasConsume: 0n };
}
exports.Params = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.denomCreationFee) {
            coin_1.Coin.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.denomCreationGasConsume !== 0n) {
            if (BigInt.asUintN(64, message.denomCreationGasConsume) !== message.denomCreationGasConsume) {
                throw new gt.Error("value provided for field message.denomCreationGasConsume of type uint64 too large");
            }
            writer.uint32(16).uint64(message.denomCreationGasConsume);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.denomCreationFee.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.denomCreationGasConsume = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            denomCreationFee: gt.Array.isArray(object?.denomCreationFee)
                ? object.denomCreationFee.map((e) => coin_1.Coin.fromJSON(e))
                : [],
            denomCreationGasConsume: isSet(object.denomCreationGasConsume) ? BigInt(object.denomCreationGasConsume) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.denomCreationFee?.length) {
            obj.denomCreationFee = message.denomCreationFee.map((e) => coin_1.Coin.toJSON(e));
        }
        if (message.denomCreationGasConsume !== 0n) {
            obj.denomCreationGasConsume = message.denomCreationGasConsume.toString();
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.denomCreationFee = object.denomCreationFee?.map((e) => coin_1.Coin.fromPartial(e)) || [];
        message.denomCreationGasConsume = object.denomCreationGasConsume ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
