// @initia/initia.proto/tendermint/version/types.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: tendermint/version/types.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Consensus = exports.App = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "tendermint.version";
function createBaseApp() {
    return { protocol: 0n, software: "" };
}
exports.App = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.protocol !== 0n) {
            if (BigInt.asUintN(64, message.protocol) !== message.protocol) {
                throw new gt.Error("value provided for field message.protocol of type uint64 too large");
            }
            writer.uint32(8).uint64(message.protocol);
        }
        if (message.software !== "") {
            writer.uint32(18).string(message.software);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseApp();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.protocol = reader.uint64();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.software = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            protocol: isSet(object.protocol) ? BigInt(object.protocol) : 0n,
            software: isSet(object.software) ? gt.String(object.software) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.protocol !== 0n) {
            obj.protocol = message.protocol.toString();
        }
        if (message.software !== "") {
            obj.software = message.software;
        }
        return obj;
    },
    create(base) {
        return exports.App.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseApp();
        message.protocol = object.protocol ?? 0n;
        message.software = object.software ?? "";
        return message;
    },
};
function createBaseConsensus() {
    return { block: 0n, app: 0n };
}
exports.Consensus = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.block !== 0n) {
            if (BigInt.asUintN(64, message.block) !== message.block) {
                throw new gt.Error("value provided for field message.block of type uint64 too large");
            }
            writer.uint32(8).uint64(message.block);
        }
        if (message.app !== 0n) {
            if (BigInt.asUintN(64, message.app) !== message.app) {
                throw new gt.Error("value provided for field message.app of type uint64 too large");
            }
            writer.uint32(16).uint64(message.app);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConsensus();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.block = reader.uint64();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.app = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { block: isSet(object.block) ? BigInt(object.block) : 0n, app: isSet(object.app) ? BigInt(object.app) : 0n };
    },
    toJSON(message) {
        const obj = {};
        if (message.block !== 0n) {
            obj.block = message.block.toString();
        }
        if (message.app !== 0n) {
            obj.app = message.app.toString();
        }
        return obj;
    },
    create(base) {
        return exports.Consensus.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConsensus();
        message.block = object.block ?? 0n;
        message.app = object.app ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
