// @initia/utils/dist/index.js
'use strict';

var viem = require('viem');
var utils = require('@noble/hashes/utils');
var encoding = require('@cosmjs/encoding');
var sha2 = require('@noble/hashes/sha2');
var sha3 = require('@noble/hashes/sha3');
var BigNumber = require('bignumber.js');
var bcs$1 = require('@mysten/bcs');
var ky = require('ky');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var BigNumber__default = /*#__PURE__*/_interopDefault(BigNumber);
var ky__default = /*#__PURE__*/_interopDefault(ky);

// src/address.ts
function removeLeadingZeros(hex) {
  const hasPrefix = hex.startsWith("0x");
  const cleanHex = hasPrefix ? hex.slice(2) : hex;
  const trimmed = cleanHex.replace(/^0+/, "") || "0";
  return hasPrefix ? `0x${trimmed}` : trimmed;
}
function createUserDerivedObjectAddress(source, derivedFrom) {
  if (!source) throw new Error("Source address cannot be empty");
  if (!derivedFrom) throw new Error("DerivedFrom address cannot be empty");
  const OBJECT_DERIVED_SCHEME = 252;
  const bytes = new Uint8Array([
    ...InitiaAddress(source, 32).bytes,
    ...InitiaAddress(derivedFrom, 32).bytes,
    OBJECT_DERIVED_SCHEME
  ]);
  return viem.bytesToHex(sha3.sha3_256.create().update(bytes).digest());
}
function createObjectAddress(source, seed) {
  if (!source) throw new Error("Source address cannot be empty");
  if (!seed) throw new Error("Seed cannot be empty");
  const OBJECT_FROM_SEED_ADDRESS_SCHEME = 254;
  const bytes = new Uint8Array([
    ...InitiaAddress(source, 32).bytes,
    ...utils.toBytes(seed),
    OBJECT_FROM_SEED_ADDRESS_SCHEME
  ]);
  return viem.bytesToHex(sha3.sha3_256.create().update(bytes).digest());
}
function denomToMetadata(denom) {
  if (!denom) throw new Error("Denom cannot be empty");
  if (denom.startsWith("move/")) {
    return removeLeadingZeros(`0x${denom.replace("move/", "")}`);
  }
  const objectAddress = createObjectAddress("0x1", denom);
  return removeLeadingZeros(objectAddress);
}
function getIbcDenom(channelId, denom) {
  if (!channelId) throw new Error("Channel ID cannot be empty");
  if (!denom) throw new Error("Denom cannot be empty");
  const path = `transfer/${channelId}/${denom}`;
  return `ibc/${viem.bytesToHex(sha2.sha256(path)).replace("0x", "").toUpperCase()}`;
}

// src/address.ts
var InitiaAddressImpl = class _InitiaAddressImpl {
  constructor(address, byteLength) {
    this.address = address;
    this.byteLength = byteLength;
    if (!address) {
      throw new Error("address is required");
    }
    if (!_InitiaAddressImpl.validate(address)) {
      throw new Error("invalid address");
    }
  }
  /**
   * Converts the address to a byte array representation.
   * Handles both hex and bech32 formats, padding with leading zeros as needed.
   */
  get bytes() {
    if (this.address.match(/^0x[0-9a-fA-F]+$/)) {
      const addressBytesLength = (this.address.length - 2) / 2;
      const targetLength2 = this.byteLength ?? (addressBytesLength > 20 ? 32 : 20);
      return utils.hexToBytes(
        this.address.replace("0x", "").padStart(targetLength2 * 2, "0")
      );
    }
    const { data } = encoding.fromBech32(this.address);
    const targetLength = this.byteLength ?? (data.length > 20 ? 32 : 20);
    const leadingZeros = Math.max(0, targetLength - data.length);
    return new Uint8Array([...Array(leadingZeros).fill(0), ...data]);
  }
  /**
   * Returns the bech32-encoded representation of the address.
   * Uses "init" as the prefix for all addresses.
   */
  get bech32() {
    return encoding.toBech32("init", this.bytes);
  }
  /**
   * Returns the raw hex string representation without 0x prefix or checksum.
   * Always returns lowercase hex string.
   */
  get rawHex() {
    return utils.bytesToHex(this.bytes);
  }
  /**
   * Returns the hex representation with 0x prefix.
   * For 20-byte addresses: returns checksummed address.
   * For 32-byte addresses: returns lowercase hex with leading zeros stripped.
   * For special Move addresses: removes leading zeros (e.g., 0x1).
   */
  get hex() {
    const rawHex = this.rawHex;
    if (rawHex === "0000000000000000000000000000000000000001") {
      return "0x1";
    }
    if (this.bytes.length === 32) {
      return removeLeadingZeros(`0x${rawHex}`);
    }
    return viem.getAddress(`0x${rawHex}`);
  }
  /**
   * Validates whether a string is a valid Initia address.
   * Accepts hex addresses (with or without leading zeros) and bech32 addresses.
   */
  static validate(address) {
    if (!address) {
      return false;
    }
    if (address === "0x1") {
      return true;
    }
    if (viem.isAddress(address)) {
      return true;
    }
    if (address.match(/^0x[0-9a-fA-F]+$/)) {
      const hexCharsCount = address.length - 2;
      return hexCharsCount > 0 && hexCharsCount <= 64;
    }
    try {
      return encoding.fromBech32(address).prefix === "init";
    } catch {
      return false;
    }
  }
  /**
   * Checks if two addresses are equal, regardless of format.
   * Compares the hex representation of both addresses.
   */
  static equals(address1, address2) {
    if (!address1 || !address2) {
      return address1 === address2;
    }
    if (!InitiaAddress.validate(address1) || !InitiaAddress.validate(address2)) {
      return false;
    }
    try {
      return InitiaAddress(address1).hex === InitiaAddress(address2).hex;
    } catch {
      return false;
    }
  }
};
var addressCache = /* @__PURE__ */ new Map();
function InitiaAddress(address, byteLength) {
  const normalizedAddress = address.toLowerCase();
  const cacheKey = byteLength ? `${normalizedAddress}:${byteLength}` : normalizedAddress;
  const cachedInstance = addressCache.get(cacheKey);
  if (cachedInstance) {
    return cachedInstance;
  }
  const newInstance = new InitiaAddressImpl(address, byteLength);
  addressCache.set(cacheKey, newInstance);
  return newInstance;
}
InitiaAddress.validate = InitiaAddressImpl.validate;
InitiaAddress.equals = InitiaAddressImpl.equals;
function addressSerializer(options) {
  return bcs.bytes(32, options).transform({
    input: (value) => InitiaAddress(value, 32).bytes,
    output: (value) => viem.bytesToHex(value)
  });
}
function createFixedPointSerializer(scaleFactor, bcsType) {
  return (options) => bcsType(options).transform({
    input: (val) => new BigNumber__default.default(val).times(scaleFactor).toFixed(0, BigNumber__default.default.ROUND_DOWN),
    output: (val) => new BigNumber__default.default(val).div(scaleFactor).toNumber()
  });
}
var SCALE_FACTOR_FIXED_POINT_32 = BigNumber__default.default(2).pow(32);
var SCALE_FACTOR_FIXED_POINT_64 = BigNumber__default.default(2).pow(64);
var SCALE_FACTOR_DECIMAL_128 = BigNumber__default.default(10).pow(18);
var SCALE_FACTOR_DECIMAL_256 = BigNumber__default.default(10).pow(18);
var fixedPoint32Serializer = createFixedPointSerializer(
  SCALE_FACTOR_FIXED_POINT_32,
  bcs$1.bcs.u64
);
var fixedPoint64Serializer = createFixedPointSerializer(
  SCALE_FACTOR_FIXED_POINT_64,
  bcs$1.bcs.u128
);
var decimal128Serializer = createFixedPointSerializer(
  SCALE_FACTOR_DECIMAL_128,
  bcs$1.bcs.u128
);
var decimal256Serializer = createFixedPointSerializer(
  SCALE_FACTOR_DECIMAL_256,
  bcs$1.bcs.u256
);
function bigdecimalSerializer(options) {
  return bcs.vector(bcs.u8(options)).transform({
    input: (value) => {
      const number = BigNumber__default.default(value).times(BigNumber__default.default(10).pow(18)).toFixed(0, BigNumber__default.default.ROUND_DOWN);
      return toLittleEndian(BigInt(number));
    },
    output: (val) => {
      const number = fromLittleEndian(new Uint8Array(val));
      return BigNumber__default.default(number).div(BigNumber__default.default(10).pow(18)).toFixed();
    }
  });
}
var bcs = {
  ...bcs$1.bcs,
  address: addressSerializer,
  object: addressSerializer,
  fixedPoint32: fixedPoint32Serializer,
  fixedPoint64: fixedPoint64Serializer,
  decimal128: decimal128Serializer,
  decimal256: decimal256Serializer,
  bigdecimal: bigdecimalSerializer
};
function resolveBcsType(typeStr) {
  const type = typeStr.replaceAll(/0x1::(\w+)::(\w+)/g, "$2").toLowerCase();
  if (type.startsWith("object")) {
    return bcs.object();
  }
  const genericMatch = type.match(/(\w+)<(.+)>$/);
  if (genericMatch) {
    const [, container, inner] = genericMatch;
    const innerBcs = resolveBcsType(inner);
    return bcs[container](innerBcs);
  }
  return bcs[type]();
}
function toLittleEndian(value) {
  if (value < 0n) {
    throw new Error("negative values are not supported");
  }
  if (value === 0n) {
    return new Uint8Array([0]);
  }
  const bytes = [];
  let n = value;
  while (n > 0n) {
    bytes.push(Number(n & 0xffn));
    n >>= 8n;
  }
  return new Uint8Array(bytes);
}
function fromLittleEndian(bytes) {
  let result = 0n;
  for (let i = 0; i < bytes.length; i++) {
    result += BigInt(bytes[i]) << BigInt(8 * i);
  }
  return result;
}
var ABBREVIATIONS = [
  { value: 1e12, suffix: "T" },
  { value: 1e9, suffix: "B" },
  { value: 1e6, suffix: "M" },
  { value: 1e3, suffix: "K" }
];
function toBigNumber(value) {
  if (value === null || value === void 0 || value === "") {
    return null;
  }
  try {
    const num = new BigNumber__default.default(value.toString());
    return num.isNaN() || !num.isFinite() ? null : num;
  } catch {
    return null;
  }
}
function getPowerOf10(exponent) {
  return new BigNumber__default.default(10).pow(exponent);
}
function addCommas(value) {
  const parts = value.split(".");
  const integerPart = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
  return parts[1] ? `${integerPart}.${parts[1]}` : integerPart;
}
function formatWithPrecision(num, dp, roundingMode, preserveTrailingZeros) {
  if (preserveTrailingZeros) {
    const fixed = num.toFixed(dp, roundingMode);
    return addCommas(fixed);
  }
  return num.decimalPlaces(dp, roundingMode).toFormat();
}
function formatNumber(value, options) {
  const {
    dp = 2,
    abbr = false,
    fallback = "",
    roundingMode = BigNumber__default.default.ROUND_DOWN
  } = options || {};
  const num = toBigNumber(value);
  if (!num) return fallback;
  if (num.isZero()) return "0";
  const preserveTrailingZeros = options?.dp !== void 0;
  if (!abbr) {
    return formatWithPrecision(num, dp, roundingMode, preserveTrailingZeros);
  }
  const absNum = num.abs();
  const isNegative = num.isNegative();
  for (const { value: threshold, suffix } of ABBREVIATIONS) {
    if (absNum.gte(threshold)) {
      const abbreviated = absNum.div(threshold);
      const formatted = abbreviated.toFormat(dp, roundingMode);
      return isNegative ? `-${formatted}${suffix}` : `${formatted}${suffix}`;
    }
  }
  return formatWithPrecision(num, dp, roundingMode, preserveTrailingZeros);
}
function formatAmount(value, options) {
  const {
    decimals = 0,
    dp,
    abbr,
    fallback = "",
    roundingMode = BigNumber__default.default.ROUND_DOWN
  } = options || {};
  const num = toBigNumber(value);
  if (!num) return fallback;
  if (num.isZero()) return "0";
  const result = num.div(getPowerOf10(decimals));
  const decimalPlaces = dp !== void 0 ? dp : Math.min(decimals, 6);
  const preserveTrailingZeros = dp !== void 0 || decimals > 0 && !abbr;
  if (!abbr && preserveTrailingZeros) {
    const fixed = result.toFixed(decimalPlaces, roundingMode);
    return addCommas(fixed);
  }
  return formatNumber(result, { dp: decimalPlaces, abbr, roundingMode });
}
function fromBaseUnit(value, options) {
  const {
    decimals = 0,
    fallback = "",
    roundingMode = BigNumber__default.default.ROUND_DOWN
  } = options || {};
  const num = toBigNumber(value);
  if (!num) return fallback;
  const result = num.div(getPowerOf10(decimals));
  const decimalPlaces = Math.min(decimals, 6);
  return result.decimalPlaces(decimalPlaces, roundingMode).toFixed();
}
function toBaseUnit(value, options) {
  const {
    decimals = 0,
    fallback = "",
    roundingMode = BigNumber__default.default.ROUND_DOWN
  } = options || {};
  const num = toBigNumber(value);
  if (!num) return fallback;
  const result = num.times(getPowerOf10(decimals));
  return result.integerValue(roundingMode).toFixed();
}
function formatPercent(value, options) {
  const {
    dp,
    fallback = "",
    roundingMode = BigNumber__default.default.ROUND_DOWN
  } = options || {};
  const num = toBigNumber(value);
  if (!num) return fallback;
  const percentage = num.times(100);
  const decimalPlaces = dp !== void 0 ? dp : percentage.gte(100) ? 0 : 2;
  return `${percentage.toFixed(decimalPlaces, roundingMode)}%`;
}
var createMoveClient = (restUrl) => {
  const restClient = ky__default.default.create({ prefixUrl: restUrl });
  const viewFunction = async (params) => {
    const {
      moduleAddress,
      moduleName,
      functionName,
      typeArgs = [],
      args = []
    } = params;
    const path = `initia/move/v1/accounts/${moduleAddress}/modules/${moduleName}/view_functions/${functionName}`;
    const payload = { type_args: typeArgs, args };
    const response = await restClient.post(path, { json: payload }).json();
    if (response.message) {
      throw new Error(response.message);
    }
    return JSON.parse(response.data);
  };
  return {
    viewFunction
  };
};

// src/text.ts
function truncate(str = "", lengths = [6, 6]) {
  const [prefixLength, suffixLength] = lengths;
  const minLength = prefixLength + suffixLength + 3;
  if (str.length <= minLength) {
    return str;
  }
  const prefix = str.slice(0, prefixLength);
  const suffix = str.slice(-suffixLength);
  return `${prefix}...${suffix}`;
}

exports.InitiaAddress = InitiaAddress;
exports.bcs = bcs;
exports.createMoveClient = createMoveClient;
exports.createObjectAddress = createObjectAddress;
exports.createUserDerivedObjectAddress = createUserDerivedObjectAddress;
exports.denomToMetadata = denomToMetadata;
exports.formatAmount = formatAmount;
exports.formatNumber = formatNumber;
exports.formatPercent = formatPercent;
exports.fromBaseUnit = fromBaseUnit;
exports.getIbcDenom = getIbcDenom;
exports.removeLeadingZeros = removeLeadingZeros;
exports.resolveBcsType = resolveBcsType;
exports.toBaseUnit = toBaseUnit;
exports.truncate = truncate;
