// @initia/initia.proto/initia/crypto/v1/derivable/keys.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: initia/crypto/v1/derivable/keys.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.PubKey = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "initia.crypto.v1.derivable";
function createBasePubKey() {
    return { moduleAddress: "", moduleName: "", functionName: "", abstractPublicKey: new Uint8Array(0) };
}
exports.PubKey = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.moduleAddress !== "") {
            writer.uint32(10).string(message.moduleAddress);
        }
        if (message.moduleName !== "") {
            writer.uint32(18).string(message.moduleName);
        }
        if (message.functionName !== "") {
            writer.uint32(26).string(message.functionName);
        }
        if (message.abstractPublicKey.length !== 0) {
            writer.uint32(34).bytes(message.abstractPublicKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePubKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.moduleAddress = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.moduleName = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.functionName = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.abstractPublicKey = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            moduleAddress: isSet(object.moduleAddress) ? gt.String(object.moduleAddress) : "",
            moduleName: isSet(object.moduleName) ? gt.String(object.moduleName) : "",
            functionName: isSet(object.functionName) ? gt.String(object.functionName) : "",
            abstractPublicKey: isSet(object.abstractPublicKey)
                ? bytesFromBase64(object.abstractPublicKey)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.moduleAddress !== "") {
            obj.moduleAddress = message.moduleAddress;
        }
        if (message.moduleName !== "") {
            obj.moduleName = message.moduleName;
        }
        if (message.functionName !== "") {
            obj.functionName = message.functionName;
        }
        if (message.abstractPublicKey.length !== 0) {
            obj.abstractPublicKey = base64FromBytes(message.abstractPublicKey);
        }
        return obj;
    },
    create(base) {
        return exports.PubKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePubKey();
        message.moduleAddress = object.moduleAddress ?? "";
        message.moduleName = object.moduleName ?? "";
        message.functionName = object.functionName ?? "";
        message.abstractPublicKey = object.abstractPublicKey ?? new Uint8Array(0);
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (gt.Buffer) {
        return Uint8Array.from(gt.Buffer.from(b64, "base64"));
    }
    else {
        const bin = gt.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (gt.Buffer) {
        return gt.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(gt.String.fromCharCode(byte));
        });
        return gt.btoa(bin.join(""));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
