// @initia/initia.proto/ibc/core/channel/v1/genesis.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v7.34.1
// source: ibc/core/channel/v1/genesis.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.PacketSequence = exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const channel_1 = require("./channel");
exports.protobufPackage = "ibc.core.channel.v1";
function createBaseGenesisState() {
    return {
        channels: [],
        acknowledgements: [],
        commitments: [],
        receipts: [],
        sendSequences: [],
        recvSequences: [],
        ackSequences: [],
        nextChannelSequence: 0n,
        params: undefined,
    };
}
exports.GenesisState = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.channels) {
            channel_1.IdentifiedChannel.encode(v, writer.uint32(10).fork()).join();
        }
        for (const v of message.acknowledgements) {
            channel_1.PacketState.encode(v, writer.uint32(18).fork()).join();
        }
        for (const v of message.commitments) {
            channel_1.PacketState.encode(v, writer.uint32(26).fork()).join();
        }
        for (const v of message.receipts) {
            channel_1.PacketState.encode(v, writer.uint32(34).fork()).join();
        }
        for (const v of message.sendSequences) {
            exports.PacketSequence.encode(v, writer.uint32(42).fork()).join();
        }
        for (const v of message.recvSequences) {
            exports.PacketSequence.encode(v, writer.uint32(50).fork()).join();
        }
        for (const v of message.ackSequences) {
            exports.PacketSequence.encode(v, writer.uint32(58).fork()).join();
        }
        if (message.nextChannelSequence !== 0n) {
            if (BigInt.asUintN(64, message.nextChannelSequence) !== message.nextChannelSequence) {
                throw new gt.Error("value provided for field message.nextChannelSequence of type uint64 too large");
            }
            writer.uint32(64).uint64(message.nextChannelSequence);
        }
        if (message.params !== undefined) {
            channel_1.Params.encode(message.params, writer.uint32(74).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.channels.push(channel_1.IdentifiedChannel.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.acknowledgements.push(channel_1.PacketState.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.commitments.push(channel_1.PacketState.decode(reader, reader.uint32()));
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.receipts.push(channel_1.PacketState.decode(reader, reader.uint32()));
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.sendSequences.push(exports.PacketSequence.decode(reader, reader.uint32()));
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.recvSequences.push(exports.PacketSequence.decode(reader, reader.uint32()));
                    continue;
                }
                case 7: {
                    if (tag !== 58) {
                        break;
                    }
                    message.ackSequences.push(exports.PacketSequence.decode(reader, reader.uint32()));
                    continue;
                }
                case 8: {
                    if (tag !== 64) {
                        break;
                    }
                    message.nextChannelSequence = reader.uint64();
                    continue;
                }
                case 9: {
                    if (tag !== 74) {
                        break;
                    }
                    message.params = channel_1.Params.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            channels: gt.Array.isArray(object?.channels)
                ? object.channels.map((e) => channel_1.IdentifiedChannel.fromJSON(e))
                : [],
            acknowledgements: gt.Array.isArray(object?.acknowledgements)
                ? object.acknowledgements.map((e) => channel_1.PacketState.fromJSON(e))
                : [],
            commitments: gt.Array.isArray(object?.commitments)
                ? object.commitments.map((e) => channel_1.PacketState.fromJSON(e))
                : [],
            receipts: gt.Array.isArray(object?.receipts) ? object.receipts.map((e) => channel_1.PacketState.fromJSON(e)) : [],
            sendSequences: gt.Array.isArray(object?.sendSequences)
                ? object.sendSequences.map((e) => exports.PacketSequence.fromJSON(e))
                : [],
            recvSequences: gt.Array.isArray(object?.recvSequences)
                ? object.recvSequences.map((e) => exports.PacketSequence.fromJSON(e))
                : [],
            ackSequences: gt.Array.isArray(object?.ackSequences)
                ? object.ackSequences.map((e) => exports.PacketSequence.fromJSON(e))
                : [],
            nextChannelSequence: isSet(object.nextChannelSequence) ? BigInt(object.nextChannelSequence) : 0n,
            params: isSet(object.params) ? channel_1.Params.fromJSON(object.params) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.channels?.length) {
            obj.channels = message.channels.map((e) => channel_1.IdentifiedChannel.toJSON(e));
        }
        if (message.acknowledgements?.length) {
            obj.acknowledgements = message.acknowledgements.map((e) => channel_1.PacketState.toJSON(e));
        }
        if (message.commitments?.length) {
            obj.commitments = message.commitments.map((e) => channel_1.PacketState.toJSON(e));
        }
        if (message.receipts?.length) {
            obj.receipts = message.receipts.map((e) => channel_1.PacketState.toJSON(e));
        }
        if (message.sendSequences?.length) {
            obj.sendSequences = message.sendSequences.map((e) => exports.PacketSequence.toJSON(e));
        }
        if (message.recvSequences?.length) {
            obj.recvSequences = message.recvSequences.map((e) => exports.PacketSequence.toJSON(e));
        }
        if (message.ackSequences?.length) {
            obj.ackSequences = message.ackSequences.map((e) => exports.PacketSequence.toJSON(e));
        }
        if (message.nextChannelSequence !== 0n) {
            obj.nextChannelSequence = message.nextChannelSequence.toString();
        }
        if (message.params !== undefined) {
            obj.params = channel_1.Params.toJSON(message.params);
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.channels = object.channels?.map((e) => channel_1.IdentifiedChannel.fromPartial(e)) || [];
        message.acknowledgements = object.acknowledgements?.map((e) => channel_1.PacketState.fromPartial(e)) || [];
        message.commitments = object.commitments?.map((e) => channel_1.PacketState.fromPartial(e)) || [];
        message.receipts = object.receipts?.map((e) => channel_1.PacketState.fromPartial(e)) || [];
        message.sendSequences = object.sendSequences?.map((e) => exports.PacketSequence.fromPartial(e)) || [];
        message.recvSequences = object.recvSequences?.map((e) => exports.PacketSequence.fromPartial(e)) || [];
        message.ackSequences = object.ackSequences?.map((e) => exports.PacketSequence.fromPartial(e)) || [];
        message.nextChannelSequence = object.nextChannelSequence ?? 0n;
        message.params = (object.params !== undefined && object.params !== null)
            ? channel_1.Params.fromPartial(object.params)
            : undefined;
        return message;
    },
};
function createBasePacketSequence() {
    return { portId: "", channelId: "", sequence: 0n };
}
exports.PacketSequence = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.portId !== "") {
            writer.uint32(10).string(message.portId);
        }
        if (message.channelId !== "") {
            writer.uint32(18).string(message.channelId);
        }
        if (message.sequence !== 0n) {
            if (BigInt.asUintN(64, message.sequence) !== message.sequence) {
                throw new gt.Error("value provided for field message.sequence of type uint64 too large");
            }
            writer.uint32(24).uint64(message.sequence);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePacketSequence();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.portId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.channelId = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.sequence = reader.uint64();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            portId: isSet(object.portId) ? gt.String(object.portId) : "",
            channelId: isSet(object.channelId) ? gt.String(object.channelId) : "",
            sequence: isSet(object.sequence) ? BigInt(object.sequence) : 0n,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.portId !== "") {
            obj.portId = message.portId;
        }
        if (message.channelId !== "") {
            obj.channelId = message.channelId;
        }
        if (message.sequence !== 0n) {
            obj.sequence = message.sequence.toString();
        }
        return obj;
    },
    create(base) {
        return exports.PacketSequence.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePacketSequence();
        message.portId = object.portId ?? "";
        message.channelId = object.channelId ?? "";
        message.sequence = object.sequence ?? 0n;
        return message;
    },
};
const gt = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function isSet(value) {
    return value !== null && value !== undefined;
}
